{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "50e6a2c4",
   "metadata": {},
   "source": [
    "# Literature Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "f2050b85",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "from Bio import Medline\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "from Bio.Blast import NCBIWWW\n",
    "from Bio import Entrez, SeqIO\n",
    "from Bio.Blast import NCBIXML\n",
    "from Bio import SearchIO\n",
    "import requests,sys,json\n",
    "from Bio.SwissProt import KeyWList\n",
    "from Bio import SwissProt\n",
    "from Bio import ExPASy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1d5f7759",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "database: PubMed\n",
      "query a procurar: DDX18\n",
      "resultados maximos: 30\n",
      "email a inserir: guilherme.lobo@ua.pt\n",
      "title: DDX18 prevents R-loop-induced DNA damage and genome instability via PARP-1.\n",
      "authors: ['Lin WL', 'Chen JK', 'Wen X', 'He W', 'Zarceno GA', 'Chen Y', 'Chen S', 'Paull TT', 'Liu HW']\n",
      "source: Cell Rep. 2022 Jul 19;40(3):111089. doi: 10.1016/j.celrep.2022.111089.\n",
      "\n",
      "title: Identification of Important Modules and Hub Gene in Chronic Kidney Disease Based on WGCNA.\n",
      "authors: ['Wang J', 'Yin Y', 'Lu Q', 'Zhao YR', 'Hu YJ', 'Hu YZ', 'Wang ZY']\n",
      "source: J Immunol Res. 2022 May 4;2022:4615292. doi: 10.1155/2022/4615292. eCollection 2022.\n",
      "\n",
      "title: The RNA-Binding Protein DDX18 Promotes Gastric Cancer by Affecting the Maturation of MicroRNA-21.\n",
      "authors: ['Zhang Y', 'Yu F', 'Ni B', 'Li Q', 'Bae SW', 'Choi JH', 'Yang HK', 'Kong SH', 'Zhu C']\n",
      "source: Front Oncol. 2021 Jan 8;10:598238. doi: 10.3389/fonc.2020.598238. eCollection 2020.\n",
      "\n",
      "title: Identification of Key Modules and Hub Genes Involved in Esophageal Squamous Cell Carcinoma Tumorigenesis Using WCGNA.\n",
      "authors: ['Chen N', 'Zhang G', 'Fu J', 'Wu Q']\n",
      "source: Cancer Control. 2020 Jan-Dec;27(1):1073274820978817. doi: 10.1177/1073274820978817.\n",
      "\n",
      "title: Identification of the Significant Genes Regulated by Estrogen Receptor in Estrogen Receptor-Positive Breast Cancer and Their Expression Pattern Changes When Tamoxifen or Fulvestrant Resistance Occurs.\n",
      "authors: ['Cheng R', 'Qi L', 'Kong X', 'Wang Z', 'Fang Y', 'Wang J']\n",
      "source: Front Genet. 2020 Sep 29;11:538734. doi: 10.3389/fgene.2020.538734. eCollection 2020.\n",
      "\n",
      "title: Identification of key gene modules and genes in colorectal cancer by co-expression analysis weighted gene co-expression network analysis.\n",
      "authors: ['Wang P', 'Zheng H', 'Zhang J', 'Wang Y', 'Liu P', 'Xuan X', 'Li Q', 'Du Y']\n",
      "source: Biosci Rep. 2020 Sep 30;40(9):BSR20202044. doi: 10.1042/BSR20202044.\n",
      "\n",
      "title: DEAD-Box Helicase 18 Counteracts PRC2 to Safeguard Ribosomal DNA in Pluripotency Regulation.\n",
      "authors: ['Zhang H', 'Wu Z', 'Lu JY', 'Huang B', 'Zhou H', 'Xie W', 'Wang J', 'Shen X']\n",
      "source: Cell Rep. 2020 Jan 7;30(1):81-97.e7. doi: 10.1016/j.celrep.2019.12.021.\n",
      "\n",
      "title: Genome-wide scan identifies opioid overdose risk locus close to MCOLN1.\n",
      "authors: ['Cheng Z', 'Yang BZ', 'Zhou H', 'Nunez Y', 'Kranzler HR', 'Gelernter J']\n",
      "source: Addict Biol. 2020 Mar;25(2):e12811. doi: 10.1111/adb.12811. Epub 2019 Jul 30.\n",
      "\n",
      "title: Integrated network analysis and machine learning approach for the identification of key genes of triple-negative breast cancer.\n",
      "authors: ['Naorem LD', 'Muthaiyan M', 'Venkatesan A']\n",
      "source: J Cell Biochem. 2019 Apr;120(4):6154-6167. doi: 10.1002/jcb.27903. Epub 2018 Oct 9.\n",
      "\n",
      "title: PNPLA3 and TM6SF2 variants as risk factors of hepatocellular carcinoma across various etiologies and severity of underlying liver diseases.\n",
      "authors: ['Yang J', 'Trepo E', 'Nahon P', 'Cao Q', 'Moreno C', 'Letouze E', 'Imbeaud S', 'Gustot T', 'Deviere J', 'Debette S', 'Amouyel P', 'Bioulac-Sage P', 'Calderaro J', 'Ganne-Carrie N', 'Laurent A', 'Blanc JF', 'Guyot E', 'Sutton A', 'Ziol M', 'Zucman-Rossi J', 'Nault JC']\n",
      "source: Int J Cancer. 2019 Feb 1;144(3):533-544. doi: 10.1002/ijc.31910. Epub 2018 Nov 9.\n",
      "\n",
      "title: [Data mining of esophageal squamous cell carcinoma from The Cancer Genome Atlas database].\n",
      "authors: ['He SY', 'Wang XB', 'Jiao YC']\n",
      "source: Zhonghua Zhong Liu Za Zhi. 2018 Jul 23;40(7):517-522. doi: 10.3760/cma.j.issn.0253-3766.2018.07.007.\n",
      "\n",
      "title: Cellular DEAD-box RNA helicase 18 (DDX18) Promotes the PRRSV Replication via Interaction with Virus nsp2 and nsp10.\n",
      "authors: ['Jin H', 'Zhou L', 'Ge X', 'Zhang H', 'Zhang R', 'Wang C', 'Wang L', 'Zhang Z', 'Yang H', 'Guo X']\n",
      "source: Virus Res. 2017 Jun 15;238:204-212. doi: 10.1016/j.virusres.2017.05.028. Epub 2017 Jun 22.\n",
      "\n",
      "title: Genomic interaction between ER and HMGB2 identifies DDX18 as a novel driver of endocrine resistance in breast cancer cells.\n",
      "authors: ['Redmond AM', 'Byrne C', 'Bane FT', 'Brown GD', 'Tibbitts P', \"O'Brien K\", 'Hill AD', 'Carroll JS', 'Young LS']\n",
      "source: Oncogene. 2015 Jul;34(29):3871-80. doi: 10.1038/onc.2014.323. Epub 2014 Oct 6.\n",
      "\n",
      "title: Ddx18 is essential for cell-cycle progression in zebrafish hematopoietic cells and is mutated in human AML.\n",
      "authors: ['Payne EM', 'Bolli N', 'Rhodes J', 'Abdel-Wahab OI', 'Levine R', 'Hedvat CV', 'Stone R', 'Khanna-Gupta A', 'Sun H', 'Kanki JP', 'Gazda HT', 'Beggs AH', 'Cotter FE', 'Look AT']\n",
      "source: Blood. 2011 Jul 28;118(4):903-15. doi: 10.1182/blood-2010-11-318022. Epub 2011 Jun 7.\n",
      "\n",
      "title: Comparative structural analysis of human DEAD-box RNA helicases.\n",
      "authors: ['Schutz P', 'Karlberg T', 'van den Berg S', 'Collins R', 'Lehtio L', 'Hogbom M', 'Holmberg-Schiavone L', 'Tempel W', 'Park HW', 'Hammarstrom M', 'Moche M', 'Thorsell AG', 'Schuler H']\n",
      "source: PLoS One. 2010 Sep 30;5(9):e12791. doi: 10.1371/journal.pone.0012791.\n",
      "\n",
      "title: Enriching protein-protein and functional interaction networks in human embryonic stem cells.\n",
      "authors: ['Zuo C', 'Liang S', 'Wang Z', 'Li H', 'Zheng W', 'Ma W']\n",
      "source: Int J Mol Med. 2009 Jun;23(6):811-9. doi: 10.3892/ijmm_00000197.\n",
      "\n",
      "title: Cellular studies of MrDb (DDX18).\n",
      "authors: ['Dubaele S', 'Chene P']\n",
      "source: Oncol Res. 2007;16(12):549-56. doi: 10.3727/000000007783630015.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#modo de apesentação geral para a nossa pesquisa com artigos exemplo\n",
    "#na minha opinião apenas mostra que dá para fazer uma pesquisa automática, contudo sempre que trabalhei com artigas envolvia uma leitura prévia e não apenas escolher os primeiros que aparecem com os filtros\n",
    "\n",
    "database = input('database: ')#For literature will be PubMed\n",
    "word = input('query a procurar: ') #I put ddx18\n",
    "res= int(input('resultados maximos: '))#I choose a limit of thirty\n",
    "email= \"guilherme.lobo@ua.pt\"\n",
    "\n",
    "Entrez.email= email\n",
    "handle=Entrez.esearch(db = database, term=word, retmax= res)\n",
    "record=Entrez.read(handle)\n",
    "handle.close()\n",
    "idlist= record['IdList']\n",
    "\n",
    "handle = Entrez.efetch(db=database, id=idlist, rettype=\"medline\", retmode=\"text\") #para este caso em especifico devemos ter como database a PubMed(\"Medline\")\n",
    "records = Medline.parse(handle)\n",
    "\n",
    "\n",
    "for record in records:\n",
    "    #print(record) \n",
    "    #print(\"PMID:\",record.get(\"PMID\",\"-\"))\n",
    "    print(\"title:\", record.get(\"TI\", \"-\"))\n",
    "    #print('abstract:', record.get('AB', '-'))\n",
    "    print(\"authors:\", record.get(\"AU\", \"-\"))\n",
    "    print(\"source:\", record.get(\"SO\", \"-\"))\n",
    "    print(\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e57a99e8",
   "metadata": {},
   "source": [
    "-Same thing as above with the difference that we are the ones who have to make a prior choice of articles and download a file accordingly-(pubmed-irs-1ANDdi-set.txt)\n",
    "-We think that this way is more suitable, despite not being completely automated, because when choosing articles it is always necessary to make an adequate pre-selection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "627372e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "6it [00:00, 2523.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       PMID                                              Title  \\\n",
      "1  15727801  IRS-1 and vascular complications in diabetes m...   \n",
      "2  14584587            IRS-1 regulation in health and disease.   \n",
      "3  26704265  Novel compounds reducing IRS-1 serine phosphor...   \n",
      "4  12169433      IRS proteins and the common path to diabetes.   \n",
      "5  10593864  Low cellular IRS 1 gene and protein expression...   \n",
      "6  11272176               IRS proteins and beta-cell function.   \n",
      "\n",
      "                                            Abstract  \\\n",
      "1  The expected explosive increase in the number ...   \n",
      "2  The global incidence of diabetes is increasing...   \n",
      "3  Activation of various interacting stress kinas...   \n",
      "4  Although a full understanding of insulin/insul...   \n",
      "5  We examined the gene and protein expression of...   \n",
      "6  Insulin receptor substrate (IRS) proteins medi...   \n",
      "\n",
      "                                             Authors  \\\n",
      "1                  [Andrade Ferreira I, Akkerman JW]   \n",
      "2                  [Schmitz-Peiffer C, Whitehead JP]   \n",
      "3  [Simon-Szabó L, Kokas M, Greff Z, Boros S, Bán...   \n",
      "4                                         [White MF]   \n",
      "5  [Carvalho E, Jansson PA, Axelsen M, Eriksson J...   \n",
      "6                               [Burks DJ, White MF]   \n",
      "\n",
      "                                              Source  \n",
      "1  Vitam Horm. 2005;70:25-67. doi: 10.1016/S0083-...  \n",
      "2  IUBMB Life. 2003 Jul;55(7):367-74. doi: 10.108...  \n",
      "3  Bioorg Med Chem Lett. 2016 Jan 15;26(2):424-42...  \n",
      "4  Am J Physiol Endocrinol Metab. 2002 Sep;283(3)...  \n",
      "5  FASEB J. 1999 Dec;13(15):2173-8. doi: 10.1096/...  \n",
      "6  Diabetes. 2001 Feb;50 Suppl 1:S140-5. doi: 10....  \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "alldata=[]\n",
    "with open(\"pubmed-irs-1ANDdi-set.txt\",encoding=\"utf-8\") as f: #For this examples I chosse articles from PubMed about Diabetes type II\n",
    "    pmids= Medline.parse(f)\n",
    "    for pmid in tqdm(pmids):\n",
    "        try:\n",
    "            pid=pmid[\"PMID\"]\n",
    "        except:\n",
    "            pid=\"-\"\n",
    "        try:\n",
    "            Title=pmid[\"TI\"]\n",
    "        except:\n",
    "            Title=\"-\"\n",
    "        try:        \n",
    "            Abstract=pmid[\"AB\"]\n",
    "        except:\n",
    "            Abstract=\"-\"\n",
    "        try:\n",
    "            Authors=pmid[\"AU\"]\n",
    "        except:\n",
    "            Authors=\"-\"\n",
    "        try:\n",
    "            Source=pmid[\"SO\"]\n",
    "        except:\n",
    "            Source=\"-\"\n",
    "        \n",
    "        dic={\"PMID\":pid,\n",
    "             \"Title\":Title,\n",
    "             \"Abstract\":Abstract,\n",
    "             \"Authors\":Authors,\n",
    "             \"Source\":Source\n",
    "            }\n",
    "        alldata.append(dic)\n",
    "#print(alldata)\n",
    "df = pd.DataFrame.from_records(alldata,index=['1', '2',\"3\",\"4\",\"5\",\"6\"])\n",
    "print(df)\n",
    "#writer = pd.ExcelWriter('articles_diabetes.xlsx')\n",
    "#df.to_Exel(writer)\n",
    "#writer.save()\n",
    "#df=pd.DataFrame(alldata)\n",
    "#df.to_csv(\"articles.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc561e03",
   "metadata": {},
   "source": [
    "# Analysis of the sequence and features present in the NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54b6f5ae",
   "metadata": {},
   "source": [
    "## DDX18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7d9b3c58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "database: nucleotide\n",
      "query a procurar: ddx18 and homo sapiens and Chromosome 2\n",
      "resultados maximos: 30\n",
      "NM_006773.4 Homo sapiens DEAD-box helicase 18 (DDX18), mRNA\n",
      "NC_060926.1 Homo sapiens isolate CHM13 chromosome 2, alternate assembly T2T-CHM13v2.0\n",
      "NC_000002.12 Homo sapiens chromosome 2, GRCh38.p14 Primary Assembly\n",
      "AC009312.4 Homo sapiens BAC clone RP11-425F6 from 2, complete sequence\n",
      "AC009404.5 Homo sapiens BAC clone RP11-28H22 from 2, complete sequence\n",
      "CM000253.1 Homo sapiens chromosome 2, whole genome shotgun sequence\n",
      "CH471103.1 Homo sapiens 211000035839809 genomic scaffold, whole genome shotgun sequence\n",
      "DQ655976.2 Homo sapiens clone Affy08254D01, mRNA sequence\n",
      "DQ655975.2 Homo sapiens clone Affy08248B12, mRNA sequence\n",
      "NT_086328.2 Homo sapiens chromosome 2 sequence, ENCODE region ENr121\n"
     ]
    }
   ],
   "source": [
    "database = input('database: ')#nucleotide tem de ser pelos vistos\n",
    "word = input('query a procurar: ')#ddx18 and homo sapiens and Chromosome 2\n",
    "res= int(input('resultados maximos: '))#30\n",
    "Entrez.email=\"guilherme.lobo@ua.pt\"\n",
    "handle=Entrez.esearch(db = database, term=word, retmax= res)\n",
    "record=Entrez.read(handle)\n",
    "handle.close()\n",
    "idlist= record['IdList']\n",
    "\n",
    "handle = Entrez.efetch(db=database, id=idlist, rettype=\"gb\") \n",
    "records = list(SeqIO.parse(handle,\"gb\"))\n",
    "handle.close()\n",
    "for x in records:\n",
    "    print(x.id,x.description)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c39e60f8",
   "metadata": {},
   "source": [
    "Taking into account the description given, we think that the id (\"NM_006773.4\") would be the most appropriate and, as such, will be used from now on."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "ed443da9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Record sequence length:  3753\n",
      "Record features count:  27\n",
      "Nuclleotide sequence: ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCACTTGTTGGGCAGAATGTCACACCTGCCGATGAAACTCCTGCGTAAGAAGATCGAGAAGCGGAACCTCAAATTGCGGCAGCGGAACCTAAAGTTTCAGGGGGCCTCAAATCTGACCCTATCGGAAACTCAAAATGGAGATGTATCTGAAGAAACAATGGGAAGTAGAAAGGTTAAAAAATCAAAACAAAAGCCCATGAATGTGGGCTTATCAGAAACTCAAAATGGAGGCATGTCTCAAGAAGCAGTGGGAAATATAAAAGTTACAAAGTCTCCCCAGAAATCCACTGTATTAACCAATGGAGAAGCAGCAATGCAGTCTTCCAATTCAGAATCAAAAAAGAAAAAGAAGAAAAAGAGAAAAATGGTGAATGATGCTGAGCCTGATACGAAAAAAGCAAAAACTGAAAACAAAGGGAAATCTGAAGAAGAAAGTGCCGAGACTACTAAAGAAACAGAAAATAATGTGGAGAAGCCAGATAATGATGAAGATGAGAGTGAGGTGCCCAGTCTGCCCCTGGGACTGACAGGAGCTTTTGAGGATACTTCGTTTGCTTCTCTATGTAATCTTGTCAATGAAAACACTCTGAAGGCAATAAAAGAAATGGGTTTTACAAACATGACTGAAATTCAGCATAAAAGTATCAGACCACTTCTGGAAGGCAGGGATCTTCTAGCAGCTGCAAAAACAGGCAGTGGTAAAACCCTGGCTTTTCTCATCCCTGCAGTTGAACTCATTGTTAAGTTAAGGTTCATGCCCAGGAATGGAACAGGAGTCCTTATTCTCTCACCTACTAGAGAACTAGCCATGCAAACCTTTGGTGTTCTTAAGGAGCTGATGACTCACCACGTGCATACCTATGGCTTGATAATGGGTGGCAGTAACAGATCTGCTGAAGCACAGAAACTTGGTAATGGGATCAACATCATTGTGGCCACACCAGGCCGTCTGCTGGACCATATGCAGAATACCCCAGGATTTATGTATAAAAACCTGCAGTGTCTGGTTATTGATGAAGCTGATCGTATCTTGGATGTGGGGTTTGAAGAGGAATTAAAGCAAATTATTAAACTTTTGCCAACACGTAGACAGACTATGCTCTTTTCTGCCACCCAAACTCGAAAAGTTGAAGACCTGGCAAGGATTTCTCTGAAAAAGGAGCCATTGTATGTTGGCGTTGATGATGATAAAGCGAATGCAACAGTGGATGGTCTTGAACAGGGATATGTTGTTTGTCCTTCTGAAAAGAGATTCCTTCTGCTCTTTACATTCCTTAAGAAGAACCGAAAGAAGAAGCTTATGGTCTTCTTTTCATCTTGTATGTCTGTGAAATACCACTATGAGTTGCTGAACTACATTGATTTGCCCGTCTTGGCCATTCATGGAAAGCAAAAGCAAAATAAGCGTACAACCACATTCTTCCAGTTCTGCAATGCAGATTCGGGAACACTATTGTGTACGGATGTGGCAGCGAGAGGACTAGACATTCCTGAAGTCGACTGGATTGTTCAGTATGACCCTCCGGATGACCCTAAGGAATATATTCATCGTGTGGGTAGAACAGCCAGAGGCCTAAATGGGAGAGGGCATGCCTTGCTCATTTTGCGCCCAGAAGAATTGGGTTTTCTTCGCTACTTGAAACAATCCAAGGTTCCATTAAGTGAATTTGACTTTTCCTGGTCTAAAATTTCTGACATTCAGTCTCAGCTTGAGAAATTGATTGAAAAGAATTACTTTCTTCATAAGTCAGCCCAGGAAGCATATAAGTCATACATACGAGCCTATGATTCCCATTCTCTGAAACAGATCTTTAATGTTAATAACCTAAATTTGCCTCAGGTTGCTCTGTCATTTGGTTTCAAGGTGCCTCCCTTCGTTGATCTGAACGTCAACAGTAATGAAGGCAAGCAGAAAAAGCGAGGAGGTGGTGGTGGATTTGGCTACCAGAAAACCAAGAAAGTTGAGAAATCCAAAATCTTTAAACACATTAGCAAGAAATCATCTGACAGCAGGCAGTTCTCTCACTGAACACATGCCTTCCTTTCATCTTGAATAACTTTGTCCTAAAATGAATTTTTTTTCCCCTTGATTTAACAGGATTTTTGTAGACTTTAGAATTTGGACTTACCTAACAAGAGTATAAATTGACTTGGGTTGCAAGCACTGAGCACTGTTACTTCTATCACGTCTCTCTTTTATTTCTGGGATATAAAACAGGCTTTAAGTTTCTTGGTTGCCCAAGGGCAGAGCAAGGAATATCTGGTGTTTCTTGTGATGATAATATTTTAATTTTAAATATCCCTCCCTCATACAAGTGTATGTTACCATTTTAATATAATTCTTTTTGTACCTTTCCTTCTTGTTTTGCGAAGATTTTTGTGGCATGGATTGCTGTGCTCACTGCTGTAAAAGGTGACCTAGTGTACTGGGCAGCTGGTGGCGGTGCAGAAAAGAGTCTCAGGTTATTTTTTGTTTTTAGTTATTTCTTGGACCTTGACAGTATCTAATGACTCCTCCTGAAAATGCTGCAGTATAAAAGAGCAAAGAGCTTTGGGAAATACCTAAGAAGCACCTTAAGATTAGGGTGGCATTGCTTTTATAGATTCTTGATTTTAAAGCAACAGGCCTTTCTCAGGTGTTGCATTTTTTGGAGCAAAAACTATGGGTTGTAATTTGAATAAAGTGTCACTAAGCAGTTATAACGTTTGATGGCTGGGGGGTAGGAAGAGGATGGAATTGAGATGTTTGAGCCTCATTTACATCAATAGAGGTGTAATGTACTGCATTTCTTCATTTGGTAACATAACAAAGACTTTCATACAAAGAACGATGATGCTCCTCATTAAGATTTGTTTAATTCAAGGTGGTTTGGATTTGGTAAGCCTTTGCACTCTGTAGAGTACTTAGAAGACAAGGGCAACTTACTTGGAGTTAGAGCCAAGCTGTCAGACGGTGCCCAGCACACATTAATGTTAGCTTCTTTCTGAGAAAAAAATACCTCTTCCAGGCCCTGAAACAAAAAATACATTTGCTGTGAAGATTGAAAATGAACAAAGTTAGAAAAAAAAACAGCAAAATCAGTGATTTAGTCAGATGAGTTTTTCGTTGTAGGAGCACTTGATTTCTAGTGTGTTTTGTACAGTATATAACTACAAGATAGTACATTTTGTAGCAGTTCAAAGCCAAAGTTGCTAGCATCATTTTGCTGTTGTGCCAGTTAATCATAGGATCCCATTAAATAAGTGTGCTAACATCGAATATAGAGAAAACTGGTAAAGAACATTCCAGTAGGAAAAGAAAAGAACAATCTTCCATTTCTGGGCTTGGCCACCATCACCCTGGTCGGACCTGTCCTGGACTTCCAACCTTGACTGCTGAGCTCCTGGCTTAGCTTCTTGGGTTCCTAATTCCTGGTGTTTAATAATTCTCTCCACGATCATGTTTTTCTGATTTTTTTTTTCAGAAATAATGTTTTTTAAAAGACAAAAACAAAGGGAAGAATATTTAATTACTGAGCAGAAGTAAATACTGTTGGCATTTTGTACATAATCTAATTTTTATATGCATGTTCATGCTTTTTAATTTTTTTATCAAAAATTAAGTCATCTACCTACTACTTGTAACCAGCTTGTTTCATAACATGTTATTTTCCTGTGTCATTAAATAATTACTTCAATGTTG\n",
      "ddx18 gene synonyms: ['Has1; MrDb']\n",
      "name: ['ATP-dependent RNA helicase DDX18']\n",
      "Protein sequence: ['MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPTRELAMQTFGVLKELMTHHVHTYGLIMGGSNRSAEAQKLGNGINIIVATPGRLLDHMQNTPGFMYKNLQCLVIDEADRILDVGFEEELKQIIKLLPTRRQTMLFSATQTRKVEDLARISLKKEPLYVGVDDDKANATVDGLEQGYVVCPSEKRFLLLFTFLKKNRKKKLMVFFSSCMSVKYHYELLNYIDLPVLAIHGKQKQNKRTTTFFQFCNADSGTLLCTDVAARGLDIPEVDWIVQYDPPDDPKEYIHRVGRTARGLNGRGHALLILRPEELGFLRYLKQSKVPLSEFDFSWSKISDIQSQLEKLIEKNYFLHKSAQEAYKSYIRAYDSHSLKQIFNVNNLNLPQVALSFGFKVPPFVDLNVNSNEGKQKKRGGGGGFGYQKTKKVEKSKIFKHISKKSSDSRQFSH']\n",
      "Location: [0:3753](+)\n",
      "Gene: ['DDX18']\n"
     ]
    }
   ],
   "source": [
    "for rec in records:\n",
    "    #print(rec)\n",
    "    if rec.id == \"NM_006773.4\":\n",
    "        print(\"Record sequence length: \", len(rec.seq))\n",
    "        print(\"Record features count: \", len(rec.features))\n",
    "        print( \"Nuclleotide sequence:\", str(rec.seq)) #useful for blastn\n",
    "        \n",
    "        cds = []\n",
    "        for i,feature in enumerate(rec.features):\n",
    "            if feature.type==\"CDS\":\n",
    "                cds.append(i)\n",
    "                for k in cds:\n",
    "                    print(\"ddx18 gene synonyms:\",rec.features[k].qualifiers[\"gene_synonym\"])#can help latter in the search\n",
    "                    print(\"name:\",rec.features[k].qualifiers[\"product\"])\n",
    "                    sequence_protein=rec.features[k].qualifiers[\"translation\"]#get the protein seq of my cds\n",
    "                    print(\"Protein sequence:\",sequence_protein)\n",
    "        gene = []\n",
    "        for j in range(len(rec.features)):\n",
    "            if rec.features[j].type == \"gene\":\n",
    "                gene.append(j)\n",
    "                for g in gene:\n",
    "                    print(\"Location:\",rec.features[g].location) \n",
    "                    print(\"Gene:\",rec.features[g].qualifiers[\"gene\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "57b0950e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPTRELAMQTFGVLKELMTHHVHTYGLIMGGSNRSAEAQKLGNGINIIVATPGRLLDHMQNTPGFMYKNLQCLVIDEADRILDVGFEEELKQIIKLLPTRRQTMLFSATQTRKVEDLARISLKKEPLYVGVDDDKANATVDGLEQGYVVCPSEKRFLLLFTFLKKNRKKKLMVFFSSCMSVKYHYELLNYIDLPVLAIHGKQKQNKRTTTFFQFCNADSGTLLCTDVAARGLDIPEVDWIVQYDPPDDPKEYIHRVGRTARGLNGRGHALLILRPEELGFLRYLKQSKVPLSEFDFSWSKISDIQSQLEKLIEKNYFLHKSAQEAYKSYIRAYDSHSLKQIFNVNNLNLPQVALSFGFKVPPFVDLNVNSNEGKQKKRGGGGGFGYQKTKKVEKSKIFKHISKKSSDSRQFSH'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_p_ddx18=\"\".join(sequence_protein)\n",
    "seq_p_ddx18\n",
    "#this is to put my sequence protein of cds in a string "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8008dae",
   "metadata": {},
   "source": [
    "# Homology analysis by BLAST or Diamond"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7734736",
   "metadata": {},
   "source": [
    "## DDX18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "0105965a",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle=NCBIWWW.qblast(\"blastn\",\"nt\",rec.seq)#fazer o blast como tem no tuturial.pdf\n",
    "with open(\"blast_file_ddx18.txt\",\"w\") as out_handle: #guardar neste ficheirotenho em txt, e xml\n",
    "    out_handle.write(result_handle.read())\n",
    "result_handle.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "080e3130",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exist 50 alignments in the register\n",
      "gi|1519243738|ref|NM_006773.4| Homo sapiens DEAD-box helicase 18 (DDX18), mRNA\n",
      "           Length = 3753\n",
      "\n"
     ]
    }
   ],
   "source": [
    "Blast=open(\"blast_file_ddx18.txt\")\n",
    "Blast_record=NCBIXML.parse(Blast)\n",
    "for x in Blast_record:\n",
    "    print(\"Exist {} alignments in the register\".format(len(x.alignments))) #by defaut\n",
    "    print(x.alignments[0])#best-one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d4254e07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----Aligment----\n",
      "sequence: gi|1519243738|ref|NM_006773.4| Homo sapiens DEAD-box helicase 18 (DDX18), mRNA\n",
      "lenght: 3753\n",
      "e.value: 0.0\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "----Aligment----\n",
      "sequence: gi|19353238|gb|BC024739.1| Homo sapiens DEAD (Asp-Glu-Ala-Asp) box polypeptide 18, mRNA (cDNA clone MGC:29902 IMAGE:4995871), complete cds\n",
      "lenght: 3764\n",
      "e.value: 0.0\n",
      "GGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCACTTGTTGGGCAGAAT...\n",
      "GGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCACTTGTTGGGCAGAAT...\n",
      "GGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCACTTGTTGGGCAGAAT...\n",
      "----Aligment----\n",
      "sequence: gi|1849097599|ref|XM_003819127.3| PREDICTED: Pan paniscus DEAD-box helicase 18 (DDX18), mRNA\n",
      "lenght: 3779\n",
      "e.value: 0.0\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "----Aligment----\n",
      "sequence: gi|1849097599|ref|XM_003819127.3| PREDICTED: Pan paniscus DEAD-box helicase 18 (DDX18), mRNA\n",
      "lenght: 3779\n",
      "e.value: 0.000504389\n",
      "TATCGGAAACTCAAAATGGAGATGTATCTGAAGAAACAATGGGAAGTAGAAAGGTTAAAAAATC...\n",
      "TATCGGAAACTCAAAATGGAGATGTATCTGAAGAAACAATGGGAAGTAGAAAGGTTAAAAAATC...\n",
      "TATCAGAAACTCAAAATGGAGACGTGTCTCAAGAAGCAGTGGAAAATGTAAAAGTTAAAAAGTC...\n",
      "----Aligment----\n",
      "sequence: gi|1034148886|ref|XM_515753.6| PREDICTED: Pan troglodytes DEAD-box helicase 18 (DDX18), mRNA\n",
      "lenght: 3832\n",
      "e.value: 0.0\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGACAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "----Aligment----\n",
      "sequence: gi|1753062192|ref|XM_004031662.3| PREDICTED: Gorilla gorilla gorilla DEAD-box helicase 18 (DDX18), mRNA\n",
      "lenght: 3833\n",
      "e.value: 0.0\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTAGCTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "ACGTGCGGCCGGAAGGGAAGTAACGTCAGCCTGAGAACTGAGTC-CTGTACTGTGTGGCGCCTTATTCTAGGCAC...\n",
      "----Aligment----\n",
      "sequence: gi|1753062192|ref|XM_004031662.3| PREDICTED: Gorilla gorilla gorilla DEAD-box helicase 18 (DDX18), mRNA\n",
      "lenght: 3833\n",
      "e.value: 9.737e-07\n",
      "TATCGGAAACTCAAAATGGAGATGTATCTGAAGAAACAATGGGAAGTAGAAAGGTTAAAAAATC...\n",
      "TATCGGAAACTCAAAATGGAGATGTATCTGAAGAAACAATGGGAAGTAGAAAGGTTAAAAAATC...\n",
      "TATCAGAAACTCAAAATGGAGATGTGTCTCAAGAAGCAGTGGAAAATATAAAAGTTAAAAAGTC...\n",
      "----Aligment----\n",
      "sequence: gi|1753062192|ref|XM_004031662.3| PREDICTED: Gorilla gorilla gorilla DEAD-box helicase 18 (DDX18), mRNA\n",
      "lenght: 3833\n",
      "e.value: 0.000504389\n",
      "TATCAGAAACTCAAAATGGAGGCATGTCTCAAGAAGCAGTGGGAAATATAAAAGTTACAAAGTC...\n",
      "TATCAGAAACTCAAAATGGAGGCATGTCTCAAGAAGCAGTGGGAAATATAAAAGTTACAAAGTC...\n",
      "TATCAGAAACTCAAAATGGAGATGTATCTGAAGAAACAATGGGAAGTAGAAAGGTTAAAAAATC...\n"
     ]
    }
   ],
   "source": [
    "result_handle=open(\"blast_file_ddx18.xml\")\n",
    "blast_records=NCBIXML.read(result_handle)\n",
    "E_VALUE_THRESH=0.01 #could have defined another value of tresh hold\n",
    "for alignment in blast_records.alignments[0:5]: #only for the five best results in terms os e-value\n",
    "    for hsp in alignment.hsps:\n",
    "        if hsp.expect<E_VALUE_THRESH:\n",
    "            print(\"----Aligment----\")\n",
    "            print(\"sequence:\",alignment.title)\n",
    "            print(\"lenght:\",alignment.length)\n",
    "            print(\"e.value:\",hsp.expect) \n",
    "            print(hsp.query[0:75] + \"...\")\n",
    "            print(hsp.query[0:75] + \"...\")\n",
    "            print(hsp.sbjct[0:75] + \"...\")\n",
    "            "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b3a5983a",
   "metadata": {},
   "source": [
    "# Blastp"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1098927",
   "metadata": {},
   "source": [
    "## DDX18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "5316d7d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle=NCBIWWW.qblast(\"blastp\",\"swissprot\",seq_p_ddx18)#blast with the protein translate of my cds\n",
    "with open(\"blastp_file_ddx18.xml\",\"w\") as out_handle:\n",
    "    out_handle.write(result_handle.read())\n",
    "result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "1ac59fc7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sp|Q9NVP1.2| RecName: Full=ATP-dependent RNA helicase DDX18; AltName: Full=DEAD box protein 18; AltName: Full=Myc-regulated DEAD box protein; Short=MrDb [Homo sapiens]\n",
      "sp|Q8K363.1| RecName: Full=ATP-dependent RNA helicase DDX18; AltName: Full=DEAD box protein 18 [Mus musculus]\n",
      "sp|Q9VD51.2| RecName: Full=Probable ATP-dependent RNA helicase pitchoune [Drosophila melanogaster]\n",
      "sp|Q1EA54.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Coccidioides immitis RS]\n",
      "sp|A4R8B5.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Pyricularia oryzae 70-15]\n",
      "sp|A1CIQ5.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus clavatus NRRL 1]\n",
      "sp|Q5BBY1.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus nidulans FGSC A4]\n",
      "sp|Q4IEK8.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Fusarium graminearum PH-1]\n",
      "sp|A1CW14.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus fischeri NRRL 181]\n",
      "sp|Q4WQM4.2| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus fumigatus Af293]\n"
     ]
    }
   ],
   "source": [
    "Blast=open(\"blastp_file_ddx18.xml\")\n",
    "Blast_record=NCBIXML.read(Blast)\n",
    "for a in Blast_record.descriptions[0:10]:\n",
    "    print(a.title) #só quero o primeiro"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "3a05c13a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "swissprot\n",
      "BLOSUM62\n",
      "(11, 1)\n",
      "50\n"
     ]
    }
   ],
   "source": [
    "Blast=open(\"blastp_file_ddx18.xml\")\n",
    "Blast_record=NCBIXML.parse(Blast)\n",
    "for x in Blast_record:\n",
    "    print(x.database)\n",
    "    print(x.matrix)\n",
    "    print(x.gap_penalties)\n",
    "    print(len(x.alignments))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "d2f129a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Q9NVP1\n",
      "sp|Q9NVP1.2|\n",
      "RecName: Full=ATP-dependent RNA helicase DDX18; AltName: Full=DEAD box protein 18; AltName: Full=Myc-regulated DEAD box protein; Short=MrDb [Homo sapiens]\n"
     ]
    }
   ],
   "source": [
    "first=x.alignments[0] #best one\n",
    "print(first.accession)\n",
    "print(first.hit_id)\n",
    "print(first.hit_def)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "62885252",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0\n",
      "3581.0\n",
      "670\n",
      "670\n",
      "\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "1\n",
      "1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for a in first.hsps:\n",
    "    print(a.expect)\n",
    "    print(a.score)\n",
    "    print(a.align_length)\n",
    "    print(len(a.match))\n",
    "    print(\"\")\n",
    "    print(a.query[0:75]+\"...\")\n",
    "    print(a.match[0:75]+\"...\")\n",
    "    print(a.sbjct[0:75]+\"...\")\n",
    "    print(a.query_start)\n",
    "    print(a.sbjct_start)\n",
    "    print(\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "0129c939",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----Aligment----\n",
      "sequence: sp|Q9NVP1.2| RecName: Full=ATP-dependent RNA helicase DDX18; AltName: Full=DEAD box protein 18; AltName: Full=Myc-regulated DEAD box protein; Short=MrDb [Homo sapiens]\n",
      "lenght: 670\n",
      "e.value: 0.0\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "----Aligment----\n",
      "sequence: sp|Q8K363.1| RecName: Full=ATP-dependent RNA helicase DDX18; AltName: Full=DEAD box protein 18 [Mus musculus]\n",
      "lenght: 660\n",
      "e.value: 0.0\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKPMNVGLSETQNGGMSQ...\n",
      "MSQLQMKLLRRKIEKRNAKLRQRNLKLQETSDTSLSQPQNGDVPKETGKGGKVKKALKRSVPVDSAEAQSGGMPE...\n",
      "----Aligment----\n",
      "sequence: sp|Q9VD51.2| RecName: Full=Probable ATP-dependent RNA helicase pitchoune [Drosophila melanogaster]\n",
      "lenght: 680\n",
      "e.value: 0.0\n",
      "EDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPR...\n",
      "EDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPR...\n",
      "DDRSFASLKGAVSEATLRAIKEMGFTEMTEIQSKSLTPLLKGRDLVGAAQTGSGKTLAFLIPAVELINKLRFMPR...\n",
      "----Aligment----\n",
      "sequence: sp|Q1EA54.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Coccidioides immitis RS]\n",
      "lenght: 604\n",
      "e.value: 0.0\n",
      "QNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAET...\n",
      "QNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAET...\n",
      "KHGGSGNAALTSANAAKSLVKTTLSSKGEASAE--KQEMKKRKTEHPSLMEDRVSENVVDEPEEVVEEDSEASQP...\n",
      "----Aligment----\n",
      "sequence: sp|A4R8B5.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Pyricularia oryzae 70-15]\n",
      "lenght: 587\n",
      "e.value: 0.0\n",
      "NDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKA...\n",
      "NDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKA...\n",
      "NDWSDEEENEAADDAGHSSDSDVD---EGDDTIAAPAKEADGDIPGDLTLPTTAESEAQAFSEL-NL-SENTMKA...\n",
      "----Aligment----\n",
      "sequence: sp|A1CIQ5.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus clavatus NRRL 1]\n",
      "lenght: 625\n",
      "e.value: 0.0\n",
      "KVKKSKQKPMNVGLSETQNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTK...\n",
      "KVKKSKQKPMNVGLSETQNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTK...\n",
      "KKRKRKHGGNGRATAETADDASPLPAAENGAVTESSAKKEVKADKKSGKEKEKALKKRKVSHSSSDEEEEEEDDE...\n",
      "----Aligment----\n",
      "sequence: sp|Q5BBY1.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus nidulans FGSC A4]\n",
      "lenght: 609\n",
      "e.value: 0.0\n",
      "QNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAET...\n",
      "QNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAET...\n",
      "KHGGSAREATDN-GVATSPAENGV--EGKATETTTELTKDKDVELAKKVVKEEAQKKRKVSHSPSDEEESESEDE...\n",
      "----Aligment----\n",
      "sequence: sp|Q4IEK8.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Fusarium graminearum PH-1]\n",
      "lenght: 591\n",
      "e.value: 0.0\n",
      "KKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDND--------------------EDESEVPS--...\n",
      "KKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDND--------------------EDESEVPS--...\n",
      "KKSKKVKRAEPEPHD-EPEDDSSDEEEQA--LKEVEDESDQADEDVEAANSAEEEEEEGGDEDNAENNTDLPNGG...\n",
      "----Aligment----\n",
      "sequence: sp|A1CW14.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus fischeri NRRL 181]\n",
      "lenght: 622\n",
      "e.value: 0.0\n",
      "VKKSKQKPMNVGLSETQ-NGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTK...\n",
      "VKKSKQKPMNVGLSETQ-NGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTK...\n",
      "TKKRKRKHGGGARAATETDDATTQPVVENGAVTDSPEK------GEDTKKSEKN-GKDKSTKKRKVSHASSDEEN...\n",
      "----Aligment----\n",
      "sequence: sp|Q4WQM4.2| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus fumigatus Af293]\n",
      "lenght: 622\n",
      "e.value: 0.0\n",
      "VKKSKQKPMNVGLSETQ-NGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTK...\n",
      "VKKSKQKPMNVGLSETQ-NGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTK...\n",
      "TKKRKRKHGGGARAATETDDAITRPAIENGAVNDSPEK-------EEDTKKSEKNGKDKSAKKRK-VSHASSDEG...\n",
      "----Aligment----\n",
      "sequence: sp|A2Q9T6.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus niger CBS 513.88]\n",
      "lenght: 606\n",
      "e.value: 0.0\n",
      "AETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLE...\n",
      "AETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLE...\n",
      "ADPEDEDDNEGKEDGNGADLPSLDAVRLPQTDG-EPKKFTEL-NL-SEKTMKGIQDMGFETMTEIQQRTIPPLLA...\n",
      "----Aligment----\n",
      "sequence: sp|Q0UR48.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Parastagonospora nodorum SN15]\n",
      "lenght: 610\n",
      "e.value: 0.0\n",
      "TNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAET--------TKETENNVEKPDNDEDE...\n",
      "TNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAET--------TKETENNVEKPDNDEDE...\n",
      "TNGAAAVEKPRKKSKKEHTPEPE-VEDVVADASENDVESGAEEDEEQVNAELKEIAAKAKKAKKAKAAQDEDEDE...\n",
      "----Aligment----\n",
      "sequence: sp|Q54S03.1| RecName: Full=Probable ATP-dependent RNA helicase ddx18; AltName: Full=DEAD box protein 18 [Dictyostelium discoideum]\n",
      "lenght: 602\n",
      "e.value: 0.0\n",
      "EDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPR...\n",
      "EDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPR...\n",
      "ESIEFSNLP--IEENTKKSIEEMGFKKMTPIQAKSILPLLEGKDLLGAARTGSGKTLAFLIPAIEVLVKSNFKPR...\n",
      "----Aligment----\n",
      "sequence: sp|A3LNR6.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Scheffersomyces stipitis CBS 6054]\n",
      "lenght: 567\n",
      "e.value: 0.0\n",
      "VNDAEPDTKKAKTENKGKSEEES-AETTK-ETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENT...\n",
      "VNDAEPDTKKAKTENKGKSEEES-AETTK-ETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENT...\n",
      "VTDPE-EKKQSKLERKRKRDEEATAEYTKPEVVDNEDDAQNDK---------------FEDAG-------LSEPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q2UUN6.2| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus oryzae RIB40]\n",
      "lenght: 596\n",
      "e.value: 0.0\n",
      "SEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNL--------------VNENTLKAIKE...\n",
      "SEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNL--------------VNENTLKAIKE...\n",
      "SSDEEQKNTEESEQ--EASDNEEEKSDAEETP---AANGEDLPSADTIRLPQQDGDPVKFTELGLSEKTMKGIEG...\n",
      "----Aligment----\n",
      "sequence: sp|Q7S2N9.1| RecName: Full=ATP-dependent RNA helicase has-1 [Neurospora crassa OR74A]\n",
      "lenght: 578\n",
      "e.value: 0.0\n",
      "EPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKE...\n",
      "EPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKE...\n",
      "ENDAQKAEEQQDGDEVIPTGNGDKEDSDDKDAEAGDELTKTNDSL-IAPSIATNATDFSEL-NL-SDKTMKAIAE...\n",
      "----Aligment----\n",
      "sequence: sp|Q09916.1| RecName: Full=ATP-dependent RNA helicase has1 [Schizosaccharomyces pombe 972h-]\n",
      "lenght: 578\n",
      "e.value: 0.0\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSENIQKAIKEMGFETMTEIQKRSIPPLLAGRDVLGAAKTGSGKTLAFLIPTIEMLYALKFKPRNGTGVIIISPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q03532.1| RecName: Full=ATP-dependent RNA helicase HAS1; AltName: Full=Helicase associated with SET1 protein 1 [Saccharomyces cerevisiae S288C]\n",
      "lenght: 505\n",
      "e.value: 0.0\n",
      "AKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNM...\n",
      "AKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNM...\n",
      "ATPSNKRSRDSESTEEPVVDEKSTSKQNNAAPEGEQTTC----VEKFEELK-------LSQPTLKAIEKMGFTTM...\n",
      "----Aligment----\n",
      "sequence: sp|Q6CXB7.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Kluyveromyces lactis NRRL Y-1140]\n",
      "lenght: 497\n",
      "e.value: 0.0\n",
      "DNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGK...\n",
      "DNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGK...\n",
      "EQDHAETETVVKDTATTGEVEG-SFSDLK--LSDGTMKAIGKMGFTQMTAVQTRTIPPLLAGKDVLGAAKTGSGK...\n",
      "----Aligment----\n",
      "sequence: sp|Q6BH93.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Debaryomyces hansenii CBS767]\n",
      "lenght: 568\n",
      "e.value: 0.0\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSEPTLKAIKDMGFSKMTQVQAKTIPPLLAGRDVLGAAKTGSGKTLAFLIPAIEMLYSLRFKPRNGTGVVVVSPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q6FIL3.1| RecName: Full=ATP-dependent RNA helicase HAS1 [[Candida] glabrata CBS 138]\n",
      "lenght: 494\n",
      "e.value: 0.0\n",
      "KPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGS...\n",
      "KPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGS...\n",
      "KRSRDESEKEEVVVKADVESSDVDHSFKSL-NL-SQPTMRAIEKMGFSKMTPVQARTIPPLMAGRDVLGAAKTGS...\n",
      "----Aligment----\n",
      "sequence: sp|Q74Z73.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Eremothecium gossypii ATCC 10895]\n",
      "lenght: 504\n",
      "e.value: 0.0\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSSQTLKAIGKMGFTKMTQVQARTIPPLMAGRDVLGAAKTGSGKTLAFLLPAIEMLHSLKFKPRNGTGVIVITPT...\n",
      "----Aligment----\n",
      "sequence: sp|P0CQ84.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Cryptococcus neoformans var. neoformans JEC21] >sp|P0CQ85.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Cryptococcus neoformans var. neoformans B-3501A]\n",
      "lenght: 607\n",
      "e.value: 0.0\n",
      "SPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDE...\n",
      "SPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDE...\n",
      "TPSHGNAHSSAHRAMSSTKPPTTTNKRKR----------TSNAHDEAPAKRVPEASSSKVTLDDSQPAPATSSDA...\n",
      "----Aligment----\n",
      "sequence: sp|A5E2Z9.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Lodderomyces elongisporus NRRL YB-4239]\n",
      "lenght: 559\n",
      "e.value: 0.0\n",
      "EEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIR...\n",
      "EEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIR...\n",
      "EKRLQELTKPKVSNTSEAEADEP---------GVNYSFEKADFS-------EPTMKAIKEMGFQKMTKVQAKTIP...\n",
      "----Aligment----\n",
      "sequence: sp|A5DID7.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Meyerozyma guilliermondii ATCC 6260]\n",
      "lenght: 569\n",
      "e.value: 0.0\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSEPTMRAIKDMGFEKMTKVQEKTIPPLLAGRDVLGAAKTGSGKTLAFLIPAIEMLYSLKFKPRNGTGVIVVSPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q6C7D2.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Yarrowia lipolytica CLIB122]\n",
      "lenght: 605\n",
      "e.value: 0.0\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSENTMQSLKDMGFETMTPVQEKTIPPLLAGRDVLGAAKTGSGKTLAFLIPAIEMLRKLKFKPRNGTGVIVVSPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q4P6N0.2| RecName: Full=ATP-dependent RNA helicase HAS1 [Ustilago maydis 521]\n",
      "lenght: 574\n",
      "e.value: 0.0\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSEPTRKAIDAMGFKTMTEVQARCIPPLMAGKDVLGAAQTGSGKTLSFLIPAIEMLHRLKFKPRNGTGAIIISPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q0CZS8.1| RecName: Full=ATP-dependent RNA helicase has1 [Aspergillus terreus NIH2624]\n",
      "lenght: 576\n",
      "e.value: 0.0\n",
      "SESKKKKKKKRKMVNDAEPDTKKAKTE--------------------NKGKSEEESAETTKETENNVEKPDNDED...\n",
      "SESKKKKKKKRKMVNDAEPDTKKAKTE--------------------NKGKSEEESAETTKETENNVEKPDNDED...\n",
      "SAQDQTKKRKRQKVSDGSKSSKSRVVEADDQRELTETTTKPKKQKTEDPPTDETPDAEDVEQTENGETAQDDAAK...\n",
      "----Aligment----\n",
      "sequence: sp|Q5AK59.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Candida albicans SC5314]\n",
      "lenght: 565\n",
      "e.value: 0.0\n",
      "FEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMP...\n",
      "FEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMP...\n",
      "FENADFS-------EPTMKAIKEMGFTKMTKVQAKTIPPLLAGRDVLGAAKTGSGKTLAFLIPAIELLYSLKIKP...\n",
      "----Aligment----\n",
      "sequence: sp|Q2GMX1.1| RecName: Full=ATP-dependent RNA helicase HAS1 [Chaetomium globosum CBS 148.51]\n",
      "lenght: 586\n",
      "e.value: 0.0\n",
      "SSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDES--EVPSLPLGLTGAF...\n",
      "SSNSESKKKKKKKRKMVNDAEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDES--EVPSLPLGLTGAF...\n",
      "ASSSKAKKLKRAPTPEDSDAERDNS---SDPEIENQEPEVEVPGPPQDNSEDQEAEENANAFERQDEPAGLDSAI...\n",
      "----Aligment----\n",
      "sequence: sp|Q84T03.1| RecName: Full=DEAD-box ATP-dependent RNA helicase 27 [Oryza sativa Japonica Group]\n",
      "lenght: 590\n",
      "e.value: 0.0\n",
      "GLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVK...\n",
      "GLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVK...\n",
      "GGSGILTNMLFSELG--VSEPTARAIREMNYTYLTQIQARSIPHLLNGKDVMGAAKTGSGKTLAFLIPAIEMLHH...\n",
      "----Aligment----\n",
      "sequence: sp|Q9LIH9.1| RecName: Full=DEAD-box ATP-dependent RNA helicase 51 [Arabidopsis thaliana]\n",
      "lenght: 568\n",
      "e.value: 0.0\n",
      "GAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRF...\n",
      "GAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRF...\n",
      "GIMTNVTFDSLD--LSEQTSIAIKEMGFQYMTQIQAGSIQPLLEGKDVLGAARTGSGKTLAFLIPAVELLFKERF...\n",
      "----Aligment----\n",
      "sequence: sp|Q9SB89.2| RecName: Full=DEAD-box ATP-dependent RNA helicase 27 [Arabidopsis thaliana]\n",
      "lenght: 633\n",
      "e.value: 0.0\n",
      "TGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLR...\n",
      "TGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLR...\n",
      "TSIMTNKTFESLS--LSDNTYKSIKEMGFARMTQIQAKAIPPLMMGEDVLGAARTGSGKTLAFLIPAVELLYRVK...\n",
      "----Aligment----\n",
      "sequence: sp|Q0DBS1.2| RecName: Full=Putative DEAD-box ATP-dependent RNA helicase 51 [Oryza sativa Japonica Group]\n",
      "lenght: 590\n",
      "e.value: 0.0\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "ISDLTANAIRDMNYTHLTEIQARSIPPLMLGSDVMASAKTGSGKTLAFLIPAIELLCRLRFSPRNGTGVIVLCPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q13206.2| RecName: Full=Probable ATP-dependent RNA helicase DDX10; AltName: Full=DEAD box protein 10 [Homo sapiens]\n",
      "lenght: 875\n",
      "e.value: 3.99271e-126\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSKKTLKGLQEAQYRLVTEIQKQTIGLALQGKDVLGAAKTGSGKTLAFLVPVLEALYRLQWTSTDGLGVLIISPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q80Y44.2| RecName: Full=Probable ATP-dependent RNA helicase DDX10; AltName: Full=DEAD box protein 10 [Mus musculus]\n",
      "lenght: 875\n",
      "e.value: 5.2043e-126\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSKKTLKGLQEAQYRLVTEIQKQTIGLALQGKDVLGAAKTGSGKTLAFLVPVLEALYRLQWTSTDGLGVLIISPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q5ZJF6.1| RecName: Full=Probable ATP-dependent RNA helicase DDX10; AltName: Full=DEAD box protein 10 [Gallus gallus]\n",
      "lenght: 875\n",
      "e.value: 1.04757e-125\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSKKTLKGLQEAQYRMVTEIQRQTIGLALQGKDVLGAAKTGSGKTLAFIVPALELLYRLQWTSADGLGVLIISPT...\n",
      "----Aligment----\n",
      "sequence: sp|A1CTZ2.1| RecName: Full=ATP-dependent RNA helicase dbp4 [Aspergillus clavatus NRRL 1]\n",
      "lenght: 823\n",
      "e.value: 7.15138e-119\n",
      "ESEVPSLPLGLTGAFEDTSFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAF...\n",
      "ESEVPSLPLGLTGAFEDTSFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAF...\n",
      "QEELSSLIQRVEGLDLKESFESFSDLPLSEPTLSGLTSSHFKTLTDIQSRAISHALKGRDVLGAAKTGSGKTLAF...\n",
      "----Aligment----\n",
      "sequence: sp|Q4P5U4.2| RecName: Full=ATP-dependent RNA helicase DBP4 [Ustilago maydis 521]\n",
      "lenght: 916\n",
      "e.value: 7.62405e-119\n",
      "SAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPL...\n",
      "SAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPL...\n",
      "AAPHLKSRSKNARKLKRSAEDAEIAQLEQGIQAFVSPIDLKQFTQLPLSDRTCRGLKRAGYTDMTDIQAKSLSLS...\n",
      "----Aligment----\n",
      "sequence: sp|Q1EB31.1| RecName: Full=ATP-dependent RNA helicase DBP4 [Coccidioides immitis RS]\n",
      "lenght: 806\n",
      "e.value: 1.4393e-116\n",
      "SFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGT...\n",
      "SFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGT...\n",
      "SFSELP--LSDATLQGLSASHFKTLTDIQSRAVPHAIKGRDILGAAKTGSGKTLAFLVPVLENLYRKQWTEYDGL...\n",
      "----Aligment----\n",
      "sequence: sp|Q0CMM5.2| RecName: Full=ATP-dependent RNA helicase dbp4 [Aspergillus terreus NIH2624]\n",
      "lenght: 804\n",
      "e.value: 2.78647e-116\n",
      "SFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNG...\n",
      "SFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNG...\n",
      "SFKAFTDLPLSEPTLSGLSASHYKTLTDIQSRAVSHALKGRDILGAAKTGSGKTLAFLIPVLENLYRKQWAEHDG...\n",
      "----Aligment----\n",
      "sequence: sp|Q4WM60.1| RecName: Full=ATP-dependent RNA helicase dbp4 [Aspergillus fumigatus Af293]\n",
      "lenght: 787\n",
      "e.value: 3.02172e-116\n",
      "LGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIV...\n",
      "LGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIV...\n",
      "LDLKGIFK--SFSDLP--LSEPTASGLASSHYKTLTDIQSRAISHALKGRDILGAAKTGSGKTLAFLVPVLENLY...\n",
      "----Aligment----\n",
      "sequence: sp|A1DNF9.1| RecName: Full=ATP-dependent RNA helicase dbp4 [Aspergillus fischeri NRRL 181]\n",
      "lenght: 810\n",
      "e.value: 1.67914e-115\n",
      "SFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNG...\n",
      "SFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNG...\n",
      "TFKSFSDLPLSEPTASGLASSHYKTLTDIQSRAISHALKGRDVLGAAKTGSGKTLAFLVPVLENLYRKQWAEHDG...\n",
      "----Aligment----\n",
      "sequence: sp|A7ESL8.1| RecName: Full=ATP-dependent RNA helicase dbp4 [Sclerotinia sclerotiorum 1980 UF-70]\n",
      "lenght: 808\n",
      "e.value: 9.7641e-115\n",
      "FTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPTRELAMQTFGVLKE...\n",
      "FTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPTRELAMQTFGVLKE...\n",
      "FKTLTDVQSKAVPLALKGKDILGAAKTGSGKTLAFLVPVLENLYRQKWTELDGLGALIISPTRELAIQIFEVLRK...\n",
      "----Aligment----\n",
      "sequence: sp|Q2H2J1.1| RecName: Full=ATP-dependent RNA helicase DBP4 [Chaetomium globosum CBS 148.51]\n",
      "lenght: 825\n",
      "e.value: 1.01171e-114\n",
      "LCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLI...\n",
      "LCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLI...\n",
      "LC----EATASGLRASHFEVLTDVQRAAIPLALKGRDILGAAKTGSGKTLAFLVPVLEKLYHAKWTEYDGLGALI...\n",
      "----Aligment----\n",
      "sequence: sp|Q5BFU1.1| RecName: Full=ATP-dependent RNA helicase dbp4 [Aspergillus nidulans FGSC A4]\n",
      "lenght: 812\n",
      "e.value: 2.99652e-114\n",
      "LGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIV...\n",
      "LGLTGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIV...\n",
      "LDLKAKYE--SFSDLP--ISEPTLSGLTSSHFKTLTDIQSRAISHALKGRDVLGAAKTGSGKTLAFLVPILENLY...\n",
      "----Aligment----\n",
      "sequence: sp|Q0UMB9.1| RecName: Full=ATP-dependent RNA helicase DBP4 [Parastagonospora nodorum SN15]\n",
      "lenght: 803\n",
      "e.value: 3.3871e-114\n",
      "TGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLR...\n",
      "TGAFEDTSFASLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLR...\n",
      "TGTYNDFSDLPL----SDPTKQGLKACHFAVMTDIQRKAVPLALKGHDILGAAKTGSGKTLSFIIPVLENLYRLQ...\n",
      "----Aligment----\n",
      "sequence: sp|Q2UHB7.1| RecName: Full=ATP-dependent RNA helicase dbp4 [Aspergillus oryzae RIB40]\n",
      "lenght: 796\n",
      "e.value: 3.84631e-114\n",
      "SFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNG...\n",
      "SFASLCNL-VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNG...\n",
      "TYKSFSDLPLSEPTASGLASSHFKTLTDIQSRAIGHALKGRDILGAAKTGSGKTLAFLIPVLENLYRKQWSEHDG...\n",
      "----Aligment----\n",
      "sequence: sp|A2QS00.1| RecName: Full=ATP-dependent RNA helicase dbp4 [Aspergillus niger CBS 513.88]\n",
      "lenght: 802\n",
      "e.value: 3.80145e-113\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "LSEPTASGLASSHYKTLTDIQSRAISHALKGRDILGAAKTGSGKTLAFLIPILENLYRKQWSEHDGLGALVLSPT...\n",
      "----Aligment----\n",
      "sequence: sp|Q54Q94.1| RecName: Full=Probable ATP-dependent RNA helicase ddx10; AltName: Full=DEAD box protein 10 [Dictyostelium discoideum]\n",
      "lenght: 878\n",
      "e.value: 3.70908e-112\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "VNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVELIVKLRFMPRNGTGVLILSPT...\n",
      "ISQLTLKALTESKFLKLTDIQRASLPHTLCGRDILGAAKTGSGKTLSFILPILETLWRNRWGRDDGIGAIVLSPT...\n"
     ]
    }
   ],
   "source": [
    "#podia ir para comentário ou retirar  diz a mesma coisa que em cima mas talvez mais bonito\n",
    "result_handle=open(\"blastp_file_ddx18.xml\")\n",
    "blast_record=NCBIXML.read(result_handle)\n",
    "E_VALUE_THRESH=0.01\n",
    "for alignment in blast_record.alignments:\n",
    "    for hsp in alignment.hsps:\n",
    "        if hsp.expect<E_VALUE_THRESH:\n",
    "            print(\"----Aligment----\")\n",
    "            print(\"sequence:\",alignment.title)\n",
    "            print(\"lenght:\",alignment.length)\n",
    "            print(\"e.value:\",hsp.expect) #can we get a value with more detail?\n",
    "            print(hsp.query[0:75] + \"...\")\n",
    "            print(hsp.query[0:75] + \"...\")\n",
    "            print(hsp.sbjct[0:75] + \"...\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "9929ee9e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Query: unnamed protein product\n",
      "        Hit: sp|Q9NVP1.2| RecName: Full=ATP-dependent RNA helicase DDX18; Alt...\n",
      "Query range: [0:670] (0)\n",
      "  Hit range: [0:670] (0)\n",
      "Quick stats: evalue 0; bitscore 1384.01\n",
      "  Fragments: 1 (670 columns)\n",
      "     Query - MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQK~~~RQFSH\n",
      "             MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQK~~~RQFSH\n",
      "       Hit - MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQK~~~RQFSH\n",
      "Alignment with 2 rows and 670 columns\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVS...FSH unnamed\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVS...FSH sp|Q9NVP1.2|\n"
     ]
    }
   ],
   "source": [
    "blast_qresult=SearchIO.read(\"blastp_file_ddx18.xml\",\"blast-xml\")\n",
    "blast_hsp=blast_qresult[0][0]\n",
    "print(blast_qresult[0][0])#best hsp\n",
    "print(blast_hsp.aln)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4cc51d4",
   "metadata": {},
   "source": [
    "## Uniprot research"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "abb618c3",
   "metadata": {},
   "source": [
    "## DDX18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "3c81f64e",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of results: 2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "WEBSITE_API=\"https://rest.uniprot.org\"\n",
    "def get_url(url,**kwargs):\n",
    "    response=requests.get(url,**kwargs);\n",
    "    if not response.ok:\n",
    "        print(response.text)\n",
    "        response.raise_for_status()\n",
    "        sys.exit()\n",
    "    return response\n",
    "r=get_url(f\"{WEBSITE_API}/uniprotkb/search?query=ddx18 AND (taxonomy_id:9606) AND (reviewed:true)\")#more detail possible\n",
    "data=r.json()\n",
    "n_results=len(data[\"results\"])\n",
    "print(f\"Number of results: {n_results}\\n\")\n",
    "total=r.headers.get(\"x-total-results\")\n",
    "page_total=len(data[\"results\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "e82abbc1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Q9NVP1,Q76FK4,']"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#one way to list the ids\n",
    "r=get_url(f\"{WEBSITE_API}//uniprotkb/search?query=DDX18 AND (taxonomy_id:9606) AND (reviewed:true)&format=list&size=2\")\n",
    "#r.text\n",
    "\n",
    "accessions=[r.text.replace(\"\\n\",\",\")]\n",
    "accessions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "3745a4be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">sp|Q9NVP1|DDX18_HUMAN ATP-dependent RNA helicase DDX18 OS=Homo sapiens OX=9606 GN=DDX18 PE=1 SV=2\n",
      "MSHLPMKLLRKKIEKRNLKLRQRNLKFQGASNLTLSETQNGDVSEETMGSRKVKKSKQKP\n",
      "MNVGLSETQNGGMSQEAVGNIKVTKSPQKSTVLTNGEAAMQSSNSESKKKKKKKRKMVND\n",
      "AEPDTKKAKTENKGKSEEESAETTKETENNVEKPDNDEDESEVPSLPLGLTGAFEDTSFA\n",
      "SLCNLVNENTLKAIKEMGFTNMTEIQHKSIRPLLEGRDLLAAAKTGSGKTLAFLIPAVEL\n",
      "IVKLRFMPRNGTGVLILSPTRELAMQTFGVLKELMTHHVHTYGLIMGGSNRSAEAQKLGN\n",
      "GINIIVATPGRLLDHMQNTPGFMYKNLQCLVIDEADRILDVGFEEELKQIIKLLPTRRQT\n",
      "MLFSATQTRKVEDLARISLKKEPLYVGVDDDKANATVDGLEQGYVVCPSEKRFLLLFTFL\n",
      "KKNRKKKLMVFFSSCMSVKYHYELLNYIDLPVLAIHGKQKQNKRTTTFFQFCNADSGTLL\n",
      "CTDVAARGLDIPEVDWIVQYDPPDDPKEYIHRVGRTARGLNGRGHALLILRPEELGFLRY\n",
      "LKQSKVPLSEFDFSWSKISDIQSQLEKLIEKNYFLHKSAQEAYKSYIRAYDSHSLKQIFN\n",
      "VNNLNLPQVALSFGFKVPPFVDLNVNSNEGKQKKRGGGGGFGYQKTKKVEKSKIFKHISK\n",
      "KSSDSRQFSH\n",
      ">sp|Q76FK4|NOL8_HUMAN Nucleolar protein 8 OS=Homo sapiens OX=9606 GN=NOL8 PE=1 SV=1\n",
      "MKVNRETKRLYVGGLSQDISEADLQNQFSRFGEVSDVEIITRKDDQGNPQKVFAYINISV\n",
      "AEADLKKCMSVLNKTKWKGGTLQIQLAKESFLHRLAQEREAAKAKKEESTTGNANLLEKT\n",
      "GGVDFHMKAVPGTEVPGHKNWVVSKFGRVLPVLHLKNQHKRKIIKYDPSKYCHNLKKIGE\n",
      "DFSNTIPISSLTWELEGGNDPMSKKRRGEFSDFHGPPKKIIKVQKDESSTGSLAMSTRPR\n",
      "RVIERPPLTQQQAAQKRTCDSITPSKSSPVPVSDTQKLKNLPFKTSGLETAKKRNSISDD\n",
      "DTDSEDELRMMIAKEENLQRTTQPSINESESDPFEVVRDDFKSGVHKLHSLIGLGIKNRV\n",
      "SCHDSDDDIMRNDREYDSGDTDEIIAMKKNVAKVKNSTEFSQMEKSTKKTSFKNRENCEL\n",
      "SDHCIKLQKRKSNVESALSHGLKSLNRKSPSHSSSSEDADSASELADSEGGEEYNAMMKN\n",
      "CLRVNLTLADLEQLAGSDLKVPNEDTKSDGPETTTQCKFDRGSKSPKTPTGLRRGRQCIR\n",
      "PAEIVASLLEGEENTCGKQKPKENNLKPKFQAFKGVGCLYEKESMKKSLKDSVASNNKDQ\n",
      "NSMKHEDPSIISMEDGSPYVNGSLGEVTPCQHAKKANGPNYIQPQKRQTTFESQDRKAVS\n",
      "PSSSEKRSKNPISRPLEGKKSLSLSAKTHNIGFDKDSCHSTTKTEASQEERSDSSGLTSL\n",
      "KKSPKVSSKDTREIKTDFSLSISNSSDVSAKDKHAEDNEKRLAALEARQKAKEVQKKLVH\n",
      "NALANLDGHPEDKPTHIIFGSDSECETEETSTQEQSHPGEEWVKESMGKTSGKLFDSSDD\n",
      "DESDSEDDSNRFKIKPQFEGRAGQKLMDLQSHFGTDDRFRMDSRFLETDSEEEQEEVNEK\n",
      "KTAEEEELAEEKKKALNVVQSVLQINLSNSTNRGSVAAKKFKDIIHYDPTKQDHATYERK\n",
      "RDDKPKESKAKRKKKREEAEKLPEVSKEMYYNIAMDLKEIFQTTKYTSEKEEGTPWNEDC\n",
      "GKEKPEEIQDPAALTSDAEQPSGFTFSFFDSDTKDIKEETYRVETVKPGKIVWQEDPRLQ\n",
      "DSSSEEEDVTEETDHRNSSPGEASLLEKETTRFFFFSKNDERLQGSDLFWRGVGSNMSRN\n",
      "SWEARTTNLRMDCRKKHKDAKRKMKPK\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#to format fasta that alow see the protein sequences and id in a form more familiary\n",
    "r=get_url(f\"{WEBSITE_API}//uniprotkb/search?query=DDX18  AND (taxonomy_id:9606) AND (reviewed:true)&format=fasta\")\n",
    "fasta=r.text\n",
    "print(fasta)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e25049da",
   "metadata": {},
   "source": [
    "-we choose the first one the id Q9NVP1 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "c0f55a08",
   "metadata": {},
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "Input \u001b[1;32mIn [59]\u001b[0m, in \u001b[0;36m<cell line: 4>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mBio\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m SeqIO\n\u001b[0;32m      4\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m ExPASy\u001b[38;5;241m.\u001b[39mget_sprot_raw(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mQ9NVP1\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m handle:\n\u001b[1;32m----> 5\u001b[0m     seq_record \u001b[38;5;241m=\u001b[39m \u001b[43mSeqIO\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mswiss\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m      6\u001b[0m     \u001b[38;5;28mprint\u001b[39m(seq_record\u001b[38;5;241m.\u001b[39mid)\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SeqIO\\__init__.py:656\u001b[0m, in \u001b[0;36mread\u001b[1;34m(handle, format, alphabet)\u001b[0m\n\u001b[0;32m    654\u001b[0m iterator \u001b[38;5;241m=\u001b[39m parse(handle, \u001b[38;5;28mformat\u001b[39m, alphabet)\n\u001b[0;32m    655\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 656\u001b[0m     record \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mnext\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43miterator\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    657\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mStopIteration\u001b[39;00m:\n\u001b[0;32m    658\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mNo records found in handle\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28mNone\u001b[39m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SeqIO\\SwissIO.py:78\u001b[0m, in \u001b[0;36mSwissIterator\u001b[1;34m(source)\u001b[0m\n\u001b[0;32m     58\u001b[0m \u001b[38;5;124;03m\"\"\"Break up a Swiss-Prot/UniProt file into SeqRecord objects.\u001b[39;00m\n\u001b[0;32m     59\u001b[0m \n\u001b[0;32m     60\u001b[0m \u001b[38;5;124;03mArgument source is a file-like object or a path to a file.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     74\u001b[0m \u001b[38;5;124;03mparser via Bio.SeqIO.parse(..., format=\"swiss\") instead.\u001b[39;00m\n\u001b[0;32m     75\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m     76\u001b[0m swiss_records \u001b[38;5;241m=\u001b[39m SwissProt\u001b[38;5;241m.\u001b[39mparse(source)\n\u001b[1;32m---> 78\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m swiss_record \u001b[38;5;129;01min\u001b[39;00m swiss_records:\n\u001b[0;32m     79\u001b[0m     \u001b[38;5;66;03m# Convert the SwissProt record to a SeqRecord\u001b[39;00m\n\u001b[0;32m     80\u001b[0m     record \u001b[38;5;241m=\u001b[39m SeqRecord(\n\u001b[0;32m     81\u001b[0m         Seq(swiss_record\u001b[38;5;241m.\u001b[39msequence),\n\u001b[0;32m     82\u001b[0m         \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39mswiss_record\u001b[38;5;241m.\u001b[39maccessions[\u001b[38;5;241m0\u001b[39m],\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     85\u001b[0m         features\u001b[38;5;241m=\u001b[39mswiss_record\u001b[38;5;241m.\u001b[39mfeatures,\n\u001b[0;32m     86\u001b[0m     )\n\u001b[0;32m     87\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m cross_reference \u001b[38;5;129;01min\u001b[39;00m swiss_record\u001b[38;5;241m.\u001b[39mcross_references:\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SwissProt\\__init__.py:239\u001b[0m, in \u001b[0;36mparse\u001b[1;34m(source)\u001b[0m\n\u001b[0;32m    237\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    238\u001b[0m     \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m--> 239\u001b[0m         record \u001b[38;5;241m=\u001b[39m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    240\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m record:\n\u001b[0;32m    241\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SwissProt\\__init__.py:371\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(handle)\u001b[0m\n\u001b[0;32m    369\u001b[0m     _read_kw(record, value)\n\u001b[0;32m    370\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m key \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFT\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m--> 371\u001b[0m     \u001b[43m_read_ft\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrecord\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mline\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    372\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m key \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSQ\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m    373\u001b[0m     cols \u001b[38;5;241m=\u001b[39m value\u001b[38;5;241m.\u001b[39msplit()\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SwissProt\\__init__.py:830\u001b[0m, in \u001b[0;36m_read_ft\u001b[1;34m(record, line)\u001b[0m\n\u001b[0;32m    828\u001b[0m \u001b[38;5;66;03m# this line is a continuation of the description of the previous feature\u001b[39;00m\n\u001b[0;32m    829\u001b[0m keys \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(feature\u001b[38;5;241m.\u001b[39mqualifiers\u001b[38;5;241m.\u001b[39mkeys())\n\u001b[1;32m--> 830\u001b[0m key \u001b[38;5;241m=\u001b[39m \u001b[43mkeys\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m-\u001b[39;49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[0;32m    831\u001b[0m description \u001b[38;5;241m=\u001b[39m value\u001b[38;5;241m.\u001b[39mrstrip(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m    832\u001b[0m old_description \u001b[38;5;241m=\u001b[39m feature\u001b[38;5;241m.\u001b[39mqualifiers[key]\n",
      "\u001b[1;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "with ExPASy.get_sprot_raw(\"Q9NVP1\") as handle:\n",
    "    seq_record = SeqIO.read(handle, \"swiss\")\n",
    "    print(seq_record.id)\n",
    "    #print(seq_record.entry_name, \"\\n\")\n",
    "    #print(\", \".join(seq_record.accessions), \"\\n\")\n",
    "    #print(seq_record.keywords, \"\\n\")\n",
    "    #print(seq_record.organism, \"\\n\")\n",
    "    #print(len(seq_record.sequence), \"aa\", \"\\n\")\n",
    "    #print(seq_record.sequence)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8e6898a",
   "metadata": {},
   "source": [
    "We can't understand why the id of ddx18 doesn't work and the other genes already work"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "6f017152",
   "metadata": {},
   "outputs": [],
   "source": [
    "#if we choose to go to the net and download a swiss prot txt file\n",
    "handle=open(\"file.txt\")\n",
    "records=KeyWList.parse(handle)\n",
    "for record in records:\n",
    "    print(record[\"ID\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f92799f9",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#in tuturorial we can explore prosite, maybe in the future"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0706b8b8",
   "metadata": {},
   "source": [
    "# Explore the 3D Strucuture"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef6aa36e",
   "metadata": {},
   "source": [
    "## DDX18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "18e42025",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "email a inserir: guilherme.lobo@ua.pt\n",
      "['82027', '77547', '68579', '66053', '61661', '61107', '60750', '59685', '45556', '45374', '38425']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "b'<?xml version=\"1.0\" ?>\\n<!DOCTYPE eEfetchResult PUBLIC \"-//NLM//DTD efetch 20020605//EN\" \"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20020605/uilist.dtd\">\\n<IdList>\\n    <Id>82027</Id>\\n</IdList>\\n'"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "email= \"guilherme.lobo@ua.pt\"\n",
    "Entrez.email= email\n",
    "handle=Entrez.esearch(db = \"Structure\", term=\"ddx18\", retmax= \"20\")\n",
    "record=Entrez.read(handle)\n",
    "handle.close()\n",
    "print(record[\"IdList\"])\n",
    "#idlist= record['IdList']\n",
    "#print(record)\n",
    "#dir(record)\n",
    "handle = Entrez.efetch(db=\"Structure\", id=\"82027\", rettype=\"fasta\") \n",
    "records = handle.read()\n",
    "handle.close()\n",
    "records\n",
    "#Try to find the id or the identification to explore my protein in PDB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "2e9b299b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Structure exists: '.\\pdb3ly5.ent' \n",
      "<Structure id=3LY5>\n"
     ]
    }
   ],
   "source": [
    "#like we cant ind the id we go to the the web and choose the apropriate id- 3LY5\n",
    "from Bio.PDB import *\n",
    "pdb=PDBList()\n",
    "pdb.retrieve_pdb_file(\"3LY5\",pdir=\".\",file_format=\"pdb\")\n",
    "parser=PDBParser(PERMISSIVE=True,QUIET=True)\n",
    "data=parser.get_structure(\"3LY5\",\"pdb3LY5.ent\")\n",
    "print(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "9185e19c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\guilh\\anaconda3\\lib\\site-packages\\Bio\\PDB\\StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 3736.\n",
      "  warnings.warn(\n",
      "C:\\Users\\guilh\\anaconda3\\lib\\site-packages\\Bio\\PDB\\StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 3746.\n",
      "  warnings.warn(\n",
      "C:\\Users\\guilh\\anaconda3\\lib\\site-packages\\Bio\\PDB\\StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 3749.\n",
      "  warnings.warn(\n",
      "C:\\Users\\guilh\\anaconda3\\lib\\site-packages\\Bio\\PDB\\StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 3774.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom OH>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom SG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom P>\n",
      "<Atom O1>\n",
      "<Atom O2>\n",
      "<Atom O3>\n",
      "<Atom O4>\n",
      "<Atom MG>\n",
      "<Atom MG>\n",
      "<Atom S>\n",
      "<Atom C>\n",
      "<Atom N>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom OH>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom ND1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom OH>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom ND2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom SG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom NE>\n",
      "<Atom CZ>\n",
      "<Atom NH1>\n",
      "<Atom NH2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom SD>\n",
      "<Atom CE>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom CE1>\n",
      "<Atom CE2>\n",
      "<Atom CZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom NE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom OE1>\n",
      "<Atom OE2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom OD1>\n",
      "<Atom OD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom CD1>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom OG>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom CE>\n",
      "<Atom NZ>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom CD1>\n",
      "<Atom CD2>\n",
      "<Atom N>\n",
      "<Atom CA>\n",
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG>\n",
      "<Atom N>\n",
      "<Atom CA>\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Atom C>\n",
      "<Atom O>\n",
      "<Atom CB>\n",
      "<Atom CG1>\n",
      "<Atom CG2>\n",
      "<Atom S>\n",
      "<Atom C>\n",
      "<Atom N>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n",
      "<Atom O>\n"
     ]
    }
   ],
   "source": [
    "p = PDBParser()\n",
    "structure = p.get_structure('X', 'pdb3LY5.ent')\n",
    "for model in structure:\n",
    "    for chain in model:\n",
    "        for residue in chain:\n",
    "            for atom in residue:\n",
    "                print(atom)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "fee5e589",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Residue THR het=  resseq=171 icode= >\n",
      "<Residue GLY het=  resseq=172 icode= >\n",
      "<Residue ALA het=  resseq=173 icode= >\n",
      "<Residue PHE het=  resseq=174 icode= >\n",
      "<Residue GLU het=  resseq=175 icode= >\n",
      "<Residue ASP het=  resseq=176 icode= >\n",
      "<Residue THR het=  resseq=177 icode= >\n",
      "<Residue SER het=  resseq=178 icode= >\n",
      "<Residue PHE het=  resseq=179 icode= >\n",
      "<Residue LEU het=  resseq=185 icode= >\n",
      "<Residue VAL het=  resseq=186 icode= >\n",
      "<Residue ASN het=  resseq=187 icode= >\n",
      "<Residue GLU het=  resseq=188 icode= >\n",
      "<Residue ASN het=  resseq=189 icode= >\n",
      "<Residue THR het=  resseq=190 icode= >\n",
      "<Residue LEU het=  resseq=191 icode= >\n",
      "<Residue LYS het=  resseq=192 icode= >\n",
      "<Residue ALA het=  resseq=193 icode= >\n",
      "<Residue ILE het=  resseq=194 icode= >\n",
      "<Residue LYS het=  resseq=195 icode= >\n",
      "<Residue GLU het=  resseq=196 icode= >\n",
      "<Residue MET het=  resseq=197 icode= >\n",
      "<Residue GLY het=  resseq=198 icode= >\n",
      "<Residue PHE het=  resseq=199 icode= >\n",
      "<Residue THR het=  resseq=200 icode= >\n",
      "<Residue ASN het=  resseq=201 icode= >\n",
      "<Residue MET het=  resseq=202 icode= >\n",
      "<Residue THR het=  resseq=203 icode= >\n",
      "<Residue GLU het=  resseq=204 icode= >\n",
      "<Residue ILE het=  resseq=205 icode= >\n",
      "<Residue GLN het=  resseq=206 icode= >\n",
      "<Residue HIS het=  resseq=207 icode= >\n",
      "<Residue LYS het=  resseq=208 icode= >\n",
      "<Residue SER het=  resseq=209 icode= >\n",
      "<Residue ILE het=  resseq=210 icode= >\n",
      "<Residue ARG het=  resseq=211 icode= >\n",
      "<Residue PRO het=  resseq=212 icode= >\n",
      "<Residue LEU het=  resseq=213 icode= >\n",
      "<Residue LEU het=  resseq=214 icode= >\n",
      "<Residue GLU het=  resseq=215 icode= >\n",
      "<Residue GLY het=  resseq=216 icode= >\n",
      "<Residue ARG het=  resseq=217 icode= >\n",
      "<Residue ASP het=  resseq=218 icode= >\n",
      "<Residue LEU het=  resseq=219 icode= >\n",
      "<Residue LEU het=  resseq=220 icode= >\n",
      "<Residue ALA het=  resseq=221 icode= >\n",
      "<Residue ALA het=  resseq=222 icode= >\n",
      "<Residue ALA het=  resseq=223 icode= >\n",
      "<Residue LYS het=  resseq=224 icode= >\n",
      "<Residue THR het=  resseq=225 icode= >\n",
      "<Residue GLY het=  resseq=226 icode= >\n",
      "<Residue SER het=  resseq=227 icode= >\n",
      "<Residue GLY het=  resseq=228 icode= >\n",
      "<Residue LYS het=  resseq=229 icode= >\n",
      "<Residue THR het=  resseq=230 icode= >\n",
      "<Residue LEU het=  resseq=231 icode= >\n",
      "<Residue ALA het=  resseq=232 icode= >\n",
      "<Residue PHE het=  resseq=233 icode= >\n",
      "<Residue LEU het=  resseq=234 icode= >\n",
      "<Residue ILE het=  resseq=235 icode= >\n",
      "<Residue PRO het=  resseq=236 icode= >\n",
      "<Residue ALA het=  resseq=237 icode= >\n",
      "<Residue VAL het=  resseq=238 icode= >\n",
      "<Residue GLU het=  resseq=239 icode= >\n",
      "<Residue LEU het=  resseq=240 icode= >\n",
      "<Residue ILE het=  resseq=241 icode= >\n",
      "<Residue VAL het=  resseq=242 icode= >\n",
      "<Residue LYS het=  resseq=243 icode= >\n",
      "<Residue LEU het=  resseq=244 icode= >\n",
      "<Residue ARG het=  resseq=245 icode= >\n",
      "<Residue PHE het=  resseq=246 icode= >\n",
      "<Residue MET het=  resseq=247 icode= >\n",
      "<Residue PRO het=  resseq=248 icode= >\n",
      "<Residue ARG het=  resseq=249 icode= >\n",
      "<Residue ASN het=  resseq=250 icode= >\n",
      "<Residue GLY het=  resseq=251 icode= >\n",
      "<Residue THR het=  resseq=252 icode= >\n",
      "<Residue GLY het=  resseq=253 icode= >\n",
      "<Residue VAL het=  resseq=254 icode= >\n",
      "<Residue LEU het=  resseq=255 icode= >\n",
      "<Residue ILE het=  resseq=256 icode= >\n",
      "<Residue LEU het=  resseq=257 icode= >\n",
      "<Residue SER het=  resseq=258 icode= >\n",
      "<Residue PRO het=  resseq=259 icode= >\n",
      "<Residue THR het=  resseq=260 icode= >\n",
      "<Residue ARG het=  resseq=261 icode= >\n",
      "<Residue GLU het=  resseq=262 icode= >\n",
      "<Residue LEU het=  resseq=263 icode= >\n",
      "<Residue ALA het=  resseq=264 icode= >\n",
      "<Residue MET het=  resseq=265 icode= >\n",
      "<Residue GLN het=  resseq=266 icode= >\n",
      "<Residue THR het=  resseq=267 icode= >\n",
      "<Residue PHE het=  resseq=268 icode= >\n",
      "<Residue GLY het=  resseq=269 icode= >\n",
      "<Residue VAL het=  resseq=270 icode= >\n",
      "<Residue LEU het=  resseq=271 icode= >\n",
      "<Residue LYS het=  resseq=272 icode= >\n",
      "<Residue GLU het=  resseq=273 icode= >\n",
      "<Residue LEU het=  resseq=274 icode= >\n",
      "<Residue MET het=  resseq=275 icode= >\n",
      "<Residue THR het=  resseq=276 icode= >\n",
      "<Residue HIS het=  resseq=277 icode= >\n",
      "<Residue HIS het=  resseq=278 icode= >\n",
      "<Residue VAL het=  resseq=279 icode= >\n",
      "<Residue HIS het=  resseq=280 icode= >\n",
      "<Residue THR het=  resseq=281 icode= >\n",
      "<Residue TYR het=  resseq=282 icode= >\n",
      "<Residue GLY het=  resseq=283 icode= >\n",
      "<Residue LEU het=  resseq=284 icode= >\n",
      "<Residue ILE het=  resseq=285 icode= >\n",
      "<Residue MET het=  resseq=286 icode= >\n",
      "<Residue GLY het=  resseq=287 icode= >\n",
      "<Residue GLY het=  resseq=288 icode= >\n",
      "<Residue SER het=  resseq=289 icode= >\n",
      "<Residue ASN het=  resseq=290 icode= >\n",
      "<Residue ARG het=  resseq=291 icode= >\n",
      "<Residue SER het=  resseq=292 icode= >\n",
      "<Residue ALA het=  resseq=293 icode= >\n",
      "<Residue GLU het=  resseq=294 icode= >\n",
      "<Residue ALA het=  resseq=295 icode= >\n",
      "<Residue GLN het=  resseq=296 icode= >\n",
      "<Residue LYS het=  resseq=297 icode= >\n",
      "<Residue LEU het=  resseq=298 icode= >\n",
      "<Residue GLY het=  resseq=299 icode= >\n",
      "<Residue ASN het=  resseq=300 icode= >\n",
      "<Residue GLY het=  resseq=301 icode= >\n",
      "<Residue ILE het=  resseq=302 icode= >\n",
      "<Residue ASN het=  resseq=303 icode= >\n",
      "<Residue ILE het=  resseq=304 icode= >\n",
      "<Residue ILE het=  resseq=305 icode= >\n",
      "<Residue VAL het=  resseq=306 icode= >\n",
      "<Residue ALA het=  resseq=307 icode= >\n",
      "<Residue THR het=  resseq=308 icode= >\n",
      "<Residue PRO het=  resseq=309 icode= >\n",
      "<Residue GLY het=  resseq=310 icode= >\n",
      "<Residue ARG het=  resseq=311 icode= >\n",
      "<Residue LEU het=  resseq=312 icode= >\n",
      "<Residue LEU het=  resseq=313 icode= >\n",
      "<Residue ASP het=  resseq=314 icode= >\n",
      "<Residue HIS het=  resseq=315 icode= >\n",
      "<Residue MET het=  resseq=316 icode= >\n",
      "<Residue GLN het=  resseq=317 icode= >\n",
      "<Residue ASN het=  resseq=318 icode= >\n",
      "<Residue THR het=  resseq=319 icode= >\n",
      "<Residue PRO het=  resseq=320 icode= >\n",
      "<Residue GLY het=  resseq=321 icode= >\n",
      "<Residue PHE het=  resseq=322 icode= >\n",
      "<Residue MET het=  resseq=323 icode= >\n",
      "<Residue TYR het=  resseq=324 icode= >\n",
      "<Residue LYS het=  resseq=325 icode= >\n",
      "<Residue ASN het=  resseq=326 icode= >\n",
      "<Residue LEU het=  resseq=327 icode= >\n",
      "<Residue GLN het=  resseq=328 icode= >\n",
      "<Residue CYS het=  resseq=329 icode= >\n",
      "<Residue LEU het=  resseq=330 icode= >\n",
      "<Residue VAL het=  resseq=331 icode= >\n",
      "<Residue ILE het=  resseq=332 icode= >\n",
      "<Residue ASP het=  resseq=333 icode= >\n",
      "<Residue GLU het=  resseq=334 icode= >\n",
      "<Residue ALA het=  resseq=335 icode= >\n",
      "<Residue ASP het=  resseq=336 icode= >\n",
      "<Residue ARG het=  resseq=337 icode= >\n",
      "<Residue ILE het=  resseq=338 icode= >\n",
      "<Residue LEU het=  resseq=339 icode= >\n",
      "<Residue ASP het=  resseq=340 icode= >\n",
      "<Residue VAL het=  resseq=341 icode= >\n",
      "<Residue GLY het=  resseq=342 icode= >\n",
      "<Residue PHE het=  resseq=343 icode= >\n",
      "<Residue GLU het=  resseq=344 icode= >\n",
      "<Residue GLU het=  resseq=345 icode= >\n",
      "<Residue GLU het=  resseq=346 icode= >\n",
      "<Residue LEU het=  resseq=347 icode= >\n",
      "<Residue LYS het=  resseq=348 icode= >\n",
      "<Residue GLN het=  resseq=349 icode= >\n",
      "<Residue ILE het=  resseq=350 icode= >\n",
      "<Residue ILE het=  resseq=351 icode= >\n",
      "<Residue LYS het=  resseq=352 icode= >\n",
      "<Residue LEU het=  resseq=353 icode= >\n",
      "<Residue LEU het=  resseq=354 icode= >\n",
      "<Residue PRO het=  resseq=355 icode= >\n",
      "<Residue THR het=  resseq=356 icode= >\n",
      "<Residue ARG het=  resseq=357 icode= >\n",
      "<Residue ARG het=  resseq=358 icode= >\n",
      "<Residue GLN het=  resseq=359 icode= >\n",
      "<Residue THR het=  resseq=360 icode= >\n",
      "<Residue MET het=  resseq=361 icode= >\n",
      "<Residue LEU het=  resseq=362 icode= >\n",
      "<Residue PHE het=  resseq=363 icode= >\n",
      "<Residue SER het=  resseq=364 icode= >\n",
      "<Residue ALA het=  resseq=365 icode= >\n",
      "<Residue THR het=  resseq=366 icode= >\n",
      "<Residue GLN het=  resseq=367 icode= >\n",
      "<Residue THR het=  resseq=368 icode= >\n",
      "<Residue ARG het=  resseq=369 icode= >\n",
      "<Residue LYS het=  resseq=370 icode= >\n",
      "<Residue VAL het=  resseq=371 icode= >\n",
      "<Residue GLU het=  resseq=372 icode= >\n",
      "<Residue ASP het=  resseq=373 icode= >\n",
      "<Residue LEU het=  resseq=374 icode= >\n",
      "<Residue ALA het=  resseq=375 icode= >\n",
      "<Residue ARG het=  resseq=376 icode= >\n",
      "<Residue ILE het=  resseq=377 icode= >\n",
      "<Residue SER het=  resseq=378 icode= >\n",
      "<Residue LEU het=  resseq=379 icode= >\n",
      "<Residue LYS het=  resseq=380 icode= >\n",
      "<Residue LYS het=  resseq=381 icode= >\n",
      "<Residue GLU het=  resseq=382 icode= >\n",
      "<Residue PRO het=  resseq=383 icode= >\n",
      "<Residue LEU het=  resseq=384 icode= >\n",
      "<Residue TYR het=  resseq=385 icode= >\n",
      "<Residue VAL het=  resseq=386 icode= >\n",
      "<Residue PO4 het=H_PO4 resseq=1 icode= >\n",
      "<Residue  MG het=H_ MG resseq=388 icode= >\n",
      "<Residue  MG het=H_ MG resseq=389 icode= >\n",
      "<Residue SCN het=H_SCN resseq=390 icode= >\n",
      "<Residue HOH het=W resseq=2 icode= >\n",
      "<Residue HOH het=W resseq=5 icode= >\n",
      "<Residue HOH het=W resseq=7 icode= >\n",
      "<Residue HOH het=W resseq=15 icode= >\n",
      "<Residue HOH het=W resseq=16 icode= >\n",
      "<Residue HOH het=W resseq=18 icode= >\n",
      "<Residue HOH het=W resseq=19 icode= >\n",
      "<Residue HOH het=W resseq=22 icode= >\n",
      "<Residue HOH het=W resseq=23 icode= >\n",
      "<Residue HOH het=W resseq=25 icode= >\n",
      "<Residue HOH het=W resseq=27 icode= >\n",
      "<Residue HOH het=W resseq=28 icode= >\n",
      "<Residue HOH het=W resseq=29 icode= >\n",
      "<Residue HOH het=W resseq=30 icode= >\n",
      "<Residue HOH het=W resseq=31 icode= >\n",
      "<Residue HOH het=W resseq=32 icode= >\n",
      "<Residue HOH het=W resseq=34 icode= >\n",
      "<Residue HOH het=W resseq=37 icode= >\n",
      "<Residue HOH het=W resseq=38 icode= >\n",
      "<Residue HOH het=W resseq=39 icode= >\n",
      "<Residue HOH het=W resseq=45 icode= >\n",
      "<Residue HOH het=W resseq=48 icode= >\n",
      "<Residue HOH het=W resseq=49 icode= >\n",
      "<Residue HOH het=W resseq=50 icode= >\n",
      "<Residue HOH het=W resseq=52 icode= >\n",
      "<Residue GLY het=  resseq=169 icode= >\n",
      "<Residue LEU het=  resseq=170 icode= >\n",
      "<Residue THR het=  resseq=171 icode= >\n",
      "<Residue GLY het=  resseq=172 icode= >\n",
      "<Residue ALA het=  resseq=173 icode= >\n",
      "<Residue PHE het=  resseq=174 icode= >\n",
      "<Residue GLU het=  resseq=175 icode= >\n",
      "<Residue ASP het=  resseq=176 icode= >\n",
      "<Residue THR het=  resseq=177 icode= >\n",
      "<Residue SER het=  resseq=178 icode= >\n",
      "<Residue PHE het=  resseq=179 icode= >\n",
      "<Residue LEU het=  resseq=185 icode= >\n",
      "<Residue VAL het=  resseq=186 icode= >\n",
      "<Residue ASN het=  resseq=187 icode= >\n",
      "<Residue GLU het=  resseq=188 icode= >\n",
      "<Residue ASN het=  resseq=189 icode= >\n",
      "<Residue THR het=  resseq=190 icode= >\n",
      "<Residue LEU het=  resseq=191 icode= >\n",
      "<Residue LYS het=  resseq=192 icode= >\n",
      "<Residue ALA het=  resseq=193 icode= >\n",
      "<Residue ILE het=  resseq=194 icode= >\n",
      "<Residue LYS het=  resseq=195 icode= >\n",
      "<Residue GLU het=  resseq=196 icode= >\n",
      "<Residue MET het=  resseq=197 icode= >\n",
      "<Residue GLY het=  resseq=198 icode= >\n",
      "<Residue PHE het=  resseq=199 icode= >\n",
      "<Residue THR het=  resseq=200 icode= >\n",
      "<Residue ASN het=  resseq=201 icode= >\n",
      "<Residue MET het=  resseq=202 icode= >\n",
      "<Residue THR het=  resseq=203 icode= >\n",
      "<Residue GLU het=  resseq=204 icode= >\n",
      "<Residue ILE het=  resseq=205 icode= >\n",
      "<Residue GLN het=  resseq=206 icode= >\n",
      "<Residue HIS het=  resseq=207 icode= >\n",
      "<Residue LYS het=  resseq=208 icode= >\n",
      "<Residue SER het=  resseq=209 icode= >\n",
      "<Residue ILE het=  resseq=210 icode= >\n",
      "<Residue ARG het=  resseq=211 icode= >\n",
      "<Residue PRO het=  resseq=212 icode= >\n",
      "<Residue LEU het=  resseq=213 icode= >\n",
      "<Residue LEU het=  resseq=214 icode= >\n",
      "<Residue GLU het=  resseq=215 icode= >\n",
      "<Residue GLY het=  resseq=216 icode= >\n",
      "<Residue ARG het=  resseq=217 icode= >\n",
      "<Residue ASP het=  resseq=218 icode= >\n",
      "<Residue LEU het=  resseq=219 icode= >\n",
      "<Residue LEU het=  resseq=220 icode= >\n",
      "<Residue ALA het=  resseq=221 icode= >\n",
      "<Residue ALA het=  resseq=222 icode= >\n",
      "<Residue ALA het=  resseq=223 icode= >\n",
      "<Residue LYS het=  resseq=224 icode= >\n",
      "<Residue THR het=  resseq=225 icode= >\n",
      "<Residue GLY het=  resseq=226 icode= >\n",
      "<Residue SER het=  resseq=227 icode= >\n",
      "<Residue GLY het=  resseq=228 icode= >\n",
      "<Residue LYS het=  resseq=229 icode= >\n",
      "<Residue THR het=  resseq=230 icode= >\n",
      "<Residue LEU het=  resseq=231 icode= >\n",
      "<Residue ALA het=  resseq=232 icode= >\n",
      "<Residue PHE het=  resseq=233 icode= >\n",
      "<Residue LEU het=  resseq=234 icode= >\n",
      "<Residue ILE het=  resseq=235 icode= >\n",
      "<Residue PRO het=  resseq=236 icode= >\n",
      "<Residue ALA het=  resseq=237 icode= >\n",
      "<Residue VAL het=  resseq=238 icode= >\n",
      "<Residue GLU het=  resseq=239 icode= >\n",
      "<Residue LEU het=  resseq=240 icode= >\n",
      "<Residue ILE het=  resseq=241 icode= >\n",
      "<Residue VAL het=  resseq=242 icode= >\n",
      "<Residue LYS het=  resseq=243 icode= >\n",
      "<Residue LEU het=  resseq=244 icode= >\n",
      "<Residue ARG het=  resseq=245 icode= >\n",
      "<Residue PHE het=  resseq=246 icode= >\n",
      "<Residue MET het=  resseq=247 icode= >\n",
      "<Residue PRO het=  resseq=248 icode= >\n",
      "<Residue ARG het=  resseq=249 icode= >\n",
      "<Residue ASN het=  resseq=250 icode= >\n",
      "<Residue GLY het=  resseq=251 icode= >\n",
      "<Residue THR het=  resseq=252 icode= >\n",
      "<Residue GLY het=  resseq=253 icode= >\n",
      "<Residue VAL het=  resseq=254 icode= >\n",
      "<Residue LEU het=  resseq=255 icode= >\n",
      "<Residue ILE het=  resseq=256 icode= >\n",
      "<Residue LEU het=  resseq=257 icode= >\n",
      "<Residue SER het=  resseq=258 icode= >\n",
      "<Residue PRO het=  resseq=259 icode= >\n",
      "<Residue THR het=  resseq=260 icode= >\n",
      "<Residue ARG het=  resseq=261 icode= >\n",
      "<Residue GLU het=  resseq=262 icode= >\n",
      "<Residue LEU het=  resseq=263 icode= >\n",
      "<Residue ALA het=  resseq=264 icode= >\n",
      "<Residue MET het=  resseq=265 icode= >\n",
      "<Residue GLN het=  resseq=266 icode= >\n",
      "<Residue THR het=  resseq=267 icode= >\n",
      "<Residue PHE het=  resseq=268 icode= >\n",
      "<Residue GLY het=  resseq=269 icode= >\n",
      "<Residue VAL het=  resseq=270 icode= >\n",
      "<Residue LEU het=  resseq=271 icode= >\n",
      "<Residue LYS het=  resseq=272 icode= >\n",
      "<Residue GLU het=  resseq=273 icode= >\n",
      "<Residue LEU het=  resseq=274 icode= >\n",
      "<Residue MET het=  resseq=275 icode= >\n",
      "<Residue THR het=  resseq=276 icode= >\n",
      "<Residue HIS het=  resseq=277 icode= >\n",
      "<Residue HIS het=  resseq=278 icode= >\n",
      "<Residue VAL het=  resseq=279 icode= >\n",
      "<Residue HIS het=  resseq=280 icode= >\n",
      "<Residue THR het=  resseq=281 icode= >\n",
      "<Residue TYR het=  resseq=282 icode= >\n",
      "<Residue GLY het=  resseq=283 icode= >\n",
      "<Residue LEU het=  resseq=284 icode= >\n",
      "<Residue ILE het=  resseq=285 icode= >\n",
      "<Residue MET het=  resseq=286 icode= >\n",
      "<Residue GLY het=  resseq=287 icode= >\n",
      "<Residue GLY het=  resseq=288 icode= >\n",
      "<Residue SER het=  resseq=289 icode= >\n",
      "<Residue ASN het=  resseq=290 icode= >\n",
      "<Residue ARG het=  resseq=291 icode= >\n",
      "<Residue SER het=  resseq=292 icode= >\n",
      "<Residue ALA het=  resseq=293 icode= >\n",
      "<Residue GLU het=  resseq=294 icode= >\n",
      "<Residue ALA het=  resseq=295 icode= >\n",
      "<Residue GLN het=  resseq=296 icode= >\n",
      "<Residue LYS het=  resseq=297 icode= >\n",
      "<Residue LEU het=  resseq=298 icode= >\n",
      "<Residue GLY het=  resseq=299 icode= >\n",
      "<Residue ASN het=  resseq=300 icode= >\n",
      "<Residue GLY het=  resseq=301 icode= >\n",
      "<Residue ILE het=  resseq=302 icode= >\n",
      "<Residue ASN het=  resseq=303 icode= >\n",
      "<Residue ILE het=  resseq=304 icode= >\n",
      "<Residue ILE het=  resseq=305 icode= >\n",
      "<Residue VAL het=  resseq=306 icode= >\n",
      "<Residue ALA het=  resseq=307 icode= >\n",
      "<Residue THR het=  resseq=308 icode= >\n",
      "<Residue PRO het=  resseq=309 icode= >\n",
      "<Residue GLY het=  resseq=310 icode= >\n",
      "<Residue ARG het=  resseq=311 icode= >\n",
      "<Residue LEU het=  resseq=312 icode= >\n",
      "<Residue LEU het=  resseq=313 icode= >\n",
      "<Residue ASP het=  resseq=314 icode= >\n",
      "<Residue HIS het=  resseq=315 icode= >\n",
      "<Residue MET het=  resseq=316 icode= >\n",
      "<Residue GLN het=  resseq=317 icode= >\n",
      "<Residue ASN het=  resseq=318 icode= >\n",
      "<Residue THR het=  resseq=319 icode= >\n",
      "<Residue PRO het=  resseq=320 icode= >\n",
      "<Residue GLY het=  resseq=321 icode= >\n",
      "<Residue PHE het=  resseq=322 icode= >\n",
      "<Residue MET het=  resseq=323 icode= >\n",
      "<Residue TYR het=  resseq=324 icode= >\n",
      "<Residue LYS het=  resseq=325 icode= >\n",
      "<Residue ASN het=  resseq=326 icode= >\n",
      "<Residue LEU het=  resseq=327 icode= >\n",
      "<Residue GLN het=  resseq=328 icode= >\n",
      "<Residue CYS het=  resseq=329 icode= >\n",
      "<Residue LEU het=  resseq=330 icode= >\n",
      "<Residue VAL het=  resseq=331 icode= >\n",
      "<Residue ILE het=  resseq=332 icode= >\n",
      "<Residue ASP het=  resseq=333 icode= >\n",
      "<Residue GLU het=  resseq=334 icode= >\n",
      "<Residue ALA het=  resseq=335 icode= >\n",
      "<Residue ASP het=  resseq=336 icode= >\n",
      "<Residue ARG het=  resseq=337 icode= >\n",
      "<Residue ILE het=  resseq=338 icode= >\n",
      "<Residue LEU het=  resseq=339 icode= >\n",
      "<Residue ASP het=  resseq=340 icode= >\n",
      "<Residue VAL het=  resseq=341 icode= >\n",
      "<Residue GLY het=  resseq=342 icode= >\n",
      "<Residue PHE het=  resseq=343 icode= >\n",
      "<Residue GLU het=  resseq=344 icode= >\n",
      "<Residue GLU het=  resseq=345 icode= >\n",
      "<Residue GLU het=  resseq=346 icode= >\n",
      "<Residue LEU het=  resseq=347 icode= >\n",
      "<Residue LYS het=  resseq=348 icode= >\n",
      "<Residue GLN het=  resseq=349 icode= >\n",
      "<Residue ILE het=  resseq=350 icode= >\n",
      "<Residue ILE het=  resseq=351 icode= >\n",
      "<Residue LYS het=  resseq=352 icode= >\n",
      "<Residue LEU het=  resseq=353 icode= >\n",
      "<Residue LEU het=  resseq=354 icode= >\n",
      "<Residue PRO het=  resseq=355 icode= >\n",
      "<Residue THR het=  resseq=356 icode= >\n",
      "<Residue ARG het=  resseq=357 icode= >\n",
      "<Residue ARG het=  resseq=358 icode= >\n",
      "<Residue GLN het=  resseq=359 icode= >\n",
      "<Residue THR het=  resseq=360 icode= >\n",
      "<Residue MET het=  resseq=361 icode= >\n",
      "<Residue LEU het=  resseq=362 icode= >\n",
      "<Residue PHE het=  resseq=363 icode= >\n",
      "<Residue SER het=  resseq=364 icode= >\n",
      "<Residue ALA het=  resseq=365 icode= >\n",
      "<Residue THR het=  resseq=366 icode= >\n",
      "<Residue GLN het=  resseq=367 icode= >\n",
      "<Residue THR het=  resseq=368 icode= >\n",
      "<Residue ARG het=  resseq=369 icode= >\n",
      "<Residue LYS het=  resseq=370 icode= >\n",
      "<Residue VAL het=  resseq=371 icode= >\n",
      "<Residue GLU het=  resseq=372 icode= >\n",
      "<Residue ASP het=  resseq=373 icode= >\n",
      "<Residue LEU het=  resseq=374 icode= >\n",
      "<Residue ALA het=  resseq=375 icode= >\n",
      "<Residue ARG het=  resseq=376 icode= >\n",
      "<Residue ILE het=  resseq=377 icode= >\n",
      "<Residue SER het=  resseq=378 icode= >\n",
      "<Residue LEU het=  resseq=379 icode= >\n",
      "<Residue LYS het=  resseq=380 icode= >\n",
      "<Residue LYS het=  resseq=381 icode= >\n",
      "<Residue GLU het=  resseq=382 icode= >\n",
      "<Residue PRO het=  resseq=383 icode= >\n",
      "<Residue LEU het=  resseq=384 icode= >\n",
      "<Residue TYR het=  resseq=385 icode= >\n",
      "<Residue VAL het=  resseq=386 icode= >\n",
      "<Residue SCN het=H_SCN resseq=2 icode= >\n",
      "<Residue HOH het=W resseq=6 icode= >\n",
      "<Residue HOH het=W resseq=8 icode= >\n",
      "<Residue HOH het=W resseq=10 icode= >\n",
      "<Residue HOH het=W resseq=11 icode= >\n",
      "<Residue HOH het=W resseq=12 icode= >\n",
      "<Residue HOH het=W resseq=13 icode= >\n",
      "<Residue HOH het=W resseq=14 icode= >\n",
      "<Residue HOH het=W resseq=17 icode= >\n",
      "<Residue HOH het=W resseq=20 icode= >\n",
      "<Residue HOH het=W resseq=21 icode= >\n",
      "<Residue HOH het=W resseq=24 icode= >\n",
      "<Residue HOH het=W resseq=33 icode= >\n",
      "<Residue HOH het=W resseq=40 icode= >\n",
      "<Residue HOH het=W resseq=42 icode= >\n",
      "<Residue HOH het=W resseq=46 icode= >\n",
      "<Residue HOH het=W resseq=47 icode= >\n",
      "<Residue HOH het=W resseq=51 icode= >\n",
      "<Residue HOH het=W resseq=53 icode= >\n",
      "<Residue HOH het=W resseq=55 icode= >\n",
      "<Residue HOH het=W resseq=56 icode= >\n"
     ]
    }
   ],
   "source": [
    "residues = model.get_residues()\n",
    "for residue in residues:\n",
    "    print(residue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "8e504619",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.8\n",
      "x-ray diffraction\n"
     ]
    }
   ],
   "source": [
    "#some extra information maybe useless\n",
    "resolution = structure.header['resolution']\n",
    "print(resolution)\n",
    "structure_reference=structure.header[\"structure_method\"]\n",
    "print(structure_reference)\n",
    "#structure_method,name,head,deposition_date,release_date,journal_reference,author,compound"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8beb1bff",
   "metadata": {},
   "source": [
    "-the goal is print the 3d structure, bur for this we need pymol so:\n",
    "How can we install pymol or a package pymol??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "6e1fe3c3",
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'pymol'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Input \u001b[1;32mIn [81]\u001b[0m, in \u001b[0;36m<cell line: 2>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mBio\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mPDB\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;241m*\u001b[39m\n\u001b[1;32m----> 2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mpymol\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m cmd\n\u001b[0;32m      3\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mPDB\u001b[39m(ID):\n\u001b[0;32m      4\u001b[0m     \u001b[38;5;124;03m\"\"\"\"\u001b[39;00m\n\u001b[0;32m      5\u001b[0m \u001b[38;5;124;03m    ID: fornecer o ID da PDB\u001b[39;00m\n\u001b[0;32m      6\u001b[0m \u001b[38;5;124;03m    Return: A estrutura 3D da proteína\u001b[39;00m\n\u001b[0;32m      7\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'pymol'"
     ]
    }
   ],
   "source": [
    "#tenho de importar o pymol\n",
    "from Bio.PDB import *\n",
    "from pymol import cmd\n",
    "def PDB(ID):\n",
    "    \"\"\"\"\n",
    "    ID: fornecer o ID da PDB\n",
    "    Return: A estrutura 3D da proteína\n",
    "    \"\"\"\n",
    "    pdbl = PDBList()\n",
    "    file = pdbl.retrieve_pdb_file(ID, pdir='.', file_format='pdb')\n",
    "    pymol.finish_launching()\n",
    "\n",
    "    pymol.cmd.load(file, ID)\n",
    "    pymol.cmd.disable(\"all\")\n",
    "    pymol.cmd.enable(ID)\n",
    "    pymol.cmd.orient()\n",
    "    pymol.cmd.zoom()\n",
    "    pymol.cmd.png(\"ORF3a_6XDC.png\", 3000, 3000, dpi=500, ray=1)\n",
    "\n",
    "    pymol.cmd.quit()\n",
    "PDB(\"3ly5\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e230f6cf",
   "metadata": {},
   "source": [
    "# Multiple Alignment and Phylogenies"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28281ed8",
   "metadata": {},
   "source": [
    "## DDX18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "f2b7949d",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIXML \n",
    "from Bio.Blast import NCBIWWW \n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Seq import Seq\n",
    "from Bio import AlignIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "a4c4d727",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle=NCBIWWW.qblast(\"blastp\",\"nr\",seq_p_ddx18)#blast with the protein translate of my cds\n",
    "with open(\"blastp_ddx18.txt\",\"w\") as out_handle:\n",
    "    out_handle.write(result_handle.read())\n",
    "result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "2dc1797f",
   "metadata": {},
   "outputs": [],
   "source": [
    "record = SeqIO.parse(open(\"blastp_file_ddx18.txt\"), format=\"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "9713569f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment with 0 rows and 0 columns\n",
      "0\n",
      "[]\n"
     ]
    }
   ],
   "source": [
    "sequences = []\n",
    "for seq in record:\n",
    "    sequences.append(seq)\n",
    "\n",
    "alin = MultipleSeqAlignment(sequences)\n",
    "print(alin)\n",
    "print(alin.get_alignment_length())\n",
    "print(sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "7affa730",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Unknown format 'text'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Input \u001b[1;32mIn [82]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[43mSeqIO\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconvert\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mblastp_file_ddx18.txt\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtext\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mblastp_file_ddx18.fasta\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mfasta\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SeqIO\\__init__.py:1077\u001b[0m, in \u001b[0;36mconvert\u001b[1;34m(in_file, in_format, out_file, out_format, molecule_type)\u001b[0m\n\u001b[0;32m   1075\u001b[0m     count \u001b[38;5;241m=\u001b[39m f(in_file, out_file)\n\u001b[0;32m   1076\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m-> 1077\u001b[0m     records \u001b[38;5;241m=\u001b[39m \u001b[43mparse\u001b[49m\u001b[43m(\u001b[49m\u001b[43min_file\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43min_format\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1078\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m molecule_type:\n\u001b[0;32m   1079\u001b[0m         \u001b[38;5;66;03m# Edit the records on the fly to set molecule type\u001b[39;00m\n\u001b[0;32m   1081\u001b[0m         \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mover_ride\u001b[39m(record):\n",
      "File \u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SeqIO\\__init__.py:611\u001b[0m, in \u001b[0;36mparse\u001b[1;34m(handle, format, alphabet)\u001b[0m\n\u001b[0;32m    608\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mformat\u001b[39m \u001b[38;5;129;01min\u001b[39;00m AlignIO\u001b[38;5;241m.\u001b[39m_FormatToIterator:\n\u001b[0;32m    609\u001b[0m     \u001b[38;5;66;03m# Use Bio.AlignIO to read in the alignments\u001b[39;00m\n\u001b[0;32m    610\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m (r \u001b[38;5;28;01mfor\u001b[39;00m alignment \u001b[38;5;129;01min\u001b[39;00m AlignIO\u001b[38;5;241m.\u001b[39mparse(handle, \u001b[38;5;28mformat\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m r \u001b[38;5;129;01min\u001b[39;00m alignment)\n\u001b[1;32m--> 611\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mUnknown format \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m \u001b[38;5;28mformat\u001b[39m)\n",
      "\u001b[1;31mValueError\u001b[0m: Unknown format 'text'"
     ]
    }
   ],
   "source": [
    "SeqIO.convert(\"blastp_file_ddx18.txt\", \"text\", \"blastp_file_ddx18.fasta\", \"fasta\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20930c3c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
