{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "59e9a631",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIXML \n",
    "from Bio.Blast import NCBIWWW\n",
    "import requests, sys, json\n",
    "import re\n",
    "from Bio import SeqIO\n",
    "from Bio import Entrez\n",
    "from Bio import Medline\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "from Bio import SearchIO\n",
    "from Bio.SwissProt import KeyWList\n",
    "from Bio import SwissProt\n",
    "from Bio import ExPASy\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Seq import Seq\n",
    "from Bio import AlignIO\n",
    "from Bio.Align import AlignInfo\n",
    "from Bio.SeqFeature import SeqFeature, FeatureLocation\n",
    "from Bio.Phylo.TreeConstruction import DistanceCalculator\n",
    "from Bio.Phylo.TreeConstruction import DistanceTreeConstructor\n",
    "from Bio import Phylo"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8eceb6ab",
   "metadata": {},
   "source": [
    "## Searching for literature for GLI2 gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3cd6b9fd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PMID: 36662414\n",
      "title: Gli2-induced lncRNA Peg13 alleviates cerebral ischemia-reperfusion injury by suppressing Yy1 transcription in a PRC2 complex-dependent manner.\n",
      "abstract: Endothelial cell dysfunction plays an important role in cerebral ischemia-reperfusion (I/R) injury. LncRNA Peg13 is reported to be down-regulated in brain microvascular endothelial cells (BMVECs) induced by glucose-oxygen deprivation (OGD), but the mechanism of its involvement in I/R progression remains to be further explored. Here, mouse BMVECs (bEnd.3 cells) were treated with OGD / reoxygenation (OGD/R) to simulate I/R injury in vitro. Peg13 and Gli2 expression was decreased in OGD/R-treated bEnd.3 cells. And overexpression of Peg13 or Gli2 prevented OGD/R-induced reduction in cell migration and angiogenesis, as well as upregulation in cell apoptosis and oxidative stress levels. Mechanism exploration showed that Gli2 promoted the transcription of Peg13. And Peg13 repressed Yy1 transcription by binding to Ezh2 (a key subunit of PRC2 complex) and inducing the enrichment of H3K27me3 in Yy1 promoter region, thereby suppressing the transcriptional inhibition effect of Yy1 on Notch3 and promoting the expression of Notch3. Consistently, Notch3 overexpression hindered OGD/R-induced endothelium dysfunction. In addition, a brain I/R injury model was established using middle cerebral artery occlusion surgery. And lentivirus-mediated Gli2 and Peg13 overexpression vectors were injected into mice via the lateral ventricle one week before surgery. The results showed that overexpression of Peg13 or Gli2 alleviated I/R-induced neurological deficit, cerebral infarct and cerebral edema. And simultaneous overexpression of Peg13 and Gli2 showed a better protective effect than overexpression of Gli2 or Peg13 alone. In conclusion, Peg13 regulated by Gli2 inhibits Yy1 transcription in a PCR2 complex-dependent manner, and blocks the transcriptional repression of Notch3 by Yy1, thereby exerting neuroprotective effects on cerebral I/R injury.\n",
      "authors: ['Li Y', 'Liu C', 'Fan H', 'Du Y', 'Zhang R', 'Zhan S', 'Zhang G', 'Bu N']\n",
      "source: Metab Brain Dis. 2023 Jan 20. doi: 10.1007/s11011-023-01159-w.\n",
      "\n",
      "PMID: 36651276\n",
      "title: Annotation of uORFs in the OMIM genes allows to reveal pathogenic variants in 5'UTRs.\n",
      "abstract: An increasing number of studies emphasize the role of non-coding variants in the development of hereditary diseases. However, the interpretation of such variants in clinical genetic testing still remains a critical challenge due to poor knowledge of their pathogenicity mechanisms. It was previously shown that variants in 5'-untranslated regions (5'UTRs) can lead to hereditary diseases due to disruption of upstream open reading frames (uORFs). Here, we performed a manual annotation of upstream translation initiation sites (TISs) in human disease-associated genes from the OMIM database and revealed  approximately 4.7 thousand of TISs related to uORFs. We compared our TISs with the previous studies and provided a list of 'high confidence' uORFs. Using a luciferase assay, we experimentally validated the translation of uORFs in the ETFDH, PAX9, MAST1, HTT, TTN,GLI2 and COL2A1 genes, as well as existence of N-terminal CDS extension in the ZIC2 gene. Besides, we created a tool to annotate the effects of genetic variants located in uORFs. We revealed the variants from the HGMD and ClinVar databases that disrupt uORFs and thereby could lead to Mendelian disorders. We also showed that the distribution of uORFs-affecting variants differs between pathogenic and population variants. Finally, drawing on manually curated data, we developed a machine-learning algorithm that allows us to predict the TISs in other human genes.\n",
      "authors: ['Filatova A', 'Reveguk I', 'Piatkova M', 'Bessonova D', 'Kuziakova O', 'Demakova V', 'Romanishin A', 'Fishman V', 'Imanmalik Y', 'Chekanov N', 'Skitchenko R', 'Barbitoff Y', 'Kardymon O', 'Skoblov M']\n",
      "source: Nucleic Acids Res. 2023 Jan 18:gkac1247. doi: 10.1093/nar/gkac1247.\n",
      "\n",
      "PMID: 36650051\n",
      "title: A temporal in vivo catalogue of chromatin accessibility and expression profiles in pineoblastoma reveals a prevalent role for repressor elements.\n",
      "abstract: Pediatric pineoblastomas (PBs) are rare and aggressive tumors of grade IV histology. Although some oncogenic drivers are characterized, including germline mutations in RB1 and DICER1, the role of epigenetic deregulation and cis-regulatory regions in PB pathogenesis and progression is largely unknown. Here, we generated genome-wide gene expression, chromatin accessibility, and H3K27ac profiles covering key time-points of PB initiation and progression from pineal tissues of a mouse model of Ccnd1-driven pineoblastoma. We identified PB-specific enhancers and super-enhancers and found that, in some cases, the accessible genome dynamic precedes the transcriptome, a characteristic that is underexplored in tumor progression. During progression of PB, newly acquired open chromatin regions lacking H3K27ac signal become enriched for repressive state elements and harbor motifs of repressor transcription factors like HINFP, GLI2, and YY1. Copy number variant analysis identified deletion events specific to the tumorigenic stage, affecting among others the histone genes cluster and Gas1, the growth arrest specific gene. Gene set enrichment analysis and gene expression signatures positioned the model used here close to human PB samples demonstrating the potential of our findings for exploring new avenues in PB management and therapy. Overall, this study reports the first temporal and in vivo cis-regulatory, expression, and accessibility maps in PB.\n",
      "authors: ['Idriss S', 'Hallal M', 'El-Kurdi A', 'Zalzali H', 'El-Rassi I', 'Ehli EA', 'Davis CM', 'Chung PED', 'Gendoo DMA', 'Zacksenhaus E', 'Saab R', 'Khoueiry P']\n",
      "source: Genome Res. 2023 Jan 17:gr.277037.122. doi: 10.1101/gr.277037.122.\n",
      "\n",
      "PMID: 36598638\n",
      "title: Reciprocal FGF19-GLI2 signaling induces epithelial-to-mesenchymal transition to promote lung squamous cell carcinoma metastasis.\n",
      "abstract: PURPOSE: Metastatic lung squamous cell carcinoma (LUSC) is one of the most common causes of cancer death worldwide. As yet, however, the molecular mechanism underlying LUSC metastasis remains elusive. In this study, we report a novel mechanism involving signaling interactions between FGF19 and GLI2 that could drive the progression of LUSC. METHODS: The expression of FGF19 in human LUSC samples was assessed by immunohistochemistry. The concentration of FGF19 in serum samples was assessed by ELISA. RNA sequencing, scratch wound-healing, trans-well, GO analysis, GSEA, luciferase reporter, Western blotting, immunofluorescence and immunohistochemistry assays, as well as an animal model were used to investigate the molecular mechanism underlying FGF19 driven LUSC progression. The therapeutic effect of a GLI2 inhibitor was determined using both in vitro cellular and in vivo animal experiments. RESULTS: We found that FGF19, a member of the fibroblast growth factor family, plays a crucial role in the invasion and metastasis of LUSC, and identified GLI2 as an important downstream effector of FGF19 involved in metastasis. Surprisingly, we found that FGF19 and GLI2 could reciprocally induce the expression of each other, and form a positive feedback loop to promote LUSC cell invasion and metastasis. These findings were corroborated by an association between a poor prognosis of LUSC patients and FGF19/GLI2 co-expression. In addition, we found that the GLI inhibitor GANT61 could effectively reduce FGF19-mediated LUSC invasion and metastasis. CONCLUSION: Our data suggest that FGF19 may serve as a novel biomarker for predicting metastatic LUSC. Intervening with the FGF19-GLI2 feedback loop may be a strategy for the treatment of FGF19-driven LUSC metastasis.\n",
      "authors: ['Zhang Y', 'Wu T', 'Wang Y', 'Chen Z', 'Chen J', 'Lu S', 'Xia W']\n",
      "source: Cell Oncol (Dordr). 2023 Jan 4. doi: 10.1007/s13402-022-00760-y.\n",
      "\n",
      "PMID: 36573458\n",
      "title: LOXL2 reduces 5-FU sensitivity through the Hedgehog/BCL2 signaling pathway in colorectal cancer.\n",
      "abstract: Elevated expression of lysyl oxidase-like 2 (LOXL2) contributes to the malignant tumor progression in multiple cancers. However, the role of LOXL2 in the 5-fluorouracil (5-FU) resistance of colorectal cancer (CRC) remains unclear. This study aimed to explore the effects of LOXL2 on 5-FU sensitivity in CRC. The mRNA and protein levels of LOXL2 were explored in public databases by bioinformatics, validated in clinical tissues using immunohistochemistry, and detected in 5-FU treated cell lines. The 50% inhibitory concentrations (IC50) values were quantified based on the cell viability at different concentrations of 5-FU with CCK-8 assays. Colony formation and flow cytometry assays were performed to measure the proliferation and apoptosis rates. Gene set enrichment and correlation analyses were conducted to identify the probable mechanism of LOXL2 in TCGA samples. Critical molecules of the Hedgehog signaling pathway and anti-apoptotic BCL2 in protein levels were detected with Western blotting. It concluded that LOXL2 was up-regulated and positively linked to the unfavorable prognosis of CRC patients. The LOXL2 expression increased with the rising 5-FU concentrations, especially at 20 and 40 muM. Elevated LOXL2 promoted the resistance to 5-FU, augmented the proliferation, and inhibited 5-FU-induced apoptosis of CRC cells. LOXL2 activated the Hedgehog signaling pathway by promoting the expression of SMO, GLI1, and GLI2, leading to the upregulation of downstream target gene BCL2 in CRC cells. Moreover, the Hedgehog signaling pathway inhibitor cyclopamine blocked the BCL2 upregulation mediated by LOXL2. This study has demonstrated that LOXL2 can reduce 5-FU sensitivity through the Hedgehog/BCL2 signaling pathway in CRC.\n",
      "authors: ['Qiu Z', 'Qiu S', 'Mao W', 'Lin W', 'Peng Q', 'Chang H']\n",
      "source: Exp Biol Med (Maywood). 2022 Dec 27:15353702221139203. doi: 10.1177/15353702221139203.\n",
      "\n",
      "PMID: 36532860\n",
      "title: Expression analysis and regulation of GLI and its correlation with stemness and metabolic alteration in human brain tumor.\n",
      "abstract: GLI gene-mediated hedgehog (Hh) signaling pathway plays a substantial role in brain cancer development and growth including glioblastoma multiforme (GBM), lower-grade glioma (LGG), and medulloblastoma (MB). GLI2 and GLI3 gene expression levels are extremely enhanced in these cancers with poor patient survival. Moreover, GLI genes are correlated with stemness-related factors SOX2, SOX9, POU5F1, and NANOG that work as the driving factors for brain cancer stem cells (CSCs) progression. It's critical to find new ways to combat this deadly malignancy and CSCs. Using in silico approaches, our study explored the role of GLI genes (GLI1, GLI2, and GLI3), the primary transcription factors of the sonic hedgehog (SHH) signaling pathway, in GBM, LGG, MB, and glioblastoma stem-like cells (GSCs). Additionally, we found strong association of angiogenic-related gene VEGFA, metabolic genes ENO1, ENO2, and pluripotency-related genes SOX2, SOX9, NANOG, POU5F1 with GLI genes, suggesting their role in brain tumor initiation and progression. We also studied their transcriptional network and functional category enrichment analysis about brain tumor development to find a better therapeutic strategy against brain cancer and their stem cells. SUPPLEMENTARY INFORMATION: The online version contains supplementary material available at 10.1007/s13205-022-03419-5.\n",
      "authors: ['Agrawal K', 'Chauhan S', 'Kumar D']\n",
      "source: 3 Biotech. 2023 Jan;13(1):10. doi: 10.1007/s13205-022-03419-5. Epub 2022 Dec 15.\n",
      "\n",
      "PMID: 36483737\n",
      "title: Gypenosides ameliorate ductular reaction and liver fibrosis via inhibition of hedgehog signaling.\n",
      "abstract: Backgroud and aims: Ductular reaction (DR) is a common pathological change and thought to have a key role in the pathogenesis and progression of liver fibrosis. Our previous study reported Gypenosides (GPs) ameliorated liver fibrosis, however, the anti-fibrotic mechanisms of GPs are still unclear. Methods: Liver fibrosis was induced in rats by carbon tetrachloride combining with 2-acerylaminofluorene (CCl(4)/2-AAF), and Mdr2 knockout (Mdr2 (-/-)) mice to evaluate the anti-fibrotic role of GPs. In vitro, WB-F344 cells, a hepatic progenitor cells (HPCs) line, with or without Gli1 overexpressing lentiviral vectors, were induced by sodium butyrate (SB) to validate the mechanism of GPs and NPLC0393, the main ingredient of GPs. Results: Both in CCl(4)/2-AAF-treated rats and Mdr2 (-/-) mice, GPs obviously reduced the deposition of collagen and hydroxyproline content, inhibited the activation of hepatic stellate cells and inflammatory cell infiltration. Notably, GPs reduced the expressions of Epcam, CK19, CK7, Dhh, Smo, Ptch2, Gli1 and Gli2. Furthermore, CK19(+) cells co-expressed Gli1, while the number of CK19(+)/Gli1(+) cells was decreased by GPs. In vitro, GPs and NPLC0393 inhibited the differentiation of WB-F344 cells toward a biliary phenotype. Mechanistically, GPs and NPLC0393 protected against DR by inhibiting hedgehog signaling, which was supported by the results that DR, triggered directly by Gli1 overexpressing lentiviral vector was blocked by administration with GPs or NPLC0393. Conclusion: GPs attenuated DR and liver fibrosis by inhibiting hedgehog signaling, which provided more evidences and a novel mechanism of anti-fibrotic effect of GPs.\n",
      "authors: ['Hu Y', 'He X', 'Zhou X', 'Liang Y', 'Fu Y', 'Zhang L', 'Fang J', 'Liu W', 'Chen G', 'Mu Y', 'Zhang H', 'Cai H', 'Liu C', 'Liu P', 'Chen J']\n",
      "source: Front Pharmacol. 2022 Nov 22;13:1033103. doi: 10.3389/fphar.2022.1033103. eCollection 2022.\n",
      "\n",
      "PMID: 36423790\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "title: SHH/GLI2-TGF-beta1 feedback loop between cancer cells and tumor-associated macrophages maintains epithelial-mesenchymal transition and endoplasmic reticulum homeostasis in cholangiocarcinoma.\n",
      "abstract: BACKGROUND: Tumor-associated macrophages (TAMs) play a dual role in tumors. However, the factors which drive the function of TAMs in cholangiocarcinoma remain largely undefined. METHODS: SHH signaling pathway and endoplasmic reticulum stress (ERS) indicators were detected in clinical tissues and cholangiocarcinoma cell lines. TAMs were co-cultured with cholangiocarcinoma cells under conditions of hypoxia/normoxia. Polarized TAMs were counted by flow cytometry, and TGF-beta1 levels in cell supernatants were detected by ELISA. The effects of glioma-associated oncogene GLI2 on TAMs themselves and cholangiocarcinoma cells were examined by conducting interference and overexpression assays. RESULTS: The SHH signaling pathway and ERS were both activated in tumor tissues or tumor cell lines under conditions of hypoxia. In co-culture experiments, the presence of cholangiocarcinoma cells increased the proportion of M2-polarized TAMs and the secretion of TGF-beta1 by TAMs, while knockdown of SHH expression reversed those increases. Overexpression of GLI2 in TAMS or stimulation of TAMS with Hh-Ag1.5 increased their levels of TGF-beta1 expression. Furthermore, under co-culture conditions, interference with GLI2 expression in TAMs reduced the tumor cell migration, invasion, and ER homeostasis induced by Hh-Ag1.5-pretreated TAMs. Under conditions of hypoxia, the presence of cholangiocarcinoma cells promoted the expression of GLI2 and TGF-beta1 in Tams, and in turn, TAMs inhibited the apoptosis and promoted the migration and invasion of cholangiocarcinoma cells. In vivo, an injection of cholangiocarcinoma cells plus TAMs contributed to the growth, EMT, and ER homeostasis of tumor tissue, while an injection of TAMs with GLI2 knockdown had the opposite effects. CONCLUSION: Cholangiocarcinoma cells regulated TAM polarization and TGF-beta1 secretion via a paracrine SHH signaling pathway, and in turn, TAMs promoted the growth, EMT, and ER homeostasis of cholangiocarcinoma cells via TGF-beta1.\n",
      "authors: ['Chen Z', 'Li H', 'Li Z', 'Chen S', 'Huang X', 'Zheng Z', 'Qian X', 'Zhang L', 'Long G', 'Xie J', 'Wang Q', 'Pan W', 'Zhang D']\n",
      "source: Pharmacol Res. 2023 Jan;187:106564. doi: 10.1016/j.phrs.2022.106564. Epub 2022 Nov 21.\n",
      "\n",
      "PMID: 36375472\n",
      "title: lncRNA HHIP-AS1/HHIP modulates osteogenic differentiation of BM-MSCs by regulating Hedgehog signaling pathway.\n",
      "abstract: BACKGROUND: lncRNA, a type of non-coding RNA, plays an important role in the osteogenic differentiation of bone marrow-derived mesenchymal stem cells (BM-MSCs). In this study, lncRNA and mRNA microarrays were performed to study the change of gene expression during osteogenic differentiation of BM-MSCs. We focused on Hedgehog interacting protein (HHIP), because HHIP mRNA and lncRNA HHIP-AS1 were gradually down-regulated on days 0, 7, and 14 during osteogenic differentiation. In addition, the gene coding lncRNA HHIP-AS1 is located on the anti-sense of Hhip gene, implying the potential interaction between lncRNA HHIP-AS1 and HHIP mRNA. METHODS: BM-MSCs with over-expressed or silenced lncRNA HHIP-AS1 were constructed to explore the biological role of HHIP-AS1 in osteogenic differentiation. BM-MSCs were lysed to determine the alkaline phosphatase activity. Fluorescence in situ hybridization and immunofluorescence were performed to analyze HHIP-AS1, HHIP, RUNX2 and osteocalcin. RESULTS: Overexpression of lncRNA HHIP-AS1 increased HHIP expression, which suppressed Hedgehog signaling pathway, as indicated by the reduction of SMO, Gli1 and Gli2. The suppression of Hedgehog signal was associated with the inhibited osteogenesis. HHIP knockdown abolished the suppression of osteogenesis induced by lncRNA HHIP-AS1 overexpression. Through binding to HHIP mRNA, lncRNA HHIP-AS1 recruited ELAVL1 to HHIP mRNA, whereby increasing the mRNA stability and the protein level. CONCLUSIONS: This study revealed that down-regulation of HHIP due to lncRNA HHIP-AS1 reduction promoted the osteogenic differentiation of BM-MSCs though removing the suppression of Hedgehog signal.\n",
      "authors: ['Yin XH', 'Wang XY', 'Liu SC', 'Yan L', 'He BR', 'Hao DJ', 'Yang M', 'Liu ZK']\n",
      "source: Aging (Albany NY). 2022 Nov 14;14(21):8839-8855. doi: 10.18632/aging.204381. Epub 2022 Nov 14.\n",
      "\n",
      "PMID: 36288719\n",
      "title: METTL3 regulates m(6)A methylation of PTCH1 and GLI2 in Sonic hedgehog signaling to promote tumor progression in SHH-medulloblastoma.\n",
      "abstract: SHH subgroup medulloblastoma (SHH-MB) is one of the most common malignant pediatric tumors that arises in the cerebellum. Previously, we showed that RNA m(6)A methylation participates in regulation of cerebellar development. Here we investigate whether dysregulated m(6)A methylation contributes to tumorigenesis of SHH-MB. We show that high expression of m(6)A methyltransferase METTL3 associates with worse survival in the patients with SHH-MB. A large number of hypermethylated transcripts are identified in SHH-MB tumor cells by m(6)A-seq. We find that METTL3 promotes tumor progression via activating Sonic hedgehog signaling. Mechanistically, METTL3 methylates PTCH1 and GLI2 RNAs and further regulates their RNA stability and translation. Importantly, targeting METTL3 by depleting METTL3 expression or treatment with its catalytic inhibitor STM2457 restrains tumor progression. Collectively, this study shows a critical function for METTL3 and m(6)A methylation in SHH-MB, indicative of a potential role of METTL3 as therapeutic target in SHH-MB.\n",
      "authors: ['Zhang ZW', 'Teng X', 'Zhao F', 'Ma C', 'Zhang J', 'Xiao LF', 'Wang Y', 'Chang M', 'Tian Y', 'Li C', 'Zhang Z', 'Song S', 'Tong WM', 'Liu P', 'Niu Y']\n",
      "source: Cell Rep. 2022 Oct 25;41(4):111530. doi: 10.1016/j.celrep.2022.111530.\n",
      "\n",
      "PMID: 36230980\n",
      "title: Hedgehog Signaling as a Therapeutic Target for Airway Remodeling and Inflammation in Allergic Asthma.\n",
      "abstract: Genome-wide association studies (GWAS) have shown that variants of patched homolog 1 (PTCH1) are associated with lung function abnormalities in the general population. It has also been shown that sonic hedgehog (SHH), an important ligand for PTCH1, is upregulated in the airway epithelium of patients with asthma and is suggested to be involved in airway remodeling. The contribution of hedgehog signaling to airway remodeling and inflammation in asthma is poorly described. To determine the biological role of hedgehog signaling-associated genes in asthma, gene silencing, over-expression, and pharmacologic inhibition studies were conducted after stimulating human airway epithelial cells or not with transforming growth factor beta1 (TGFbeta1), an important fibrotic mediator in asthmatic airway remodeling that also interacts with SHH pathway. TGFbeta1 increased hedgehog-signaling-related gene expression including SHH, GLI1 and GLI2. Knockdown of PTCH1 or SMO with siRNA, or use of hedgehog signaling inhibitors, consistently attenuated COL1A1 expression induced by TGFbeta1 stimulation. In contrast, Ptch1 over-expression augmented TGFbeta1-induced an increase in COL1A1 and MMP2 gene expression. We also showed an increase in hedgehog-signaling-related gene expression in primary airway epithelial cells from controls and asthmatics at different stages of cellular differentiation. GANT61, an inhibitor of GLI1/2, attenuated TGFbeta1-induced increase in COL1A1 protein expression in primary airway epithelial cells differentiated in air-liquid interface. Finally, to model airway tissue remodeling in vivo, C57BL/6 wildtype (WT) and Ptch1(+/-) mice were intranasally challenged with house dust mite (HDM) or phosphate-buffered saline (PBS) control. Ptch1(+/-) mice showed reduced sub-epithelial collagen expression and serum inflammatory proteins compared to WT mice in response to HDM challenge. In conclusion, TGFbeta1-induced airway remodeling is partially mediated through the hedgehog signaling pathway via the PTCH1-SMO-GLI axis. The Hedgehog signaling pathway is a promising new potential therapeutic target to alleviate airway tissue remodeling in patients with allergic airways disease.\n",
      "authors: ['Tam A', 'Osei ET', 'Cheung CY', 'Hughes M', 'Yang CX', 'McNagny KM', 'Dorscheid DR', 'Singhera GK', 'Hallstrand TS', 'Warner S', 'Hogg JC', 'Hackett TL', 'Lim CJ', 'Sin DD']\n",
      "source: Cells. 2022 Sep 27;11(19):3016. doi: 10.3390/cells11193016.\n",
      "\n",
      "PMID: 36210532\n",
      "title: DNA methylation profile of lip tissue from congenital nonsyndromic cleft lip and palate patients by whole-genome bisulfite sequencing.\n",
      "abstract: Congenital nonsyndromic cleft lip and palate (NSCLP) is one of the most common malformations worldwide. DNA methylation has been implicated in many diseases. However, its involvement in lip tissue from NSCLP is not well understood. This study aimed to investigate the role of dysregulated DNA methylation in NSCLP. DNA methylation profile was determined in eight injured and five self-normal lip tissue samples from children with NSCLP by whole-genome bisulfite sequencing. A total of 2,711 differentially methylated regions (DMRs), corresponding to 1,231 genes were identified. Given the important role of promoter methylation in regulating gene expression, the promoter DMR-related genes were considered. Bioinformatics analysis demonstrated that some of them showed potential associations with NSCLP. Therefore, the well-known NSCLP susceptibility gene, GLI family zinc finger 2 (GLI2) with an unknown role in its DNA methylation in NSCLP, was selected for further analysis. The promoter hypomethylation and higher mRNA expression level of GLI2 were observed in injured lip tissues by verification in additional samples. Moreover, dual luciferase reporter assay indicated that promoter hypermethylation of GLI2 inhibited its transcription. Overall, this study suggested that abnormal DNA methylation in lip tissue may be correlated with the pathogenesis of congenital NSCLP.\n",
      "authors: ['Zhang B', 'Zhang Y', 'Wu S', 'Ma D', 'Ma J']\n",
      "source: Birth Defects Res. 2022 Oct 9. doi: 10.1002/bdr2.2102.\n",
      "\n",
      "PMID: 36142286\n",
      "title: Cordycepin Inhibits Growth and Metastasis Formation of MDA-MB-231 Xenografts in Nude Mice by Modulating the Hedgehog Pathway.\n",
      "abstract: We previously found that cordycepin inhibits the growth and metastasis formation of MDA-MB-231 cells through the Hedgehog pathway but has not validated this in vivo. In this study, we confirmed cordycepin's anti-triple-negative breast cancer (TNBC) effect in nude mice and documented its mechanism. We found that cordycepin reduced the volume and weight of MDA-MB-231 xenografts and affected the expression of proliferation-, apoptosis-, epithelial-mesenchymal transition-, and matrix metalloproteinase-related proteins without side effects. RNA sequencing screening, pathway enrichment, and the protein network interaction analysis revealed enriched pathways and targets mainly concentrated on the Hedgehog pathway and its core components of SHH and GLI2. This indicates that the Hedgehog pathway plays a central role in the cordycepin-mediated regulation of growth and metastasis formation in TNBC. The database analysis of the Hedgehog pathway markers (SHH, PTCH1, SMO, GLI1, and GLI2) revealed that the Hedgehog pathway is activated in breast cancer tissues, and its high expression is not conducive to a patient's survival. Finally, we verified that cordycepin effectively inhibited the Hedgehog pathway in TNBC through Western blotting and immunohistochemistry. This study found that cordycepin could regulate the growth and metastasis formation of TNBC through the Hedgehog pathway in vivo, which provides new insights for targeting and treating breast cancer.\n",
      "authors: ['Wu W', 'Li X', 'Qi M', 'Hu X', 'Cao F', 'Wu X', 'Fu J']\n",
      "source: Int J Mol Sci. 2022 Sep 8;23(18):10362. doi: 10.3390/ijms231810362.\n",
      "\n",
      "PMID: 35969743\n",
      "title: O-GlcNAcylation promotes cerebellum development and medulloblastoma oncogenesis via SHH signaling.\n",
      "abstract: Sonic hedgehog (Shh) signaling plays a critical role in regulating cerebellum development by maintaining the physiological proliferation of granule neuron precursors (GNPs), and its dysregulation leads to the oncogenesis of medulloblastoma. O-GlcNAcylation (O-GlcNAc) of proteins is an emerging regulator of brain function that maintains normal development and neuronal circuitry. Here, we demonstrate that O-GlcNAc transferase (OGT) in GNPs mediate the cerebellum development, and the progression of the Shh subgroup of medulloblastoma. Specifically, OGT regulates the neurogenesis of GNPs by activating the Shh signaling pathway via O-GlcNAcylation at S355 of GLI family zinc finger 2 (Gli2), which in turn promotes its deacetylation and transcriptional activity via dissociation from p300, a histone acetyltransferases. Inhibition of OGT via genetic ablation or chemical inhibition improves survival in a medulloblastoma mouse model. These data uncover a critical role for O-GlcNAc signaling in cerebellar development, and pinpoint a potential therapeutic target for Shh-associated medulloblastoma.\n",
      "authors: ['Chen L', 'Li Y', 'Song Z', 'Xue S', 'Liu F', 'Chang X', 'Wu Y', 'Duan X', 'Wu H']\n",
      "source: Proc Natl Acad Sci U S A. 2022 Aug 23;119(34):e2202821119. doi: 10.1073/pnas.2202821119. Epub 2022 Aug 15.\n",
      "\n",
      "PMID: 35953110\n",
      "title: Parathyroid hormone-related protein in breast cancer bone metastasis.\n",
      "abstract: Parathyroid hormone-related protein (PTHrP) was discovered as the tumor product causing the humoral hypercalcemia of malignancy. Its structural similarity to the hormone, PTH, with 8 of the first 13 amino acids identical, was sufficient to explain the sharing by PTHrP and PTH of a common receptor, PTH1R, although the remainder of the sequences are unique. PTHrP has important roles in development of several organs, including breast and bone, and functions as a paracrine factor postnatally in these and other tissues. In addition to its hormonal role in cancer, PTHrP is produced by two thirds of primary breast cancers and 90% of bone metastases from breast cancer, leading to the concept that its production in bone by breast cancer cells promotes bone resorption, thus favoring tumor establishment and expansion, and an exit from tumor dormancy in bone through downregulation of leukemia inducing factor receptor (LIFR). Cancer production of PTHrP is increased by bone-derived growth factors, with particular attention paid to TGFbeta, as well as by promoter-driven transcriptional effects, such as the hedgehog signaling factor, GLI2, and microenvironment effects including changes in underlying stiffness of substrates for cells. Although interest has been focused on PTHrP-induced bone resorption in bone metastasis, a mechanistically separate, protective effect against tumor progression has been proposed. Although there is conflicting mouse data, there are clinical studies suggesting that increased production of PTHrP by breast cancers confers upon them a less invasive phenotype, an effect distinct from the bone resorption-stimulating action that favors bone metastasis.\n",
      "authors: ['Johnson RW', 'Rhoades J', 'Martin TJ']\n",
      "source: Vitam Horm. 2022;120:215-230. doi: 10.1016/bs.vh.2022.04.006. Epub 2022 Jul 11.\n",
      "\n",
      "PMID: 35888995\n",
      "title: Effects of Spore-Displayed p75 Protein from Lacticaseibacillus rhamnosus GG on the Transcriptional Response of HT-29 Cells.\n",
      "abstract: A Lacticaseibacillus rhamnosus GG-derived protein, p75, is one of the key molecules exhibiting probiotic activity. However, the molecular mechanism and transcriptional response of p75 in human intestinal epithelial cells are not completely understood. To gain a deeper understanding of its potential probiotic action, this study investigated genome-wide responses of HT-29 cells to stimulation by spore-displayed p75 (CotG-p75) through a transcriptome analysis based on RNA sequencing. Analysis of RNA-seq data showed significant changes of gene expression in HT-29 cells stimulated by CotG-p75 compared to the control. A total of 189 up-regulated and 314 down-regulated genes was found as differentially expressed genes. Gene ontology enrichment analysis revealed that a large number of activated genes was involved in biological processes, such as epithelial cell differentiation, development, and regulation of cell proliferation. A gene-gene interaction network analysis showed that several DEGs, including AREG, EREG, HBEGF, EPGN, FASLG, GLI2, CDKN1A, FOSL1, MYC, SERPINE1, TNFSF10, BCL6, FLG, IVL, SPRR1A, SPRR1B, SPRR3, and MUC5AC, might play a critical role in these biological processes. RNA-seq results for selected genes were verified by reverse transcription-quantitative polymerase chain reaction. Overall, these results provide extensive knowledge about the transcriptional responses of HT-29 cells to stimulation by CotG-p75. This study showed that CotG-p75 can contribute to cell survival and epithelial development in human intestinal epithelial cells.\n",
      "authors: ['Kang SJ', 'Kim MJ', 'Son DY', 'Kang SS', 'Hong KW']\n",
      "source: Microorganisms. 2022 Jun 23;10(7):1276. doi: 10.3390/microorganisms10071276.\n",
      "\n",
      "PMID: 35867772\n",
      "title: Combinatorial Gli activity directs immune infiltration and tumor growth in pancreatic cancer.\n",
      "abstract: Proper Hedgehog (HH) signaling is essential for embryonic development, while aberrant HH signaling drives pediatric and adult cancers. HH signaling is frequently dysregulated in pancreatic cancer, yet its role remains controversial, with both tumor-promoting and tumor-restraining functions reported. Notably, the GLI family of HH transcription factors (GLI1, GLI2, GLI3), remain largely unexplored in pancreatic cancer. We therefore investigated the individual and combined contributions of GLI1-3 to pancreatic cancer progression. At pre-cancerous stages, fibroblast-specific Gli2/Gli3 deletion decreases immunosuppressive macrophage infiltration and promotes T cell infiltration. Strikingly, combined loss of Gli1/Gli2/Gli3 promotes macrophage infiltration, indicating that subtle changes in Gli expression differentially regulate immune infiltration. In invasive tumors, Gli2/Gli3 KO fibroblasts exclude immunosuppressive myeloid cells and suppress tumor growth by recruiting natural killer cells. Finally, we demonstrate that fibroblasts directly regulate macrophage and T cell migration through the expression of Gli-dependent cytokines. Thus, the coordinated activity of GLI1-3 directs the fibroinflammatory response throughout pancreatic cancer progression.\n",
      "authors: ['Scales MK', 'Velez-Delgado A', 'Steele NG', 'Schrader HE', 'Stabnick AM', 'Yan W', 'Mercado Soto NM', 'Nwosu ZC', 'Johnson C', 'Zhang Y', 'Salas-Escabillas DJ', 'Menjivar RE', 'Maurer HC', 'Crawford HC', 'Bednar F', 'Olive KP', 'Pasca di Magliano M', 'Allen BL']\n",
      "source: PLoS Genet. 2022 Jul 22;18(7):e1010315. doi: 10.1371/journal.pgen.1010315. eCollection 2022 Jul.\n",
      "\n",
      "PMID: 35692978\n",
      "title: The Role of GLI in the Regulation of Hepatic Epithelial-Mesenchymal Transition in Biliary Atresia.\n",
      "abstract: OBJECTIVE: To study the regulatory role of GLI1/GLI2, a nuclear transcription factor of the Sonic hedgehog (Shh) signaling pathway, in epithelial-mesenchymal transition (EMT) related to hepatic fibrosis in patients with biliary atresia (BA). METHODS: The messenger RNA (mRNA) and protein expression levels of GLI1/GLI2, Snail/Slug, and other Shh- and EMT-related cytokines were tested in the liver tissues of BA patients and animals. Then, GLI1/GLI2 was silenced and overexpressed in mouse intrahepatic bile duct epithelial cells (mIBECs) and BA animals to investigate changes in the mRNA and protein expression of EMT key factors and liver fibrosis indicators. After silencing and overexpression of GLI1/GLI2, immunofluorescence was used to detect the expression of cytokeratin-19 (CK19) and alpha-smooth muscle actin (alpha-SMA) in mIBECs, and hematoxylin and eosin (HE) staining and Masson staining were used to observe the degree of liver fibrosis in the BA animals. RESULTS: Compared with the control, the mRNA and protein expression levels of GLI2, Snail, vimentin, and alpha-SMA were significantly increased and those of E-cadherin were significantly decreased in liver tissue from BA patients and animals. Overexpression of GLI2 increased the mRNA and protein expression levels of Snail, vimentin, and alpha-SMA and that of E-cadherin was significantly decreased in mIBECs and BA animals. After GLI2 silencing, the opposite pattern was observed. Immunofluorescence detection showed enhanced expression of the bile duct epithelial cell marker CK19 in mIBECs after GLI2 silencing and enhanced expression of the mesenchymal cell marker alpha-SMA after GLI2 overexpression. HE and Masson staining suggested that the GLI2-overexpressing group had a significantly higher degree of fibrosis. CONCLUSION: The Shh signaling pathway plays an important role in fibrogenesis in BA. GLI2 can significantly regulate EMT in mIBECs and livers of BA mice.\n",
      "authors: ['Siyu P', 'Junxiang W', 'Qi W', 'Yimao Z', 'Shuguang J']\n",
      "source: Front Pediatr. 2022 May 26;10:861826. doi: 10.3389/fped.2022.861826. eCollection 2022.\n",
      "\n",
      "PMID: 35584663\n",
      "title: The ciliary gene INPP5E confers dorsal telencephalic identity to human cortical organoids by negatively regulating Sonic hedgehog signaling.\n",
      "abstract: Defects in primary cilia, cellular antennas that control multiple intracellular signaling pathways, underlie several neurodevelopmental disorders, but it remains unknown how cilia control essential steps in human brain formation. Here, we show that cilia are present on the apical surface of radial glial cells in human fetal forebrain. Interfering with cilia signaling in human organoids by mutating the INPP5E gene leads to the formation of ventral telencephalic cell types instead of cortical progenitors and neurons. INPP5E mutant organoids also show increased Sonic hedgehog (SHH) signaling, and cyclopamine treatment partially rescues this ventralization. In addition, ciliary expression of SMO, GLI2, GPR161, and several intraflagellar transport (IFT) proteins is increased. Overall, these findings establish the importance of primary cilia for dorsal and ventral patterning in human corticogenesis, indicate a tissue-specific role of INPP5E as a negative regulator of SHH signaling, and have implications for the emerging roles of cilia in the pathogenesis of neurodevelopmental disorders.\n",
      "authors: ['Schembs L', 'Willems A', 'Hasenpusch-Theil K', 'Cooper JD', 'Whiting K', 'Burr K', 'Bostrand SMK', 'Selvaraj BT', 'Chandran S', 'Theil T']\n",
      "source: Cell Rep. 2022 May 17;39(7):110811. doi: 10.1016/j.celrep.2022.110811.\n",
      "\n",
      "PMID: 35574464\n",
      "title: Ptch2 is a Potential Regulator of Mesenchymal Stem Cells.\n",
      "abstract: Ptch receptors 1 and 2 mediate Hedgehog signaling pivotal for organ development and homeostasis. In contrast to embryonic lethal Ptch1 (-/-) phenotype, Ptch2 (-/-) mice display no effect on gross phenotype. In this brief report, we provide evidence of changes in the putative incisor mesenchymal stem cell (MSC) niches that contribute to accelerated incisor growth, as well as intriguing changes in the bones and skin which suggest a role for Ptch2 in the regulation of MSCs and their regenerative potential. We employed histological, immunostaining, and computed tomography (microCT) analyses to analyze morphological differences between Ptch2 (-/-) and wild-type incisors, long bones, and skins. In vitro CFU and differentiation assays were used to demonstrate the MSC content and differentiation potential of Ptch2 (-/-) bone marrow stromal cells. Wound healing assay was performed in vivo and in vitro on 8-week-old mice to assess the effect of Ptch2 on the wound closure. Loss of Ptch2 causes increases in the number of putative MSCs in the continuously growing incisor, associated with increased vascularization observed in the tooth mesenchyme and the neurovascular bundle. Increased length and volume of Ptch2 (-/-) bones is linked with the increased number and augmented in vitro differentiation potential of MSCs in the bone marrow. Dynamic changes in the Ptch2 (-/-) skin thickness relate to changes in the mesenchymal compartment and impact the wound closure potential. The effects of Ptch2 abrogation on the postnatal MSCs suggest a crucial role for Ptch2 in Hedgehog signaling regulation of the organ regenerative potential.\n",
      "authors: ['Juuri E', 'Tikka P', 'Domanskyi A', 'Corfe I', 'Morita W', 'Mckinnon PJ', 'Jandova N', 'Balic A']\n",
      "source: Front Physiol. 2022 Apr 28;13:877565. doi: 10.3389/fphys.2022.877565. eCollection 2022.\n",
      "\n",
      "PMID: 35505292\n",
      "title: Associations of keratinocyte cancers with snp variants in the sonic hedgehog pathway.\n",
      "abstract: BACKGROUND: Sonic Hedgehog (SHH) pathway dysregulation is implicated in basal cell carcinoma (BCC) development. To evaluate the possible wider role of SHH gene variants in skin carcinogenesis, we assessed associations of genes in the SHH pathway with lifetime development of any keratinocyte cancer (KC), and with developing either BCCs or squamous cell carcinomas (SCCs) exclusively, in a 25-year prospective, population-based study of 1,621 Australians. METHODS: We genotyped 795 unrelated adults with available blood samples: 311 cases with any KC (186 developing BCCs-only, 55 SCCs-only, 70 BCCs and SCCs) and 484 controls. We compared allele frequencies of 158 independent SNPs across 43 SHH genes between cases and controls, and performed a gene-based analysis. RESULTS: We found associations between SNP rs4848627 (GLI2) (related to DNA synthesis in keratinocytes) and development of any KC (OR = 1.53; 95% CI = 1.06-2.13, P < 0.01) and SCCs exclusively (OR = 2.12; 95%CI = 1.39-3.23, P < 0.01). SNP rs3217882 located in CCND2 was associated with exclusive BCC development (OR = 1.43, CI = 1.12-1.82, P < 0.01). The gene-based analysis suggested an association of PRKACG (protein kinase cAMP-activated catalytic subunit gamma) with any KC (P = 0.013). CONCLUSION: We conclude that variants located in genes in the SHH pathway may are involved in SCC as well as BCC development.\n",
      "authors: ['Rodriguez-Acevedo AJ', 'Antonsson A', 'Liyanage UE', 'Hughes MC', 'Gordon S', 'van der Pols J', 'Green AC']\n",
      "source: BMC Cancer. 2022 May 3;22(1):490. doi: 10.1186/s12885-022-09565-6.\n",
      "\n",
      "PMID: 35251038\n",
      "title: A Critical Function for the Transcription Factors GLI1 and GLI2 in the Proliferation and Survival of Human Mast Cells.\n",
      "abstract: Mast cell hyperactivity and accumulation in tissues are associated with allergy and other mast cell-related disorders. However, the molecular pathways regulating mast cell survival in homeostasis and disease are not completely understood. As glioma-associated oncogene (GLI) proteins are involved in both tissue homeostasis and in the hematopoietic system by regulating cell fate decisions, we sought to investigate the role for GLI proteins in the control of proliferation and survival of human mast cells. GLI1 transcripts were present in primary human mast cells and mast cell lines harboring or not activating mutations in the tyrosine kinase receptor KIT (HMC-1.1 and HMC-1.2, and LAD2 cells, respectively), while GLI2 transcripts were only present in HMC-1.1 and HMC-1.2 cells, suggesting a role for oncogenic KIT signaling in the regulation of GLI2. Reduction in GLI activity by small molecule inhibitors, or by shRNA-mediated knockdown of GLI1 or GLI2, led to increases in apoptotic cell death in both cultured human and murine mast cells, and reduced the number of peritoneal mast cells in mice. Although GLI proteins are typically activated via the hedgehog pathway, steady-state activation of GLI in mast cells occurred primarily via non-canonical pathways. Apoptosis induced by GLI silencing was associated with a downregulation in the expression of KIT and of genes that influence p53 stability and function including USP48, which promotes p53 degradation; and iASPP, which inhibits p53-induced transcription, thus leading to the induction of p53-regulated apoptotic genes. Furthermore, we found that GLI silencing inhibited the proliferation of neoplastic mast cell lines, an effect that was more pronounced in rapidly growing cells. Our findings support the conclusion that GLI1/2 transcription factors are critical regulators of mast cell survival and that their inhibition leads to a significant reduction in the number of mast cells in vitro and in vivo, even in cells with constitutively active KIT variants. This knowledge can potentially be applicable to reducing mast cell burden in mast cell-related diseases.\n",
      "authors: ['Falduto GH', 'Pfeiffer A', 'Zhang Q', 'Yin Y', 'Metcalfe DD', 'Olivera A']\n",
      "source: Front Immunol. 2022 Feb 16;13:841045. doi: 10.3389/fimmu.2022.841045. eCollection 2022.\n",
      "\n",
      "PMID: 35226587\n",
      "title: ULK3-dependent activation of GLI1 promotes DNMT3A expression upon autophagy induction.\n",
      "abstract: Macroautophagy/autophagy is a tightly regulated catabolic process, which contributes at baseline level to cellular homeostasis, and upon its stimulation to the adaptive cellular response to intra- and extracellular stress stimuli. Decrease of autophagy activity is occurring upon aging and thought to contribute to age-related-diseases. Recently, we uncovered, upon autophagy induction, the role of de novo DNMT3A (DNA methyltransferase 3 alpha)-mediated DNA methylation on expression of the MAP1LC3 (microtubule associated protein 1 light chain 3) proteins, core components of the autophagy pathway, which resulted in reduced baseline autophagy activity. Here, we report that serine/threonine kinase ULK3 (unc-51 like kinase 3)-dependent activation of GLI1 (GLI family zinc finger 1) contributes to the transcriptional upregulation of DNMT3A gene expression upon autophagy induction, thereby bringing additional understanding of the long-term effect of autophagy induction and a possible mechanism for its decline upon aging, pathological conditions, or in response to treatment interventions.Abbreviations: CBZ: carbamazepine; ChIP: chromatin immunoprecipitation; Clon: clonidine; DNMT3A: DNA methyltransferase 3 alpha; GLI1: GLI family zinc finger 1; GLI2: GLI family zinc finger 2; MAP1LC3: microtubule associated protein 1 light chain 3; MTOR: mechanistic target of rapamycin kinase; PLA: proximity ligation assay; RT-qPCR: quantitative reverse transcription PCR; shRNA: small hairpin RNA; siRNA: small interfering RNA; Treh: trehalose; ULK3: unc-51 like kinase 3.\n",
      "authors: ['Gonzalez-Rodriguez P', 'Cheray M', 'Keane L', 'Engskog-Vlachos P', 'Joseph B']\n",
      "source: Autophagy. 2022 Dec;18(12):2769-2780. doi: 10.1080/15548627.2022.2039993. Epub 2022 Feb 28.\n",
      "\n",
      "PMID: 35221935\n",
      "title: Gli2-Mediated Shh Signaling Is Required for Thalamocortical Projection Guidance.\n",
      "abstract: The thalamocortical projections are part of the most important higher level processing connections in the vertebrates and follow a highly ordered pathway from their origin in the thalamus to the cerebral cortex. Their functional complexities are not only due to an extremely elaborate axon guidance process but also due to activity-dependent mechanisms. Gli2 is an intermediary transcription factor in the Sonic hedgehog (Shh) pathway. During neural early development, Shh has an important role in dorsoventral patterning, diencephalic anteroposterior patterning, and many later developmental processes, such as axon guidance and cell migration. Using a Gli2 knockout mouse line, we have studied the role of Shh signaling mediated by Gli2 in the development of the thalamocortical projections during embryonic development. In wild-type brains, we have described the normal trajectory of the thalamocortical axons into the context of the prosomeric model. Then, we have compared it with the altered thalamocortical axons course in Gli2 homozygous embryos. The thalamocortical axons followed different trajectories and were misdirected to other territories probably due to alterations in the Robo/Slit signaling mechanism. In conclusion, the alteration of Gli2-mediated Shh signaling produces an erroneous specification of several territories related with the thalamocortical axons. This is translated into a huge modification in the pathfinding signaling mechanisms needed for the correct wiring of the thalamocortical axons.\n",
      "authors: ['Callejas-Marin A', 'Moreno-Bravo JA', 'Company V', 'Madrigal MP', 'Almagro-Garcia F', 'Martinez S', 'Puelles E']\n",
      "source: Front Neuroanat. 2022 Feb 10;16:830758. doi: 10.3389/fnana.2022.830758. eCollection 2022.\n",
      "\n",
      "PMID: 35220829\n",
      "title: Gnas Loss Causes Chondrocyte Fate Conversion in Cranial Suture Formation.\n",
      "abstract: Calvaria development is distinct from limb formation. Craniosynostosis is a skull deformity characterized by premature cranial suture fusion due to the loss of the GNAS gene and, consequently, its encoded protein Galpha(s). This birth defect requires surgery, with potential lethal consequences. So far, hardly any early-stage nonsurgical interventions for GNAS loss-related craniosynostosis are available. Here, we investigated the role of the Gnas gene in mice in guarding the distinctiveness of intramembranous ossification and how loss of Gnas triggered endochondral-like ossification within the cranial sutures. Single-cell RNA sequencing (scRNA-seq) of normal neonatal mice cranial suture chondrocytes showed a Hedgehog (Hh) inactivation pattern, which was associated with Galpha(s) signaling activation. Loss of Gnas evoked chondrocyte-to-osteoblast fate conversion and resulted in cartilage heterotopic ossification (HO) within cranial sutures and fontanels of the mouse model, leading to a skull deformity resembling craniosynostosis in patients with loss of GNAS. Activation of ectopic Hh signaling within cranial chondrocytes stimulated the conversion of cell identity through a hypertrophy-like stage, which shared features of endochondral ossification in vivo. Reduction of Gli transcription activity by crossing with a loss-of-function Gli2 allele or injecting GLI1/2 antagonist hindered the progression of cartilage HO in neonatal stage mice. Our study uncovered the role of Galpha(s) in maintaining cranial chondrocyte identity during neonatal calvaria development in mice and how reduction of Hh signaling could be a nonsurgical intervention to reduce skull deformity in craniosynostosis due to loss of GNAS.\n",
      "authors: ['Xu R', 'Liu Y', 'Zhou Y', 'Lin W', 'Yuan Q', 'Zhou X', 'Yang Y']\n",
      "source: J Dent Res. 2022 Jul;101(8):931-941. doi: 10.1177/00220345221075215. Epub 2022 Feb 26.\n",
      "\n",
      "PMID: 35159339\n",
      "title: Induction of Pro-Fibrotic CLIC4 in Dermal Fibroblasts by TGF-beta/Wnt3a Is Mediated by GLI2 Upregulation.\n",
      "abstract: Chloride intracellular channel 4 (CLIC4) is a recently discovered driver of fibroblast activation in Scleroderma (SSc) and cancer-associated fibroblasts (CAF). CLIC4 expression and activity are regulated by TGF-beta signalling through the SMAD3 transcription factor. In view of the aberrant activation of canonical Wnt-3a and Hedgehog (Hh) signalling in fibrosis, we investigated their role in CLIC4 upregulation. Here, we show that TGF-beta/SMAD3 co-operates with Wnt3a/beta-catenin and Smoothened/GLI signalling to drive CLIC4 expression in normal dermal fibroblasts, and that the inhibition of beta-catenin and GLI expression or activity abolishes TGF-beta/SMAD3-dependent CLIC4 induction. We further show that the expression of the pro-fibrotic marker alpha-smooth muscle actin strongly correlates with CLIC4 expression in dermal fibroblasts. Further investigations revealed that the inhibition of CLIC4 reverses morphogen-dependent fibroblast activation. Our data highlights that CLIC4 is a common downstream target of TGF-beta, Hh, and Wnt-3a through signalling crosstalk and we propose a potential therapeutic avenue using CLIC4 inhibitors.\n",
      "authors: ['Wasson CW', 'Caballero-Ruiz B', 'Gillespie J', 'Derrett-Smith E', 'Mankouri J', 'Denton CP', 'Canettieri G', 'Riobo-Del Galdo NA', 'Del Galdo F']\n",
      "source: Cells. 2022 Feb 3;11(3):530. doi: 10.3390/cells11030530.\n",
      "\n",
      "PMID: 35059317\n",
      "title: Hedgehog-Gli2 Signaling Promotes Chemoresistance in Ovarian Cancer Cells by Regulating MDR1.\n",
      "abstract: BACKGROUND: Cisplatin (DDP) resistance remains a key challenge in improving the clinical outcome of patients with ovarian cancer (OC). Gli2 overexpression can lead to DDP resistance in OC cells, but the specific underlying regulatory mechanism remains unclear. The membrane transporter encoding gene MDR1 positively regulates chemotherapy resistance in various cancer types. We evaluated MDR1 as a potential Gli2 downstream target and the contribution of the Gli2/MDR1 axis in promoting DDP resistance in OC cells. METHODS: To generate drug-resistant SKOV3/DDP cells, SKOV3 cells were grown for six months under continuous induction wherein the DDP concentration was steadily increased. Gli2 expression in OC cells with varying DDP sensitivities was detected using western blot. Cell counting kit-8 assays were used to assess the DDP sensitivity of SKOV3, SKOV3/DDP, A2780, and A2780/DDP cells and reversal of DDP resistance in SKOV3/DDP and A2780/DDP cells. Cell proliferation was analyzed using 5-ethynyl-2'-deoxyuridine (EdU) incorporation assays. The transcriptional regulation of MDR1 by Gli2 was determined using luciferase reporter assays. Finally, xenograft OC tumors were generated in nude mice, which were then treated with intraperitoneal DDP or phosphate-buffered saline (PBS) injections to investigate if Gli2 affected DDP resistance in OC in vivo. RESULTS: DDP-resistant SKOV3/DDP and A2780/DDP cells showed higher expression of Gli2 and MDR1 as compared with that in DDP-sensitive OC cells. Gli2 knockdown in SKOV3/DDP cells significantly reduced MDR1 expression, whereas it increased DNA damage, thereby sensitizing OC cells to DDP. Similar results were obtained after targeting Gli2 expression with the Gli-antagonist 61 inhibitor (GANT61) in SKOV3/DDP and A2780/DDP cells. In cells stably overexpressing Gli2, treatment with gradient concentrations of verapamil, an MDR1 inhibitor, significantly inhibited MDR1 expression. Our findings indicate that downregulation of MDR1 expression may reverse OC cell resistance to DDP. Moreover, dual-luciferase reporter gene assays confirmed that MDR1 is a direct downstream target of Gli2, with Gli2 positively regulating MDR1 expression. Finally, subcutaneous xenotransplantation in nude mice demonstrated that Gli2 plays a key role in regulating OC drug resistance. CONCLUSIONS: We identified a mechanism by which Hedgehog-Gli signaling regulates OC chemoresistance by modulating MDR1 expression. Hence, Gli2 and MDR1 are potential biomarkers and therapeutic targets in patients with chemoresistant OC.\n",
      "authors: ['Wang Q', 'Wei X', 'Hu L', 'Zhuang L', 'Zhang H', 'Chen Q']\n",
      "source: Front Oncol. 2022 Jan 4;11:794959. doi: 10.3389/fonc.2021.794959. eCollection 2021.\n",
      "\n",
      "PMID: 35053334\n",
      "title: GLI3 Is Required for OLIG2+ Progeny Production in Adult Dorsal Neural Stem Cells.\n",
      "abstract: The ventricular-subventricular zone (V-SVZ) is a postnatal germinal niche. It holds a large population of neural stem cells (NSCs) that generate neurons and oligodendrocytes for the olfactory bulb and (primarily) the corpus callosum, respectively. These NSCs are heterogeneous and generate different types of neurons depending on their location. Positional identity among NSCs is thought to be controlled in part by intrinsic pathways. However, extrinsic cell signaling through the secreted ligand Sonic hedgehog (Shh) is essential for neurogenesis in both the dorsal and ventral V-SVZ. Here we used a genetic approach to investigate the role of the transcription factors GLI2 and GLI3 in the proliferation and cell fate of dorsal and ventral V-SVZ NSCs. We find that while GLI3 is expressed in stem cell cultures from both dorsal and ventral V-SVZ, the repressor form of GLI3 is more abundant in dorsal V-SVZ. Despite this high dorsal expression and the requirement for other Shh pathway members, GLI3 loss affects the generation of ventrally-, but not dorsally-derived olfactory interneurons in vivo and does not affect trilineage differentiation in vitro. However, loss of GLI3 in the adult dorsal V-SVZ in vivo results in decreased numbers of OLIG2-expressing progeny, indicating a role in gliogenesis.\n",
      "authors: ['Embalabala RJ', 'Brockman AA', 'Jurewicz AR', 'Kong JA', 'Ryan K', 'Guinto CD', 'Alvarez-Buylla A', 'Chiang C', 'Ihrie RA']\n",
      "source: Cells. 2022 Jan 10;11(2):218. doi: 10.3390/cells11020218.\n",
      "\n",
      "PMID: 35048440\n",
      "title: Unique lingual expression of the Hedgehog pathway antagonist Hedgehog-interacting protein in filiform papillae during homeostasis and ectopic expression in fungiform papillae during Hedgehog signaling inhibition.\n",
      "abstract: BACKGROUND: Hedgehog (HH) signaling is essential for homeostasis in gustatory fungiform papillae (FP) and taste buds. However, activities of HH antagonists in these tissues remain unexplored. We investigated a potential role for HH-interacting protein (HHIP), an endogenous pathway antagonist, in regulating HH signaling during taste organ homeostasis. We found a restricted pattern of Hhip-expressing cells in the anterior epithelium of each nongustatory filiform papilla (FILIF) only. To test for roles in antagonism of HH signaling, we investigated HHIP after pathway inhibition with SMO inhibition via sonidegib and Smo deletion, Gli2 deletion/suppression, or with chorda tympani/lingual nerve cut. RESULTS: In all approaches, the HHIP expression pattern was retained in FILIF suggesting HH-independent regulation of HHIP. Remarkably, after pathway inhibition, HHIP expression was detected also in the conical, FILIF-like atypical FP. We found a close association of de novo expression of HHIP in atypical FP with loss of Gli1+, HH-responding cells. Further, we report that PTCH1 is another potential HH antagonist in FILIF that co-localizes with HHIP. CONCLUSIONS: After HH pathway inhibition the ectopic expression of HHIP correlates with a FILIF-like morphology in atypical FP and we propose that localized expression of the HH antagonist HHIP regulates pathway inhibition to maintain FILIF during tongue homeostasis.\n",
      "authors: ['Kumari A', 'Li L', 'Ermilov AN', 'Franks NE', 'Dlugosz AA', 'Allen BL', 'Mistretta CM']\n",
      "source: Dev Dyn. 2022 Jul;251(7):1175-1195. doi: 10.1002/dvdy.456. Epub 2022 Feb 18.\n",
      "\n",
      "PMID: 34984190\n",
      "title: Role of BCAR4 in prostate cancer cell autophagy.\n",
      "abstract: BACKGROUND: Increased autophagy of prostate cancer (PC) cells contributes to their resistance to chemotherapy. Recently, we reported that a long non-coding RNA (lncRNA)-breast-cancer anti-estrogen resistance 4 (BCAR4)-is highly expressed in PC and contributes to castration resistance through activation of GLI2 signaling. However, the role of BCAR4 in the regulation of PC cell autophagy is unknown and is the subject of the current study. METHODS: BCAR4 and Beclin-1 levels and the alteration in autophagy pathway genes were assessed in PC using a public database and in our own clinical specimens. The correlation between BCAR4 and Beclin-1 levels in PC and PC cell lines was determined and their regulatory relationship was assessed by overexpression and knockout assay. The final effect on autophagy was measured by microtubule-associated protein 1A/1B-light chain 3 (LC3) levels. The mechanism that underlies the control of Beclin-1 by BCAR4 was analyzed by cancer database and gain-of-function and loss-of-function approaches. RESULTS: BCAR4 and Beclin-1 were both upregulated in PC and were positively correlated. BCAR4 directly activated Beclin-1 at transcriptional level, which subsequently increased the ratio of LC3 II to LC3I to augment PC cell autophagy. Beclin-1 did not control levels of BCAR4. Mechanically, BCAR4 and Beclin-1 shared several targeting microRNAs, among which miR-15 and miR-146 appeared to be the mediators of the effects of BACR4 on Beclin-1. CONCLUSIONS: BCAR4 may enhance PC cell autophagy through altering miRNA-regulated Beclin-1 expression in PC.\n",
      "authors: ['Cai Z', 'Wu Y', 'Ju G', 'Wang G', 'Liu B']\n",
      "source: Transl Androl Urol. 2021 Nov;10(11):4253-4261. doi: 10.21037/tau-21-929.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "database = \"PubMed\"\n",
    "word = 'Role of GLI2'\n",
    "res= int('30')\n",
    "email= \"rodrigoce9@gmail.com\"\n",
    "\n",
    "Entrez.email= email\n",
    "handle=Entrez.esearch(db = database, term=word, retmax= res)\n",
    "record=Entrez.read(handle)\n",
    "handle.close()\n",
    "idlist= record['IdList']\n",
    "\n",
    "handle = Entrez.efetch(db=database, id=idlist, rettype=\"medline\", retmode=\"text\")\n",
    "records = Medline.parse(handle)\n",
    "\n",
    "\n",
    "for record in records:\n",
    "    #print(record) \n",
    "    print(\"PMID:\",record.get(\"PMID\",\"-\"))\n",
    "    print(\"title:\", record.get(\"TI\", \"-\"))\n",
    "    print('abstract:', record.get('AB', '-'))\n",
    "    print(\"authors:\", record.get(\"AU\", \"-\"))\n",
    "    print(\"source:\", record.get(\"SO\", \"-\"))\n",
    "    print(\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "610b23f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "title: Icariin Treatment Rescues Diabetes Induced Bone Loss via Scavenging ROS and Activating Primary Cilia/Gli2/Osteocalcin Signaling Pathway.\n",
      "authors: ['Liu J', 'Cheng Q', 'Wu X', 'Zhu H', 'Deng X', 'Wang M', 'Yang S', 'Xu J', 'Chen Q', 'Li M', 'Liu X', 'Wang C']\n",
      "source: Cells. 2022 Dec 16;11(24):4091. doi: 10.3390/cells11244091.\n",
      "\n",
      "title: FBW7 couples structural integrity with functional output of primary cilia.\n",
      "authors: ['Petsouki E', 'Gerakopoulos V', 'Szeto N', 'Chang W', 'Humphrey MB', 'Tsiokas L']\n",
      "source: Commun Biol. 2021 Sep 13;4(1):1066. doi: 10.1038/s42003-021-02504-4.\n",
      "\n",
      "title: Genome-wide association analysis of metabolic syndrome quantitative traits in the GENNID multiethnic family study.\n",
      "authors: ['Wan JY', 'Goodman DL', 'Willems EL', 'Freedland AR', 'Norden-Krichmar TM', 'Santorico SA', 'Edwards KL']\n",
      "source: Diabetol Metab Syndr. 2021 Jun 1;13(1):59. doi: 10.1186/s13098-021-00670-3.\n",
      "\n",
      "title: Efficacy of Metformin and Chemotherapeutic Agents on the Inhibition of Colony Formation and Shh/Gli1 Pathway: Metformin/Docetaxel Versus Metformin/5-Fluorouracil.\n",
      "authors: ['Fatehi-Agdam M', 'Vatankhah MA', 'Panahizadeh R', 'Jeddi F', 'Najafzadeh N']\n",
      "source: Drug Res (Stuttg). 2021 Jan;71(1):17-25. doi: 10.1055/a-1248-9008. Epub 2020 Sep 28.\n",
      "\n",
      "title: Sufu- and Spop-mediated downregulation of Hedgehog signaling promotes beta cell differentiation through organ-specific niche signals.\n",
      "authors: ['Yung T', 'Poon F', 'Liang M', 'Coquenlorge S', 'McGaugh EC', 'Hui CC', 'Wilson MD', 'Nostro MC', 'Kim TH']\n",
      "source: Nat Commun. 2019 Oct 11;10(1):4647. doi: 10.1038/s41467-019-12624-5.\n",
      "\n",
      "title: Direct repression of anoctamin 1 (ANO1) gene transcription by Gli proteins.\n",
      "authors: ['Mazzone A', 'Gibbons SJ', 'Eisenman ST', 'Strege PR', 'Zheng T', \"D'Amato M\", 'Ordog T', 'Fernandez-Zapico ME', 'Farrugia G']\n",
      "source: FASEB J. 2019 May;33(5):6632-6642. doi: 10.1096/fj.201802373R. Epub 2019 Feb 25.\n",
      "\n",
      "title: Congenital Hypopituitarism: Various Genes, Various Phenotypes.\n",
      "authors: ['Xatzipsalti M', 'Voutetakis A', 'Stamoyannou L', 'Chrousos GP', 'Kanaka-Gantenbein C']\n",
      "source: Horm Metab Res. 2019 Feb;51(2):81-90. doi: 10.1055/a-0822-3637. Epub 2019 Feb 13.\n",
      "\n",
      "title: O-GlcNAcylation of GLI transcription factors in hyperglycemic conditions augments Hedgehog activity.\n",
      "authors: ['Das S', 'Bailey SK', 'Metge BJ', 'Hanna A', 'Hinshaw DC', 'Mota M', 'Forero-Torres A', 'Chatham JC', 'Samant RS', 'Shevde LA']\n",
      "source: Lab Invest. 2019 Feb;99(2):260-270. doi: 10.1038/s41374-018-0122-8. Epub 2018 Nov 12.\n",
      "\n",
      "title: Screening a large pediatric cohort with GH deficiency for mutations in genes regulating pituitary development and GH secretion: Frequencies, phenotypes and growth outcomes.\n",
      "authors: ['Blum WF', 'Klammt J', 'Amselem S', 'Pfaffle HM', 'Legendre M', 'Sobrier ML', 'Luton MP', 'Child CJ', 'Jones C', 'Zimmermann AG', 'Quigley CA', 'Cutler GB Jr', 'Deal CL', 'Lebl J', 'Rosenfeld RG', 'Parks JS', 'Pfaffle RW']\n",
      "source: EBioMedicine. 2018 Oct;36:390-400. doi: 10.1016/j.ebiom.2018.09.026. Epub 2018 Sep 25.\n",
      "\n",
      "title: Congenital Hyperinsulinism and Hypopituitarism Attributable to a Mutation in FOXA2.\n",
      "authors: ['Vajravelu ME', 'Chai J', 'Krock B', 'Baker S', 'Langdon D', 'Alter C', 'De Leon DD']\n",
      "source: J Clin Endocrinol Metab. 2018 Mar 1;103(3):1042-1047. doi: 10.1210/jc.2017-02157.\n",
      "\n",
      "title: Hedgehog signaling via Gli2 prevents obesity induced by high-fat diet in adult mice.\n",
      "authors: ['Shi Y', 'Long F']\n",
      "source: Elife. 2017 Dec 5;6:e31649. doi: 10.7554/eLife.31649.\n",
      "\n",
      "title: Keratinocyte Sonic Hedgehog Upregulation Drives the Development of Giant Congenital Nevi via Paracrine Endothelin-1 Secretion.\n",
      "authors: ['Chitsazan A', 'Ferguson B', 'Villani R', 'Handoko HY', 'Mukhopadhyay P', 'Gabrielli B', 'Mooi WJ', 'Soyer HP', 'Lambie D', 'Khosrotehrani K', 'Morahan G', 'Walker GJ']\n",
      "source: J Invest Dermatol. 2018 Apr;138(4):893-902. doi: 10.1016/j.jid.2017.10.032. Epub 2017 Nov 11.\n",
      "\n",
      "title: Metformin suppresses the expression of Sonic hedgehog in gastric cancer cells.\n",
      "authors: ['Song Z', 'Wei B', 'Lu C', 'Huang X', 'Li P', 'Chen L']\n",
      "source: Mol Med Rep. 2017 Apr;15(4):1909-1915. doi: 10.3892/mmr.2017.6205. Epub 2017 Feb 15.\n",
      "\n",
      "title: Sonic hedgehog improves ischemia-induced neovascularization by enhancing endothelial progenitor cell function in type 1 diabetes.\n",
      "authors: ['Qin Y', 'He YH', 'Hou N', 'Zhang GS', 'Cai Y', 'Zhang GP', 'Xiao Q', 'He LS', 'Li SJ', 'Yi Q', 'Luo JD']\n",
      "source: Mol Cell Endocrinol. 2016 Mar 5;423:30-9. doi: 10.1016/j.mce.2016.01.005. Epub 2016 Jan 7.\n",
      "\n",
      "title: Putative regulatory factors associated with intramuscular fat content.\n",
      "authors: ['Cesar AS', 'Regitano LC', 'Koltes JE', 'Fritz-Waters ER', 'Lanna DP', 'Gasparin G', 'Mourao GB', 'Oliveira PS', 'Reecy JM', 'Coutinho LL']\n",
      "source: PLoS One. 2015 Jun 4;10(6):e0128350. doi: 10.1371/journal.pone.0128350. eCollection 2015.\n",
      "\n",
      "title: Sonic Hedgehog mutations are not a common cause of congenital hypopituitarism in the absence of complex midline cerebral defects.\n",
      "authors: ['Paulo SS', 'Fernandes-Rosa FL', 'Turatti W', 'Coeli-Lacchini FB', 'Martinelli CE Jr', 'Nakiri GS', 'Moreira AC', 'Santos AC', 'de Castro M', 'Antonini SR']\n",
      "source: Clin Endocrinol (Oxf). 2015 Apr;82(4):562-9. doi: 10.1111/cen.12565. Epub 2014 Sep 22.\n",
      "\n",
      "title: Holoprosencephaly: ZIC2 mutation in a case with panhypopituitarism.\n",
      "authors: ['Tasdemir S', 'Sahin I', 'Cayir A', 'Doneray H', 'Solomon BD', 'Muenke M', 'Yuce I', 'Tatar A']\n",
      "source: J Pediatr Endocrinol Metab. 2014 Jul;27(7-8):777-81. doi: 10.1515/jpem-2013-0449.\n",
      "\n",
      "title: Hedgehog signaling in human medullary thyroid carcinoma: a novel signaling pathway.\n",
      "authors: ['Bohinc B', 'Michelotti G', 'Diehl AM']\n",
      "source: Thyroid. 2013 Sep;23(9):1119-26. doi: 10.1089/thy.2012.0474.\n",
      "\n",
      "title: Hedgehog signaling regulates MyoD expression and activity.\n",
      "authors: ['Voronova A', 'Coyne E', 'Al Madhoun A', 'Fair JV', 'Bosiljcic N', 'St-Louis C', 'Li G', 'Thurig S', 'Wallace VA', 'Wiper-Bergeron N', 'Skerjanc IS']\n",
      "source: J Biol Chem. 2013 Feb 8;288(6):4389-404. doi: 10.1074/jbc.M112.400184. Epub 2012 Dec 24.\n",
      "\n",
      "title: Relatively high frequency of non-synonymous GLI2 variants in patients with congenital hypopituitarism without holoprosencephaly.\n",
      "authors: ['Franca MM', 'Jorge AA', 'Carvalho LR', 'Costalonga EF', 'Otto AP', 'Correa FA', 'Mendonca BB', 'Arnhold IJ']\n",
      "source: Clin Endocrinol (Oxf). 2013 Apr;78(4):551-7. doi: 10.1111/cen.12044.\n",
      "\n",
      "title: Osteopontin is a novel downstream target of SOX9 with diagnostic implications for progression of liver fibrosis in humans.\n",
      "authors: ['Pritchett J', 'Harvey E', 'Athwal V', 'Berry A', 'Rowe C', 'Oakley F', 'Moles A', 'Mann DA', 'Bobola N', 'Sharrocks AD', 'Thomson BJ', 'Zaitoun AM', 'Irving WL', 'Guha IN', 'Hanley NA', 'Hanley KP']\n",
      "source: Hepatology. 2012 Sep;56(3):1108-16. doi: 10.1002/hep.25758.\n",
      "\n",
      "title: Genetic disorders of the pituitary.\n",
      "authors: ['Cohen LE']\n",
      "source: Curr Opin Endocrinol Diabetes Obes. 2012 Feb;19(1):33-9. doi: 10.1097/MED.0b013e32834ed639.\n",
      "\n",
      "title: Elevated Hedgehog/Gli signaling causes beta-cell dedifferentiation in mice.\n",
      "authors: ['Landsman L', 'Parent A', 'Hebrok M']\n",
      "source: Proc Natl Acad Sci U S A. 2011 Oct 11;108(41):17010-5. doi: 10.1073/pnas.1105404108. Epub 2011 Oct 3.\n",
      "\n",
      "title: Novel heterozygous nonsense GLI2 mutations in patients with hypopituitarism and ectopic posterior pituitary lobe without holoprosencephaly.\n",
      "authors: ['Franca MM', 'Jorge AA', 'Carvalho LR', 'Costalonga EF', 'Vasques GA', 'Leite CC', 'Mendonca BB', 'Arnhold IJ']\n",
      "source: J Clin Endocrinol Metab. 2010 Nov;95(11):E384-91. doi: 10.1210/jc.2010-1050. Epub 2010 Aug 4.\n",
      "\n",
      "title: Primary cilia regulate Gli/Hedgehog activation in pancreas.\n",
      "authors: ['Cervantes S', 'Lau J', 'Cano DA', 'Borromeo-Austin C', 'Hebrok M']\n",
      "source: Proc Natl Acad Sci U S A. 2010 Jun 1;107(22):10109-14. doi: 10.1073/pnas.0909900107. Epub 2010 May 17.\n",
      "\n",
      "title: Hartsfield holoprosencephaly-ectrodactyly syndrome in five male patients: further delineation and review.\n",
      "authors: ['Vilain C', 'Mortier G', 'Van Vliet G', 'Dubourg C', 'Heinrichs C', 'de Silva D', 'Verloes A', 'Baumann C']\n",
      "source: Am J Med Genet A. 2009 Jul;149A(7):1476-81. doi: 10.1002/ajmg.a.32678.\n",
      "\n",
      "title: Sonic hedgehog-induced type 3 deiodinase blocks thyroid hormone action enhancing proliferation of normal and malignant keratinocytes.\n",
      "authors: ['Dentice M', 'Luongo C', 'Huang S', 'Ambrosio R', 'Elefante A', 'Mirebeau-Prunier D', 'Zavacki AM', 'Fenzi G', 'Grachtchouk M', 'Hutchin M', 'Dlugosz AA', 'Bianco AC', 'Missero C', 'Larsen PR', 'Salvatore D']\n",
      "source: Proc Natl Acad Sci U S A. 2007 Sep 4;104(36):14466-71. doi: 10.1073/pnas.0706754104. Epub 2007 Aug 24.\n",
      "\n",
      "title: Holoprosencephaly.\n",
      "authors: ['Dubourg C', 'Bendavid C', 'Pasquier L', 'Henry C', 'Odent S', 'David V']\n",
      "source: Orphanet J Rare Dis. 2007 Feb 2;2:8. doi: 10.1186/1750-1172-2-8.\n",
      "\n",
      "title: Hedgehog signaling and Bmi-1 regulate self-renewal of normal and malignant human mammary stem cells.\n",
      "authors: ['Liu S', 'Dontu G', 'Mantle ID', 'Patel S', 'Ahn NS', 'Jackson KW', 'Suri P', 'Wicha MS']\n",
      "source: Cancer Res. 2006 Jun 15;66(12):6063-71. doi: 10.1158/0008-5472.CAN-06-0054.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "database = \"PubMed\"\n",
    "word = 'Gli2, diabetes'\n",
    "res= int('30')\n",
    "email= \"rodrigoce9@gmail.com\"\n",
    "\n",
    "Entrez.email= email\n",
    "handle=Entrez.esearch(db = database, term=word, retmax= res)\n",
    "record=Entrez.read(handle)\n",
    "handle.close()\n",
    "idlist= record['IdList']\n",
    "\n",
    "handle = Entrez.efetch(db=database, id=idlist, rettype=\"medline\", retmode=\"text\")\n",
    "records = Medline.parse(handle)\n",
    "\n",
    "\n",
    "for record in records:\n",
    "    #print(record) \n",
    "    #print(\"PMID:\",record.get(\"PMID\",\"-\"))\n",
    "    print(\"title:\", record.get(\"TI\", \"-\"))\n",
    "    #print('abstract:', record.get('AB', '-'))\n",
    "    print(\"authors:\", record.get(\"AU\", \"-\"))\n",
    "    print(\"source:\", record.get(\"SO\", \"-\"))\n",
    "    print(\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cadec206",
   "metadata": {},
   "source": [
    "## Searching for GLI2 sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "82f64b09",
   "metadata": {},
   "outputs": [],
   "source": [
    "database = 'nucleotide'\n",
    "word = 'GLI2 and homo sapiens and Chromosome 2 and not predicted and not unverified '\n",
    "res= '15'\n",
    "email= 'rodrigoce9@gmail.com'\n",
    "Entrez.email= email\n",
    "handle_search=Entrez.esearch(db = database, term=word, retmax= res)\n",
    "record=Entrez.read(handle_search)\n",
    "handle_search.close()\n",
    "idlist= record['IdList']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6289eaac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NM_001374354.1 - Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 4, mRNA\n",
      "NM_001374353.1 - Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 3, mRNA\n",
      "NM_001371271.1 - Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 1, mRNA\n",
      "NM_005270.5 - Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 2, mRNA\n",
      "NM_003743.5 - Homo sapiens nuclear receptor coactivator 1 (NCOA1), transcript variant 1, mRNA\n",
      "NM_001362950.1 - Homo sapiens nuclear receptor coactivator 1 (NCOA1), transcript variant 4, mRNA\n",
      "NM_001362952.1 - Homo sapiens nuclear receptor coactivator 1 (NCOA1), transcript variant 5, mRNA\n",
      "NM_147223.3 - Homo sapiens nuclear receptor coactivator 1 (NCOA1), transcript variant 2, mRNA\n",
      "NM_001362954.1 - Homo sapiens nuclear receptor coactivator 1 (NCOA1), transcript variant 6, mRNA\n",
      "NM_001362955.1 - Homo sapiens nuclear receptor coactivator 1 (NCOA1), transcript variant 7, mRNA\n",
      "NM_147233.2 - Homo sapiens nuclear receptor coactivator 1 (NCOA1), transcript variant 3, mRNA\n",
      "NG_009030.2 - Homo sapiens GLI family zinc finger 2 (GLI2), RefSeqGene on chromosome 2\n",
      "AY493739.1 - Homo sapiens clone ASov4 GLI2 (GLI2) mRNA, partial cds, alternatively spliced\n",
      "AY493738.1 - Homo sapiens clone ASov1 GLI2 (GLI2) mRNA, partial cds, alternatively spliced\n",
      "AY493737.1 - Homo sapiens GLI2 (GLI2) mRNA, partial cds, alternatively spliced\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.efetch(db=database, id=idlist, rettype=\"gb\") \n",
    "records = list(SeqIO.parse(handle,\"gb\"))\n",
    "handle.close()\n",
    "for info in records:\n",
    "    print(info.id, '-', info.description)\n",
    "    #print('length of seq:', len(info.seq)) #to check the length of the sequences"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c816412",
   "metadata": {},
   "source": [
    "The selection of the id has to be done manually as there is no pattern in writing the titles of the queries.\\\n",
    "The id __NG_009030.2__ is the only one where the annotated sequence is in chromosome 2, is not a mRNA and actually is RefSeq. This means that the sequence is being used as a standard for well-characterized genes. So id __NG_009030.2__ will be used from now on."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "db4fb64a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#The correspondent information of NG_015830.1 was downloaded to a file\n",
    "import os\n",
    "Entrez.email = \"rodrigoce9@gmail.com\"\n",
    "filename = \"NG_009030.2.gb\"\n",
    "if not os.path.isfile(filename):\n",
    "    net_handle = Entrez.efetch(db=\"nucleotide\", id=\"NG_009030.2\", rettype=\"gb\", retmode=\"text\")\n",
    "    out_handle = open(filename, \"w\")\n",
    "    out_handle.write(net_handle.read())\n",
    "    out_handle.close()\n",
    "    net_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "24dd76b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The length of the sequence: 263786\n",
      "\n",
      "Type of features: {'source': 1, 'gene': 1, 'mRNA': 1, 'exon': 14, 'CDS': 1, 'misc_feature': 17}\n",
      "\n",
      "Comment from NCBI: REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
      "reference sequence was derived from AC018866.9, AC017033.5,\n",
      "KF510752.1, KF510212.1 and AC016764.8.\n",
      "This sequence is a reference standard in the RefSeqGene project.\n",
      "On Sep 14, 2022 this sequence version replaced NG_009030.1.\n",
      "Summary: This gene encodes a protein which belongs to the C2H2-type\n",
      "zinc finger protein subclass of the Gli family. Members of this\n",
      "subclass are characterized as transcription factors which bind DNA\n",
      "through zinc finger motifs. These motifs contain conserved H-C\n",
      "links. Gli family zinc finger proteins are mediators of Sonic\n",
      "hedgehog (Shh) signaling and they are implicated as potent\n",
      "oncogenes in the embryonal carcinoma cell. The protein encoded by\n",
      "this gene localizes to the cytoplasm and activates patched\n",
      "Drosophila homolog (PTCH) gene expression. It is also thought to\n",
      "play a role during embryogenesis. The encoded protein is associated\n",
      "with several phenotypes- Greig cephalopolysyndactyly syndrome,\n",
      "Pallister-Hall syndrome, preaxial polydactyly type IV, postaxial\n",
      "polydactyly types A1 and B. [provided by RefSeq, Jul 2008].\n",
      "\n",
      "Location of the CDS on the original sequence: join{[66454:66602](+), [196494:196600](+), [220376:220579](+), [224378:224564](+), [237847:238049](+), [239526:239740](+), [241074:241197](+), [244108:244243](+), [247567:247717](+), [251849:252014](+), [253604:253877](+), [255411:255748](+), [257341:259809](+)}\n"
     ]
    }
   ],
   "source": [
    "record = SeqIO.read(open(\"NG_009030.2.gb\"), format=\"genbank\")\n",
    "position=0\n",
    "record_types={}\n",
    "for x in record.features:\n",
    "    x.\n",
    "    record_types[x.type]=record_types.get(x.type,0)+1\n",
    "for i in range(len(record.features)):\n",
    "    if record.features[i].type == \"CDS\":\n",
    "        position=i\n",
    "print(\"The length of the sequence: {}\\n\".format(len(record.seq)))\n",
    "print(\"Type of features: {}\\n\".format(record_types))\n",
    "print(\"Comment from NCBI: {}\\n\".format(record.annotations[\"comment\"]))\n",
    "#checking the location of the CDS on the original sequence\n",
    "print(\"Location of the CDS on the original sequence: {}\".format(record.features[position].location))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "22a6d9ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "CDS_nuc_seq=record.seq[250000:260000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cf5ba8b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#saving the CDS_nucleotides and CD_aminoacid seqs in files\n",
    "filename_CDS_nucl = \"CDS_nucleot_GLI2_seq.fasta\"\n",
    "filename_CDS_prot = \"CDS_prot_GLI2_seq.fasta\"\n",
    "input_handle  = open(filename, \"r\")\n",
    "output_handle_nucl = open(filename_CDS_nucl, \"w\")\n",
    "output_handle_prot = open(filename_CDS_prot, \"w\")\n",
    "for seq_record in SeqIO.parse(input_handle, \"genbank\") :\n",
    "    output_handle_nucl.write(\">\\n%s\" % (CDS_nuc_seq))\n",
    "    output_handle_prot.write(\">\\n%s\" % (\"\".join(record.features[position].qualifiers['translation']))) #without join, output is a list\n",
    "    \n",
    "output_handle_nucl.close()\n",
    "output_handle_prot.close()\n",
    "input_handle.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eff68e9c",
   "metadata": {},
   "source": [
    "## BLASTN - for all organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "536dd77e",
   "metadata": {},
   "outputs": [],
   "source": [
    "record = SeqIO.read(open(\"CDS_nucleot_GLI2_seq.fasta\"), format=\"fasta\") \n",
    "print (len(record.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bff8132",
   "metadata": {},
   "outputs": [],
   "source": [
    "Blast = NCBIWWW.qblast(\"blastn\", \"nt\", record.seq)\n",
    "with open('blast_DNA_GLI2_file.xml', \"w\") as out_handle:\n",
    "        out_handle.write(Blast.read())\n",
    "Blast.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "547f0d5f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Accession:  NG_009030\n",
      "Definition:  Homo sapiens GLI family zinc finger 2 (GLI2), RefSeqGene on chromosome 2\n",
      "E-value:  0.0\n",
      "E-value:  1.23303e-94\n",
      "E-value:  2.71593e-90\n",
      "E-value:  4.03079e-88\n",
      "E-value:  1.71394e-86\n",
      "E-value:  5.98222e-86\n",
      "E-value:  2.54371e-84\n",
      "E-value:  8.87841e-84\n",
      "E-value:  8.87841e-84\n",
      "E-value:  8.87841e-84\n",
      "E-value:  3.09887e-83\n",
      "E-value:  3.77519e-82\n",
      "E-value:  4.59913e-81\n",
      "E-value:  4.59913e-81\n",
      "E-value:  1.60525e-80\n",
      "E-value:  5.60289e-80\n",
      "E-value:  2.38241e-78\n",
      "E-value:  1.23412e-75\n",
      "E-value:  4.30749e-75\n",
      "E-value:  4.30749e-75\n",
      "E-value:  1.50346e-74\n",
      "E-value:  1.50346e-74\n",
      "E-value:  1.50346e-74\n",
      "E-value:  5.2476e-74\n",
      "E-value:  6.39289e-73\n",
      "E-value:  6.39289e-73\n",
      "E-value:  2.23134e-72\n",
      "E-value:  2.23134e-72\n",
      "E-value:  9.48789e-71\n",
      "E-value:  1.40813e-68\n",
      "E-value:  3.78187e-44\n",
      "E-value:  3.78187e-44\n",
      "E-value:  5.61279e-42\n",
      "E-value:  5.61279e-42\n",
      "E-value:  3.54206e-38\n",
      "E-value:  5.9981e-29\n",
      "E-value:  3.54519e-19\n",
      "E-value:  4.31892e-18\n",
      "E-value:  4.04506e-12\n",
      "E-value:  4.92789e-11\n",
      "E-value:  6.0034e-10\n",
      "\n",
      "Accession:  AC016764\n",
      "Definition:  Homo sapiens BAC clone RP11-549G13 from 2, complete sequence\n",
      "E-value:  0.0\n",
      "E-value:  1.82998e-92\n",
      "E-value:  2.71593e-90\n",
      "E-value:  8.87841e-84\n",
      "E-value:  8.87841e-84\n",
      "E-value:  3.09887e-83\n",
      "E-value:  4.59913e-81\n",
      "E-value:  5.60289e-80\n",
      "E-value:  1.9556e-79\n",
      "E-value:  8.31542e-78\n",
      "E-value:  2.90237e-77\n",
      "E-value:  5.2476e-74\n",
      "E-value:  2.23134e-72\n",
      "E-value:  1.71545e-67\n",
      "E-value:  2.90493e-58\n",
      "E-value:  1.23521e-56\n",
      "E-value:  2.23331e-53\n",
      "E-value:  2.72073e-52\n",
      "E-value:  5.9928e-48\n",
      "E-value:  6.83778e-41\n",
      "E-value:  2.38662e-40\n",
      "E-value:  6.40419e-35\n",
      "E-value:  1.41062e-30\n",
      "E-value:  5.9981e-29\n",
      "E-value:  4.31892e-18\n",
      "\n",
      "Accession:  NM_001374354\n",
      "Definition:  Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 4, mRNA\n",
      "E-value:  0.0\n",
      "E-value:  3.52023e-171\n",
      "E-value:  1.95042e-136\n",
      "E-value:  2.90237e-77\n",
      "\n",
      "Accession:  NM_001374353\n",
      "Definition:  Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 3, mRNA\n",
      "E-value:  0.0\n",
      "E-value:  3.52023e-171\n",
      "E-value:  1.95042e-136\n",
      "E-value:  2.90237e-77\n",
      "\n",
      "Accession:  NM_001371271\n",
      "Definition:  Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 1, mRNA\n",
      "E-value:  0.0\n",
      "E-value:  3.52023e-171\n",
      "E-value:  1.95042e-136\n",
      "E-value:  2.90237e-77\n"
     ]
    }
   ],
   "source": [
    "results_Blastn = open(\"blast_DNA_GLI2_file.xml\")\n",
    "blastn_records = NCBIXML.read(results_Blastn)\n",
    "for parameter in blastn_records.alignments[0:5]:\n",
    "    print()\n",
    "    print('Accession: ', parameter.accession)\n",
    "    print('Definition: ', parameter.hit_def)    \n",
    "    for e_v in parameter.hsps:\n",
    "        print('E-value: ', e_v.expect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ffa6e58a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gi|2301838502|ref|NG_009030.2| Homo sapiens GLI family zinc finger 2 (GLI2), RefSeqGene on chromosome 2\n",
      "           Length = 263786\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2301838502|ref|NG_009030.2| Homo sapiens GLI family zinc finger 2 (GLI2), RefSeqGene on chromosome 2\n",
      "lenght: 263786\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAG...\n",
      "\n",
      "gi|18097434|gb|AC016764.8| Homo sapiens BAC clone RP11-549G13 from 2, complete sequence\n",
      "           Length = 155868\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|18097434|gb|AC016764.8| Homo sapiens BAC clone RP11-549G13 from 2, complete sequence\n",
      "lenght: 155868\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAG...\n",
      "\n",
      "gi|1748462987|ref|NM_001374354.1| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 4, mRNA\n",
      "           Length = 7030\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1748462987|ref|NM_001374354.1| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 4, mRNA\n",
      "lenght: 7030\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|1748462113|ref|NM_001374353.1| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 3, mRNA\n",
      "           Length = 7136\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1748462113|ref|NM_001374353.1| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 3, mRNA\n",
      "lenght: 7136\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|1698389211|ref|NM_001371271.1| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 1, mRNA\n",
      "           Length = 7187\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1698389211|ref|NM_001371271.1| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 1, mRNA\n",
      "lenght: 7187\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|1698388715|ref|NM_005270.5| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 2, mRNA\n",
      "           Length = 6927\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1698388715|ref|NM_005270.5| Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant 2, mRNA\n",
      "lenght: 6927\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|2217326993|ref|XM_011510971.3| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X4, mRNA\n",
      "           Length = 6705\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2217326993|ref|XM_011510971.3| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X4, mRNA\n",
      "lenght: 6705\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|2217326991|ref|XM_047443947.1| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X3, mRNA\n",
      "           Length = 6876\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2217326991|ref|XM_047443947.1| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X3, mRNA\n",
      "lenght: 6876\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|2217326989|ref|XM_017003818.2| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X2, mRNA\n",
      "           Length = 6679\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2217326989|ref|XM_017003818.2| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X2, mRNA\n",
      "lenght: 6679\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|2217326987|ref|XM_011510969.3| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X1, mRNA\n",
      "           Length = 6730\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2217326987|ref|XM_011510969.3| PREDICTED: Homo sapiens GLI family zinc finger 2 (GLI2), transcript variant X1, mRNA\n",
      "lenght: 6730\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|62088287|dbj|AB209354.1| Homo sapiens mRNA for GLI-Kruppel family member GLI2 isoform beta variant protein\n",
      "           Length = 5197\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|62088287|dbj|AB209354.1| Homo sapiens mRNA for GLI-Kruppel family member GLI2 isoform beta variant protein\n",
      "lenght: 5197\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|3061315|dbj|AB007296.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2b\n",
      "           Length = 4909\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|3061315|dbj|AB007296.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2b\n",
      "lenght: 4909\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|3061313|dbj|AB007295.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2a\n",
      "           Length = 4960\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|3061313|dbj|AB007295.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2a\n",
      "lenght: 4960\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|68164336|gb|DQ086814.1| Homo sapiens GLI-Kruppel transcription factor GLI2 (GLI2) mRNA, complete cds\n",
      "           Length = 4761\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|68164336|gb|DQ086814.1| Homo sapiens GLI-Kruppel transcription factor GLI2 (GLI2) mRNA, complete cds\n",
      "lenght: 4761\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|3061319|dbj|AB007298.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2d\n",
      "           Length = 3678\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|3061319|dbj|AB007298.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2d\n",
      "lenght: 3678\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|3061317|dbj|AB007297.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2g\n",
      "           Length = 3729\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|3061317|dbj|AB007297.1| Homo sapiens mRNA for hGLI2, complete cds, clone:hGli2g\n",
      "lenght: 3729\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GGCTCCATCCTGGAAAACTTCAGTGGCAGTGGGGGCGGCGGGCCCGCGGGGCTGCTGCCGAACCCGCGGCTGTCG...\n",
      "\n",
      "gi|2322705702|ref|NG_096463.1| Homo sapiens H3K4me1 hESC enhancer GRCh37_chr2:121740347-121740857 (LOC127274613) on chromosome 2\n",
      "           Length = 711\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2322705702|ref|NG_096463.1| Homo sapiens H3K4me1 hESC enhancer GRCh37_chr2:121740347-121740857 (LOC127274613) on chromosome 2\n",
      "lenght: 711\n",
      "TGGGCCCCCTGGGGTGCCTTGACTGACTGAACCGCCTCCTTCTAGTTCGAGGGCTGCTCGAAGGCCTACTCCCGC...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "TGGGCCCCCTGGGGTGCCTTGACTGACTGAACCGCCTCCTTCTAGTTCGAGGGCTGCTCGAAGGCCTACTCCCGC...\n",
      "\n",
      "gi|111494027|gb|BC111410.1| Homo sapiens GLI family zinc finger 2, mRNA (cDNA clone IMAGE:40001997), complete cds\n",
      "           Length = 2743\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|111494027|gb|BC111410.1| Homo sapiens GLI family zinc finger 2, mRNA (cDNA clone IMAGE:40001997), complete cds\n",
      "lenght: 2743\n",
      "CACCACAGCCACAGGCCTGTCAGGACAGCATCCAGCCCCAGCCCTTGCCCTCACCAGGGGTCAACCAGGTGTCCA...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "CACCACAGCCACAGGCCTGTCAGGACAGCATCCAGCCCCAGCCCTTGCCCTCACCAGGGGTCAACCAGGTGTCCA...\n",
      "\n",
      "gi|1049010973|gb|AH001472.2| Homo sapiens exon Y gene, complete cds\n",
      "           Length = 1102\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1049010973|gb|AH001472.2| Homo sapiens exon Y gene, complete cds\n",
      "lenght: 1102\n",
      "AAGGCCTACTCCCGCCTGGAGAACCTGAAGACACACCTGCGGTCCCACACCGGGGAGAAGCCATATGTGTGTGAG...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "AAGGCCTACTCCCGCCTGGAGAACCTGAAGACACACCTGCGGTCCCACACCGGGGAGAAGCCATATGTGTGTGAG...\n",
      "\n",
      "gi|439900|dbj|D14828.1|HUMTHP2 Homo sapiens THP2 mRNA for Tax helper protein 2, complete cds\n",
      "           Length = 2001\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|439900|dbj|D14828.1|HUMTHP2 Homo sapiens THP2 mRNA for Tax helper protein 2, complete cds\n",
      "lenght: 2001\n",
      "GCTGTGTCAGTCCAGCCCCGGGGCCCAGTCGTCCTGCAGCAGCGAGCCCTCTCCTCTGGGCAGTGCCCCCAACAA...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GCTGTGTCAGTCCAGCCCCGGGGCCCAGTCGTCCTGCAGCAGCGAGCCCTCTCCTCTGGGCAGTGCCCCCAACAA...\n",
      "\n",
      "gi|439899|dbj|D14827.1|HUMTHP1 Homo sapiens THP1 mRNA for Tax helper protein 1, complete cds\n",
      "           Length = 2105\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|439899|dbj|D14827.1|HUMTHP1 Homo sapiens THP1 mRNA for Tax helper protein 1, complete cds\n",
      "lenght: 2105\n",
      "GCTGTGTCAGTCCAGCCCCGGGGCCCAGTCGTCCTGCAGCAGCGAGCCCTCTCCTCTGGGCAGTGCCCCCAACAA...\n",
      "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||...\n",
      "GCTGTGTCAGTCCAGCCCCGGGGCCCAGTCGTCCTGCAGCAGCGAGCCCTCTCCTCTGGGCAGTGCCCCCAACAA...\n",
      "\n",
      "gi|1152003149|ref|NG_016599.2| Homo sapiens phosphorylase kinase regulatory subunit alpha 1 (PHKA1), RefSeqGene on chromosome X\n",
      "           Length = 142368\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1152003149|ref|NG_016599.2| Homo sapiens phosphorylase kinase regulatory subunit alpha 1 (PHKA1), RefSeqGene on chromosome X\n",
      "lenght: 142368\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|| ||| ||| |||||||  |||||||  |||||  |||||||| ||||||||| | |  |||||||||||||||...\n",
      "TTGTTTGTTTGTTTGAGATGGAGTCTCATTCTGTTGCCCAGGCTGGAGTGCAATGGCGTGATCTTGGCTCACTGC...\n",
      "\n",
      "gi|1870022710|gb|AC279654.1| Homo sapiens chromosome unknown clone VMRC59-219A07, complete sequence\n",
      "           Length = 176397\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1870022710|gb|AC279654.1| Homo sapiens chromosome unknown clone VMRC59-219A07, complete sequence\n",
      "lenght: 176397\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|| ||| ||| |||||||  |||||||  |||||  |||||||| ||||||||| | |  |||||||||||||||...\n",
      "TTGTTTGTTTGTTTGAGATGGAGTCTCATTCTGTTGCCCAGGCTGGAGTGCAATGGCGTGATCTTGGCTCACTGC...\n",
      "\n",
      "gi|1870022582|gb|AC279566.1| Homo sapiens chromosome unknown clone VMRC59-405N02, complete sequence\n",
      "           Length = 166133\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1870022582|gb|AC279566.1| Homo sapiens chromosome unknown clone VMRC59-405N02, complete sequence\n",
      "lenght: 166133\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|| ||| ||| |||||||  |||||||  |||||  |||||||| ||||||||| | |  |||||||||||||||...\n",
      "TTGTTTGTTTGTTTGAGATGGAGTCTCATTCTGTTGCCCAGGCTGGAGTGCAATGGCGTGATCTTGGCTCACTGC...\n",
      "\n",
      "gi|29150138|emb|BX295541.1| Homo sapiens chromosome X PAC RP1-172N19 map Xq13 markers PHKA1-DXS227, complete sequence\n",
      "           Length = 176589\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|29150138|emb|BX295541.1| Homo sapiens chromosome X PAC RP1-172N19 map Xq13 markers PHKA1-DXS227, complete sequence\n",
      "lenght: 176589\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|| ||| ||| |||||||  |||||||  |||||  |||||||| ||||||||| | |  |||||||||||||||...\n",
      "TTGTTTGTTTGTTTGAGATGGAGTCTCATTCTGTTGCCCAGGCTGGAGTGCAATGGCGTGATCTTGGCTCACTGC...\n",
      "\n",
      "gi|1909942459|dbj|AP023478.1| Homo sapiens DNA, chromosome 18, nearly complete genome\n",
      "           Length = 77846715\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1909942459|dbj|AP023478.1| Homo sapiens DNA, chromosome 18, nearly complete genome\n",
      "lenght: 77846715\n",
      "ATCCATTTGGAGTTAATTTTTTTTTTTTTT--GAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGT...\n",
      "||| ||||   |||||||||||||||||||  ||||   ||||| ||||||||||||||| | ||||||| | | ...\n",
      "ATCTATTTTAGGTTAATTTTTTTTTTTTTTTTGAGATGAAGTCTACCTCTGTCACCCAGGTTGGAGTGCAGTGGC...\n",
      "\n",
      "gi|2033714914|gb|CP068260.2| Homo sapiens isolate CHM13 chromosome 18\n",
      "           Length = 80542538\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2033714914|gb|CP068260.2| Homo sapiens isolate CHM13 chromosome 18\n",
      "lenght: 80542538\n",
      "ATCCATTTGGAGTTAATTTTTTTTTTTTTT---GAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAG...\n",
      "||| ||||   |||||||||||||||||||   ||||   ||||| ||||||||||||||| | ||||||| | |...\n",
      "ATCTATTTTAGGTTAATTTTTTTTTTTTTTTTTGAGATGAAGTCTACCTCTGTCACCCAGGTTGGAGTGCAGTGG...\n",
      "\n",
      "gi|22532582|gb|AC011774.9| Homo sapiens chromosome 18, clone RP11-380C8, complete sequence\n",
      "           Length = 208039\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|22532582|gb|AC011774.9| Homo sapiens chromosome 18, clone RP11-380C8, complete sequence\n",
      "lenght: 208039\n",
      "ATCCATTTGGAGTTAATTTTTTTTTTTTTT---GAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAG...\n",
      "||| ||||   |||||||||||||||||||   ||||   ||||| ||||||||||||||| | ||||||| | |...\n",
      "ATCTATTTTAGGTTAATTTTTTTTTTTTTTTTTGAGATGAAGTCTACCTCTGTCACCCAGGTTGGAGTGCAGTGG...\n",
      "\n",
      "gi|19745137|gb|AC015878.7| Homo sapiens, clone RP11-296E23, complete sequence\n",
      "           Length = 160569\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|19745137|gb|AC015878.7| Homo sapiens, clone RP11-296E23, complete sequence\n",
      "lenght: 160569\n",
      "ATCCATTTGGAGTTAATTTTTTTTTTTTTT---GAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAG...\n",
      "||| ||||   |||||||||||||||||||   ||||   ||||| ||||||||||||||| | ||||||| | |...\n",
      "ATCTATTTTAGGTTAATTTTTTTTTTTTTTTTTGAGATGAAGTCTACCTCTGTCACCCAGGTTGGAGTGCAGTGG...\n",
      "\n",
      "gi|321267517|ref|NG_028118.1| Homo sapiens arrestin beta 1 (ARRB1), RefSeqGene on chromosome 11\n",
      "           Length = 98710\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|321267517|ref|NG_028118.1| Homo sapiens arrestin beta 1 (ARRB1), RefSeqGene on chromosome 11\n",
      "lenght: 98710\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|||||||||||||||||||| |||||  |||| | ||||||||| ||||||| | |    ||||  |||||||||...\n",
      "TTTTTTTTTTTTTTGAGACAAAGTCTTGCTCTCTTACCCAGGCTGGAGTGCAGTGGCATGATCTCAGCTCACTGC...\n",
      "\n",
      "gi|83699636|gb|DQ314865.1| Homo sapiens arrestin, beta 1 (ARRB1) gene, complete cds\n",
      "           Length = 89549\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|83699636|gb|DQ314865.1| Homo sapiens arrestin, beta 1 (ARRB1) gene, complete cds\n",
      "lenght: 89549\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|||||||||||||||||||| |||||  |||| | ||||||||| ||||||| | |    ||||  |||||||||...\n",
      "TTTTTTTTTTTTTTGAGACAAAGTCTTGCTCTCTTACCCAGGCTGGAGTGCAGTGGCATGATCTCAGCTCACTGC...\n",
      "\n",
      "gi|13488761|dbj|AP001972.4| Homo sapiens genomic DNA, chromosome 11q, clone:CTD-2562J17, complete sequences\n",
      "           Length = 215225\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|13488761|dbj|AP001972.4| Homo sapiens genomic DNA, chromosome 11q, clone:CTD-2562J17, complete sequences\n",
      "lenght: 215225\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|||||||||||||||||||| |||||  |||| | ||||||||| ||||||| | |    ||||  |||||||||...\n",
      "TTTTTTTTTTTTTTGAGACAAAGTCTTGCTCTCTTACCCAGGCTGGAGTGCAGTGGCATGATCTCAGCTCACTGC...\n",
      "\n",
      "gi|1027250656|ref|NG_047146.1| Homo sapiens adhesion G protein-coupled receptor E2 (ADGRE2), RefSeqGene on chromosome 19\n",
      "           Length = 61372\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1027250656|ref|NG_047146.1| Homo sapiens adhesion G protein-coupled receptor E2 (ADGRE2), RefSeqGene on chromosome 19\n",
      "lenght: 61372\n",
      "TTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGCA...\n",
      "||||| | |||||||||| ||||||  || |||  |||||||| ||||||||| | || ||||||||| || |||...\n",
      "TTTTTGTGTTTTTGAGACGGAGTCTTGCTGTGTTGCCCAGGCTGGAGTGCAATGGCGCAATCTTGGCTTACCGCA...\n",
      "\n",
      "gi|2033715243|gb|CP068259.2| Homo sapiens isolate CHM13 chromosome 19\n",
      "           Length = 61707364\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2033715243|gb|CP068259.2| Homo sapiens isolate CHM13 chromosome 19\n",
      "lenght: 61707364\n",
      "TTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGCA...\n",
      "||||| | |||||||||| ||||||  || |||  |||||||| ||||||||| | || ||||||||| || |||...\n",
      "TTTTTGTGTTTTTGAGACGGAGTCTTGCTGTGTTGCCCAGGCTGGAGTGCAATGGCGCAATCTTGGCTTACCGCA...\n",
      "\n",
      "gi|1909942460|dbj|AP023479.1| Homo sapiens DNA, chromosome 19, nearly complete genome\n",
      "           Length = 59105444\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1909942460|dbj|AP023479.1| Homo sapiens DNA, chromosome 19, nearly complete genome\n",
      "lenght: 59105444\n",
      "TTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGCA...\n",
      "||||| | |||||||||| ||||||  || |||  |||||||| ||||||||| | || ||||||||| || |||...\n",
      "TTTTTGTGTTTTTGAGACGGAGTCTTGCTGTGTTGCCCAGGCTGGAGTGCAATGGCGCAATCTTGGCTTACCGCA...\n",
      "\n",
      "gi|18030136|gb|AC090427.2| Homo sapiens chromosome 19 clone CTD-2292J18, complete sequence\n",
      "           Length = 109233\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|18030136|gb|AC090427.2| Homo sapiens chromosome 19 clone CTD-2292J18, complete sequence\n",
      "lenght: 109233\n",
      "TTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGCA...\n",
      "||||| | |||||||||| ||||||  || |||  |||||||| ||||||||| | || ||||||||| || |||...\n",
      "TTTTTGTGTTTTTGAGACGGAGTCTTGCTGTGTTGCCCAGGCTGGAGTGCAATGGCGCAATCTTGGCTTACCGCA...\n",
      "\n",
      "gi|11968291|gb|AC008983.7|AC008983 Homo sapiens chromosome 19 clone LLNLF-175G11, complete sequence\n",
      "           Length = 39426\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|11968291|gb|AC008983.7|AC008983 Homo sapiens chromosome 19 clone LLNLF-175G11, complete sequence\n",
      "lenght: 39426\n",
      "TTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGCA...\n",
      "||||| | |||||||||| ||||||  || |||  |||||||| ||||||||| | || ||||||||| || |||...\n",
      "TTTTTGTGTTTTTGAGACGGAGTCTTGCTGTGTTGCCCAGGCTGGAGTGCAATGGCGCAATCTTGGCTTACCGCA...\n",
      "\n",
      "gi|2935596|gb|AC004262.1|AC004262 Homo sapiens chromosome 19, cosmid R29368, complete sequence\n",
      "           Length = 44544\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2935596|gb|AC004262.1|AC004262 Homo sapiens chromosome 19, cosmid R29368, complete sequence\n",
      "lenght: 44544\n",
      "TTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGCA...\n",
      "||||| | |||||||||| ||||||  || |||  |||||||| ||||||||| | || ||||||||| || |||...\n",
      "TTTTTGTGTTTTTGAGACGGAGTCTTGCTGTGTTGCCCAGGCTGGAGTGCAATGGCGCAATCTTGGCTTACCGCA...\n",
      "\n",
      "gi|1675131106|ref|NM_001354259.2| Homo sapiens euchromatic histone lysine methyltransferase 1 (EHMT1), transcript variant 3, mRNA\n",
      "           Length = 4585\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1675131106|ref|NM_001354259.2| Homo sapiens euchromatic histone lysine methyltransferase 1 (EHMT1), transcript variant 3, mRNA\n",
      "lenght: 4585\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTG...\n",
      "||||  |||||||||| ||||||| || |||| |||||||||| | ||| ||| | | |  |||| |||||||||...\n",
      "TTTTCCTTTTTTTTGAAACAGAGTTTCGCTCTTGTCACCCAGGTTGGAGCGCAGTGGCGTGATCTCGGCTCACTG...\n",
      "\n",
      "gi|283837826|ref|NG_011776.1| Homo sapiens euchromatic histone lysine methyltransferase 1 (EHMT1), RefSeqGene on chromosome 9\n",
      "           Length = 224136\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|283837826|ref|NG_011776.1| Homo sapiens euchromatic histone lysine methyltransferase 1 (EHMT1), RefSeqGene on chromosome 9\n",
      "lenght: 224136\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTG...\n",
      "||||  |||||||||| ||||||| || |||| |||||||||| | ||| ||| | | |  |||| |||||||||...\n",
      "TTTTCCTTTTTTTTGAAACAGAGTTTCGCTCTTGTCACCCAGGTTGGAGCGCAGTGGCGTGATCTCGGCTCACTG...\n",
      "\n",
      "gi|28569844|emb|AL611925.27| Human DNA sequence from clone RP11-188C12 on chromosome 9, complete sequence\n",
      "           Length = 90096\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|28569844|emb|AL611925.27| Human DNA sequence from clone RP11-188C12 on chromosome 9, complete sequence\n",
      "lenght: 90096\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTG...\n",
      "||||  |||||||||| ||||||| || |||| |||||||||| | ||| ||| | | |  |||| |||||||||...\n",
      "TTTTCCTTTTTTTTGAAACAGAGTTTCGCTCTTGTCACCCAGGTTGGAGCGCAGTGGCGTGATCTCGGCTCACTG...\n",
      "\n",
      "gi|19807887|gb|AC108019.3| Homo sapiens BAC clone RP11-5K16 from 4, complete sequence\n",
      "           Length = 136867\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|19807887|gb|AC108019.3| Homo sapiens BAC clone RP11-5K16 from 4, complete sequence\n",
      "lenght: 136867\n",
      "CATTTGGAGTTAATTTTTTTTTTTTTT--------GAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAAT...\n",
      "||||| |||||||||||||||||||||        ||||| ||||||| | |||||||||||||| |||||||||...\n",
      "CATTTTGAGTTAATTTTTTTTTTTTTTTTTTTCTGGAGACTGAGTCTCACCCTGTCACCCAGGCTGGAGTGCAAT...\n",
      "\n",
      "gi|2033714908|gb|CP068262.2| Homo sapiens isolate CHM13 chromosome 16\n",
      "           Length = 96330374\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|2033714908|gb|CP068262.2| Homo sapiens isolate CHM13 chromosome 16\n",
      "lenght: 96330374\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTT----TTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGC...\n",
      "||||||||||| ||||||||||||||||||    |||||||| |||| ||||||| |||| ||||||| ||||||...\n",
      "TTGATCCATTTTGAGTTAATTTTTTTTTTTGTTTTTTGAGACGGAGTTTCCCTCTTGTCATCCAGGCTGGAGTGC...\n",
      "\n",
      "gi|1909942457|dbj|AP023476.1| Homo sapiens DNA, chromosome 16, nearly complete genome\n",
      "           Length = 94690957\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1909942457|dbj|AP023476.1| Homo sapiens DNA, chromosome 16, nearly complete genome\n",
      "lenght: 94690957\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTT----TTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGC...\n",
      "||||||||||| ||||||||||||||||||    |||||||| |||| ||||||| |||| ||||||| ||||||...\n",
      "TTGATCCATTTTGAGTTAATTTTTTTTTTTGTTTTTTGAGACGGAGTTTCCCTCTTGTCATCCAGGCTGGAGTGC...\n",
      "\n",
      "gi|16596520|gb|AC008870.8| Homo sapiens chromosome 16 clone CTD-2196E14, complete sequence\n",
      "           Length = 150522\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|16596520|gb|AC008870.8| Homo sapiens chromosome 16 clone CTD-2196E14, complete sequence\n",
      "lenght: 150522\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTT----TTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGC...\n",
      "||||||||||| ||||||||||||||||||    |||||||| |||| ||||||| |||| ||||||| ||||||...\n",
      "TTGATCCATTTTGAGTTAATTTTTTTTTTTGTTTTTTGAGACGGAGTTTCCCTCTTGTCATCCAGGCTGGAGTGC...\n",
      "\n",
      "gi|29336193|gb|AC012317.7| Homo sapiens chromosome 16 clone CTD-2385L22, complete sequence\n",
      "           Length = 133289\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|29336193|gb|AC012317.7| Homo sapiens chromosome 16 clone CTD-2385L22, complete sequence\n",
      "lenght: 133289\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTT----TTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGC...\n",
      "||||||||||| ||||||||||||||||||    |||||||| |||| ||||||| |||| ||||||| ||||||...\n",
      "TTGATCCATTTTGAGTTAATTTTTTTTTTTGTTTTTTGAGACGGAGTTTCCCTCTTGTCATCCAGGCTGGAGTGC...\n",
      "\n",
      "gi|20218783|emb|AL121658.4| BAC sequence from the SPG4 candidate region at 2p21-2p22 BAC 563N04 of library RPCI-11 from chromosome 2 of Homo sapiens (Human)\n",
      "           Length = 162692\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|20218783|emb|AL121658.4| BAC sequence from the SPG4 candidate region at 2p21-2p22 BAC 563N04 of library RPCI-11 from chromosome 2 of Homo sapiens (Human)\n",
      "lenght: 162692\n",
      "TTGATCCATTTGGAGTTAATTTTTTTTTTTTTT-------GAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGT...\n",
      "||||||||||| |||||||||||||||||||||       ||||  ||||||| ||||||| |||||||| ||||...\n",
      "TTGATCCATTTTGAGTTAATTTTTTTTTTTTTTTTTTTTTGAGATGGAGTCTCGCTCTGTCGCCCAGGCTGGAGT...\n",
      "\n",
      "gi|16944896|emb|AL356212.13| Human DNA sequence from clone RP4-686B20 on chromosome 1p31.3-32.3, complete sequence\n",
      "           Length = 31403\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|16944896|emb|AL356212.13| Human DNA sequence from clone RP4-686B20 on chromosome 1p31.3-32.3, complete sequence\n",
      "lenght: 31403\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCTGTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTGC...\n",
      "|||||| |||||||||||||||||||| ||||||  |||||||| ||||||| | ||||  |||  |||||||||...\n",
      "TTTTTTGTTTTTTTGAGACAGAGTCTCACTCTGTAGCCCAGGCTGGAGTGCAGTGGTGCAGTCTCAGCTCACTGC...\n",
      "\n",
      "gi|1024846813|ref|NG_046987.1| Homo sapiens DEAD-box helicase 10 (DDX10), RefSeqGene on chromosome 11\n",
      "           Length = 282906\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|1024846813|ref|NG_046987.1| Homo sapiens DEAD-box helicase 10 (DDX10), RefSeqGene on chromosome 11\n",
      "lenght: 282906\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTG...\n",
      "||||||||||||||||||||||||||| |||| |||||||||||| ||||||||| |||| ||||||||||||||...\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCTCTCTTGTCACCCAGGCTGGAGTGCAATGGTGCGATCTTGGCTCACTG...\n",
      "\n",
      "gi|31790682|dbj|AP002453.4| Homo sapiens genomic DNA, chromosome 11 clone:RP11-801G16, complete sequence\n",
      "           Length = 211345\n",
      "\n",
      "****Alignment****\n",
      "sequence:  gi|31790682|dbj|AP002453.4| Homo sapiens genomic DNA, chromosome 11 clone:RP11-801G16, complete sequence\n",
      "lenght: 211345\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCCCTCT-GTCACCCAGGCTAGAGTGCAATAGTGCCATCTTGGCTCACTG...\n",
      "||||||||||||||||||||||||||| |||| |||||||||||| ||||||||| |||| ||||||||||||||...\n",
      "TTTTTTTTTTTTTTGAGACAGAGTCTCTCTCTTGTCACCCAGGCTGGAGTGCAATGGTGCGATCTTGGCTCACTG...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "E_VALUE_THRESH = 0.001\n",
    "results_Blast= open('blast_DNA_GLI2_file.xml')\n",
    "blast_records = NCBIXML.read(results_Blast)\n",
    "for alignment in  blast_records.alignments:\n",
    "    print (alignment)\n",
    "    for hsp in alignment.hsps:\n",
    "        if hsp.expect < E_VALUE_THRESH:\n",
    "            print('****Alignment****')\n",
    "            print('sequence: ', alignment.title)\n",
    "            print('lenght:', alignment.length)\n",
    "            print(hsp.query[0:75] + '...')\n",
    "            print(hsp.match[0:75] + '...')\n",
    "            print(hsp.sbjct[0:75] + '...')\n",
    "            print()\n",
    "            break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "f84107f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'AB007295', 'NG_009030', 'AB007296', 'AC108019', 'NG_046987', 'AC004262', 'NM_001354259', 'NG_096463', 'AC008870', 'CP068260', 'AB209354', 'AC015878', 'AB007297', 'CP068259', 'D14828', 'AC279654', 'NM_001371271', 'NM_001374354', 'AC016764', 'BX295541', 'CP068262', 'AP002453', 'AP023476', 'AH001472', 'NG_047146', 'AL356212', 'D14827', 'AC008983', 'NM_005270', 'AC011774', 'BC111410', 'NM_001374353', 'AL611925', 'NG_016599', 'NG_011776', 'DQ086814', 'AP023478', 'AP023479', 'AL121658', 'AC012317', 'AB007298', 'AC090427', 'AP001972', 'DQ314865', 'NG_028118', 'AC279566'}\n",
      "Total 16 PREDICTED seqs found.\n"
     ]
    }
   ],
   "source": [
    "#filtering the \"predicted\" alignments\n",
    "results_Blastn= open('blast_DNA_GLI2_file.xml')\n",
    "blastn_records = NCBIXML.read(results_Blastn)\n",
    "E_VALUE_THRESH = 0.001\n",
    "count_preditc, count_homo=0,0\n",
    "list_filtered_alignments=[]\n",
    "for alignment in  blastn_records.alignments:\n",
    "    for hsp in alignment.hsps:\n",
    "        if hsp.expect < E_VALUE_THRESH:\n",
    "            exist = re.search(r'PREDICTED:\\s', alignment.title)\n",
    "            if exist:\n",
    "                # print( existe[0] )\n",
    "                pre = re.match(r'PREDICTED:\\s', exist[0] )\n",
    "                if pre:\n",
    "                    count_preditc+=1\n",
    "                    #print(id)\n",
    "            else:\n",
    "                list_filtered_alignments.append(alignment.accession)\n",
    "                homo=re.search(r'Homo\\ssapiens',alignment.title)\n",
    "                if homo:\n",
    "                    count_homo+=1\n",
    "print(set(list_filtered_alignments))\n",
    "print('Total {} PREDICTED seqs found.'.format(count_preditc))\n",
    "#counting of Homo sapiens it's not conclusive, just to have an ideia, because the titles don't follow any pattern"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "23c62a78",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('CDS_GLI2_nucleotide_result_blast.txt', 'w') as f:\n",
    "    for line in list_filtered_alignments:\n",
    "        f.write(f\"{line}\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d499823c",
   "metadata": {},
   "source": [
    "## BLASTP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "941da015",
   "metadata": {},
   "outputs": [],
   "source": [
    "record = SeqIO.read(open(\"10001bp_sequence_GLI2.gb\"), format=\"gb\") \n",
    "print (len(record.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1650e800",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['METSASATASEKQEAKSGILEAAGFPDPGKKASPLVVAAAAAAAVAAQGVPQHLLPPFHAPLPIDMRHQEGRYHYEPHSVHGVHGPPALSGSPVISDISLIRLSPHPAGPGESPFNAPHPYVNPHMEHYLRSVHSSPTLSMISAARGLSPADVAQEHLKERGLFGLPAPGTTPSDYYHQMTLVAGHPAPYGDLLMQSGGAASAPHLHDYLNPVDVSRFSSPRVTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGSYGHLSAGALSPAFTFPHPINPVAYQQILSQQRGLGSAFGHTPPLIQPSPTFLAQQPMALTSINATPTQLSSSSNCLSDTNQNKQSSESAVSSTVNPVAIHKRSKVKTEPEGLRPASPLALTQEQLADLKEDLDRDDCKQEAEVVIYETNCHWEDCTKEYDTQEQLVHHINNEHIHGEKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRLENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKIPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHLRTPLLKENGDSEAGTEPGGPESTEASSTSQAVEDCLHVRAIKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGVEMPGTGPGSLGDLTALDDTPPGADTSALAAPSAGGLQLRKHMTTMHRFEQLKKEKLKSLKDSCSWAGPTPHTRNTKLPPLPGSGSILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDSSTSTVSSAYTVSRRSSGISPYFSSRRSSEASPLGAGRPHNASSADSYDPISTDASRRSSEASQCSGGSGLLNLTPAQQYSLRAKYAAATGGPPPTPLPGLERMSLRTRLALLDAPERTLPAGCPRPLGPRRGSDGPTYGHGHAGAAPAFPHEAPGGGARRASDPVRRPDALSLPRVQRFHSTHNVNPGPLPPCADRRGLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAGPEADLGLPEDDLVLPDDVVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLPSPGLHGQRRMVAADSNVGPSAPMLGGCQLGFGAPSSLNKNNMPVQWNEVSSGTVDALASQVKPPPFPQGNLAVVQQKPAFGQYPGYSPQGLQASPGGLDSTQPHLQPRSGAPSQGIPRVNYMQQLRQPVAGSQCPGMTTTMSPHACYGQVHPQLSPSTISGALNQFPQSCSNMPAKPGHLGHPQQTEVAPDPTTMGNRHRELGVPDSALAGVPPPHPVQSYPQQSHHLAASMSQEGYHQVPSLLPARQPGFMEPQTGPMGVATAGFGLVQPRPPLEPSPTGRHRGVRAVQQQLAYARATGHAMAAMPSSQETAEAVPKGAMGNMGSVPPQPPPQDAGGAPDHSMLYYYGQIHMYEQDGGLENLGSCQVMRSQPPQPQACQDSIQPQPLPSPGVNQVSSTVDSQLLEAPQIDFDAIMDDGDHSSLFSGALSPSLLHSLSQNSSRLTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT']\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "Trans=[]\n",
    "record = SeqIO.read(open(\"10001bp_sequence_GLI2.gb\"), format=\"gb\") \n",
    "for feat in record.features:\n",
    "    if feat.type == \"CDS\":\n",
    "        Trans.append(str(feat.qualifiers['translation']))\n",
    "        print(feat.qualifiers['translation'])\n",
    "        print(len(feat.qualifiers['translation']))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "011697bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1569"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq_prot='METSASATASEKQEAKSGILEAAGFPDPGKKASPLVVAAAAAAAVAAQGVPQHLLPPFHAPLPIDMRHQEGRYHYEPHSVHGVHGPPALSGSPVISDISLIRLSPHPAGPGESPFNAPHPYVNPHMEHYLRSVHSSPTLSMISAARGLSPADVAQEHLKERGLFGLPAPGTTPSDYYHQMTLVAGHPAPYGDLLMQSGGAASAPHLHDYLNPVDVSRFSSPRVTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGSYGHLSAGALSPAFTFPHPINPVAYQQILSQQRGLGSAFGHTPPLIQPSPTFLAQQPMALTSINATPTQLSSSSNCLSDTNQNKQSSESAVSSTVNPVAIHKRSKVKTEPEGLRPASPLALTQEQLADLKEDLDRDDCKQEAEVVIYETNCHWEDCTKEYDTQEQLVHHINNEHIHGEKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRLENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKIPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHLRTPLLKENGDSEAGTEPGGPESTEASSTSQAVEDCLHVRAIKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGVEMPGTGPGSLGDLTALDDTPPGADTSALAAPSAGGLQLRKHMTTMHRFEQLKKEKLKSLKDSCSWAGPTPHTRNTKLPPLPGSGSILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDSSTSTVSSAYTVSRRSSGISPYFSSRRSSEASPLGAGRPHNASSADSYDPISTDASRRSSEASQCSGGSGLLNLTPAQQYSLRAKYAAATGGPPPTPLPGLERMSLRTRLALLDAPERTLPAGCPRPLGPRRGSDGPTYGHGHAGAAPAFPHEAPGGGARRASDPVRRPDALSLPRVQRFHSTHNVNPGPLPPCADRRGLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAGPEADLGLPEDDLVLPDDVVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLPSPGLHGQRRMVAADSNVGPSAPMLGGCQLGFGAPSSLNKNNMPVQWNEVSSGTVDALASQVKPPPFPQGNLAVVQQKPAFGQYPGYSPQGLQASPGGLDSTQPHLQPRSGAPSQGIPRVNYMQQLRQPVAGSQCPGMTTTMSPHACYGQVHPQLSPSTISGALNQFPQSCSNMPAKPGHLGHPQQTEVAPDPTTMGNRHRELGVPDSALAGVPPPHPVQSYPQQSHHLAASMSQEGYHQVPSLLPARQPGFMEPQTGPMGVATAGFGLVQPRPPLEPSPTGRHRGVRAVQQQLAYARATGHAMAAMPSSQETAEAVPKGAMGNMGSVPPQPPPQDAGGAPDHSMLYYYGQIHMYEQDGGLENLGSCQVMRSQPPQPQACQDSIQPQPLPSPGVNQVSSTVDSQLLEAPQIDFDAIMDDGDHSSLFSGALSPSLLHSLSQNSSRLTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT'\n",
    "len(seq_prot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07673de5",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle = NCBIWWW.qblast(\"blastp\", \"swissprot\", Trans)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d69486f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('blast_PROT_GLI2_file.xml', \"w\") as out_handle:\n",
    "        out_handle.write(result_handle.read())\n",
    "result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0681d30d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Accession:  P10070\n",
      "Definition:  RecName: Full=Zinc finger protein GLI2; AltName: Full=GLI family zinc finger protein 2; AltName: Full=Tax helper protein [Homo sapiens]\n",
      "E-value:  0.0\n",
      "\n",
      "Accession:  Q0VGT2\n",
      "Definition:  RecName: Full=Zinc finger protein GLI2; AltName: Full=Tax helper protein [Mus musculus]\n",
      "E-value:  0.0\n",
      "\n",
      "Accession:  Q91661\n",
      "Definition:  RecName: Full=Zinc finger protein GLI4; AltName: Full=Neural-specific DNA-binding protein xGLI4; Short=xGLI-4 [Xenopus laevis]\n",
      "E-value:  0.0\n",
      "\n",
      "Accession:  Q9IA31\n",
      "Definition:  RecName: Full=Transcriptional activator GLI3; AltName: Full=GLI3 full-length protein; Short=GLI3FL; Contains: RecName: Full=Transcriptional repressor GLI3R; AltName: Full=GLI3 C-terminally truncated form [Gallus gallus]\n",
      "E-value:  0.0\n",
      "E-value:  1.2649e-18\n",
      "\n",
      "Accession:  P55879\n",
      "Definition:  RecName: Full=Zinc finger protein GLI2 [Gallus gallus]\n",
      "E-value:  0.0\n"
     ]
    }
   ],
   "source": [
    "results_Blastp = open(\"blast_PROT_GLI2_file.xml\")\n",
    "blastp_records = NCBIXML.read(results_Blastp)\n",
    "for parameter in blastp_records.alignments[0:5]:\n",
    "    print()\n",
    "    print('Accession: ', parameter.accession)\n",
    "    print('Definition: ', parameter.hit_def)    \n",
    "    for e_v in parameter.hsps:\n",
    "        print('E-value: ', e_v.expect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ad58f306",
   "metadata": {},
   "outputs": [],
   "source": [
    "#results_Blastp= open('blast_PROT_GLI2_file.xml')\n",
    "#blastp_records = NCBIXML.read(results_Blastp)\n",
    "#E_VALUE_THRESH = 0.001\n",
    "#list_filtered_alignments,list_species=[],[]\n",
    "#for alignment in  blastp_records.alignments:\n",
    "    #for hsp in alignment.hsps:\n",
    "#         print(hsp.identities)    # maybe add more \n",
    "        #if hsp.expect < E_VALUE_THRESH:\n",
    "            #list_filtered_alignments.append(alignment.accession)\n",
    "            #title_organism=re.search(r'\\[.+\\s.+\\]', alignment.title)\n",
    "            #if title_organism:\n",
    "                #m = re.match(r'\\[.+\\s.+\\]', title_organism[0] )\n",
    "                #specie = m.group(0)\n",
    "                #print(specie)\n",
    "                #list_species.append(specie)\n",
    "#for x in sorted(set(list_species)):\n",
    "    #print(\"number of times: {} that appeared specie: {}\".format(list_species.count(x),x))\n",
    "\n",
    "#print(list_filtered_alignments)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "278d5245",
   "metadata": {},
   "outputs": [],
   "source": [
    "#with open('CDS_GLI2_protein_result_blastp.txt', 'w') as f:\n",
    "    #for line in list_filtered_alignments:\n",
    "        #f.write(f\"{line}\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "4ec82e84",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1586\n",
      "__Thresholds__\n",
      "coverage: 40%   |\n",
      "identity: 40%   |\n",
      "e-value: 0.001% |\n",
      "\n",
      "   ID     %coverage  %identity    Organism\n",
      "P10070      100          99       Homo sapiens\n",
      "Q0VGT2       97          83       Mus musculus\n",
      "Q91661       86          55     Xenopus laevis\n",
      "Q9IA31       67          73      Gallus gallus\n",
      "P55879       41          79      Gallus gallus\n",
      "Q91660       70          67      Gallus gallus\n",
      "P10071       67          68     Xenopus laevis\n",
      "Q5IS56       67          68     Xenopus laevis\n",
      "Q61602       66          69       Homo sapiens\n",
      "Q91690       57          46       Homo sapiens\n",
      "P08151       44          42    Pan troglodytes\n"
     ]
    }
   ],
   "source": [
    "#has to be apart because of NCBIXML.read() and not parse\n",
    "\n",
    "filtragem= []\n",
    "results_Blastp= open('blast_PROT_GLI2_file.xml')\n",
    "blastp_records_r = NCBIXML.read(results_Blastp)\n",
    "first=blastp_records_r.alignments[0]\n",
    "len_max_seq=first.hsps[0].align_length\n",
    "print(len_max_seq)\n",
    "results_Blastp= open('blast_PROT_GLI2_file.xml')\n",
    "blastp_records = NCBIXML.parse(results_Blastp)\n",
    "list_accession,list_identities,list_coverage_start, list_coverage_stop,list_alignLen=[],[],[],[],[]\n",
    "list_organism=[]\n",
    "E_VALUE_THRESH = 0.001\n",
    "for alignment in  blastp_records:\n",
    "    for a in alignment.alignments:\n",
    "        for b in a.hsps:\n",
    "            if b.expect < E_VALUE_THRESH:\n",
    "                title_organism=re.search(r'\\[.+\\s.+\\]', a.title)\n",
    "                if title_organism:\n",
    "                    m = re.match(r'\\[(.+\\s.+)\\]', title_organism[0] )\n",
    "                    list_accession.append(a.accession)\n",
    "                    list_coverage_start.append(b.sbjct_start)\n",
    "                    list_coverage_stop.append(b.sbjct_end)\n",
    "                    list_identities.append(b.identities)\n",
    "                    list_alignLen.append(b.align_length)\n",
    "                    list_organism.append(m.group(1))\n",
    "            #print(a.accession, b.align_length, b.identities, b.expect, b.sbjct_start, b.sbjct_end)\n",
    "            \n",
    "#calculation of %identity                        \n",
    "def identity(list_i, list_a):\n",
    "    perc_list_identities=[]\n",
    "    for index in range(len(list_i)):\n",
    "        perc_list_identities.append(round(list_i[index]*100/list_a[index]))\n",
    "    return perc_list_identities\n",
    "\n",
    "#calculation of the %coverage - for all hsps of all accessions\n",
    "def coverage(list_end,list_start):\n",
    "    perc_list_coverage=[]\n",
    "    for index in range(len(list_identities)):\n",
    "        perc_list_coverage.append(round((1+list_end[index]-list_start[index])/len_max_seq*100))\n",
    "    return perc_list_coverage\n",
    "\n",
    "#selecting the max coverage and max identity per hsps/accession\n",
    "def max_values_by_id(ids, covs, idents): \n",
    "    unique_ids = []\n",
    "    max_coverage = []\n",
    "    max_identity = []\n",
    "    for id, cov, ident in zip(ids, covs,idents):\n",
    "        if id in unique_ids:\n",
    "            idx = unique_ids.index(id)\n",
    "            max_coverage[idx] = max(max_coverage[idx], cov)\n",
    "            max_identity[idx] = max(max_identity[idx], ident)\n",
    "        else:\n",
    "            unique_ids.append(id)\n",
    "            max_coverage.append(cov)\n",
    "            max_identity.append(ident)\n",
    "    return unique_ids, max_coverage, max_identity\n",
    "\n",
    "beta_max=max_values_by_id(list_accession, coverage(list_coverage_stop, list_coverage_start),identity(list_identities,list_alignLen))\n",
    "\n",
    "thresold_coverage=40\n",
    "thresold_identity=40\n",
    "print('__Thresholds__')\n",
    "print('coverage: {}%   |'.format(thresold_coverage))\n",
    "print('identity: {}%   |'.format(thresold_identity))\n",
    "print('e-value: {}% |\\n'.format(E_VALUE_THRESH))\n",
    " \n",
    "print('{:>5}{:>14}{:>11}{:>12}'.format('ID','%coverage','%identity', 'Organism'))\n",
    "count=0\n",
    "filtered_organism=[]\n",
    "    for id, cov, ident, specie in zip(beta_max[0], beta_max[1], beta_max[2], list_organism):\n",
    "        if int(cov)>thresold_coverage and ident>thresold_identity:\n",
    "            if int(cov)>100:\n",
    "                print('{:>5}{:>9}*{:>11}  {:>20}'.format(id, cov, ident, specie))\n",
    "                count+=1\n",
    "            else:\n",
    "                print('{:>5}{:>9}{:>12}  {:>17}'.format(id, cov, ident, specie))\n",
    "#                     f.write(f\"{id}\\n\")\n",
    "            filtragem.append(f\"{id} {specie}\")\n",
    "            \n",
    "if count>0:\n",
    "    print('\\n* means that the subject sequence is longer that the query sequence')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "70a8278e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['P10070 Homo sapiens', 'Q0VGT2 Mus musculus', 'Q91661 Xenopus laevis', 'Q9IA31 Gallus gallus', 'P08151 Pan troglodytes']\n"
     ]
    }
   ],
   "source": [
    "lista_sem_repetidos= []\n",
    "seen = set()\n",
    "for x in filtragem:\n",
    "    parts = x.split()\n",
    "    organi = parts[1] + ' ' + parts[2]\n",
    "    if organi in seen:\n",
    "        filtragem.remove(x)\n",
    "    else:\n",
    "        seen.add(organi)\n",
    "        lista_sem_repetidos.append(x)\n",
    "        \n",
    "print(lista_sem_repetidos)\n",
    "\n",
    "lista_ids = []\n",
    "with open('filtered_protein_result_blastp.txt', 'w') as f:\n",
    "    for x in lista_sem_repetidos:\n",
    "        IDS = x.split()\n",
    "        lista_ids.append(IDS[0])\n",
    "        f.write(f\"{IDS[0]}\\n\")\n",
    "print(lista_ids)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "423dd7ad",
   "metadata": {},
   "source": [
    "## Uniprot search of Blastp results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "524fe7c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_url(url, **kwargs):\n",
    "    response = requests.get(url, **kwargs);\n",
    "\n",
    "    if not response.ok:\n",
    "        print(response.text)\n",
    "        response.raise_for_status()\n",
    "        sys.exit()\n",
    "\n",
    "    return response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "2930675b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1106\n",
      "[1586, 1544, 1361, 1544, 1106]\n"
     ]
    }
   ],
   "source": [
    "lista_tamanho=[]\n",
    "for x in lista_ids:\n",
    "    with ExPASy.get_sprot_raw(x) as handle:\n",
    "        seq_record = SeqIO.read(handle, \"swiss\")\n",
    "        lista_tamanho.append(len(seq_record.seq))\n",
    "\n",
    "print (lista_tamanho)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "71f6228d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GLI2_HUMAN', 'GLI2_MOUSE', 'GLI4_XENLA', 'GLI3_CHICK', 'GLI1_HUMAN']\n"
     ]
    }
   ],
   "source": [
    "lista_entry_name=[]\n",
    "\n",
    "for x in lista_ids:\n",
    "    with ExPASy.get_sprot_raw(x) as handle:\n",
    "        seq_record = SwissProt.read(handle)\n",
    "        b= (seq_record.entry_name)\n",
    "        lista_entry_name.append(b)\n",
    "print(lista_entry_name)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "d0094dc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('BLASTp_new.fasta', 'w') as f:\n",
    "    for x in lista_ids:\n",
    "        with ExPASy.get_sprot_raw(x) as handle:\n",
    "            seq_record = SeqIO.read(handle, \"swiss\")\n",
    "            #print(f'>{seq_record.id}')\n",
    "            #print(seq_record.seq)\n",
    "            f.write(f'>{seq_record.id}\\n{seq_record.seq}\\n\\n')\n",
    "            \n",
    "\n",
    "           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "2d8b6440",
   "metadata": {},
   "outputs": [],
   "source": [
    "#results_Blast= open('blast_PROT_GLI2_file.xml')\n",
    "#blast_records = NCBIXML.read(results_Blast)\n",
    "#for alignment in  blast_records.alignments:\n",
    "    #print (alignment.hit_id)\n",
    "    #m = alignment.hit_id.replace(\"sp|\",\"\")\n",
    "    #a=m[0:6]\n",
    "    #print(f\"id:{a}\")\n",
    "    #with ExPASy.get_sprot_raw(a) as handle:\n",
    "        #seq_record = SwissProt.read(handle)\n",
    "        #print(seq_record.organism)\n",
    "    #with ExPASy.get_sprot_raw(a) as handle:\n",
    "        #seq_record = SeqIO.read(handle, \"swiss\")\n",
    "        #print(f'>{seq_record.id}')\n",
    "        #print(seq_record.seq)\n",
    "        #print()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "debf7409",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> GLI2_HUMAN \n",
      " RecName: Full=Zinc finger protein GLI2 {ECO:0000305}; AltName: Full=GLI family zinc finger protein 2 {ECO:0000312|HGNC:HGNC:4318}; AltName: Full=Tax helper protein {ECO:0000303|PubMed:9557682}; \n",
      " ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "keywords:['Acetylation', 'Activator', 'Alternative splicing', 'Cell projection', 'Cilium', 'Cytoplasm', 'Developmental protein', 'Disease variant', 'DNA-binding', 'Holoprosencephaly', 'Isopeptide bond', 'Metal-binding', 'Nucleus', 'Phosphoprotein', 'Reference proteome', 'Repeat', 'Repressor', 'Transcription', 'Transcription regulation', 'Ubl conjugation', 'Zinc', 'Zinc-finger']\n",
      "\n",
      "> GLI2_MOUSE \n",
      " RecName: Full=Zinc finger protein GLI2 {ECO:0000305}; AltName: Full=Tax helper protein {ECO:0000250|UniProtKB:P10070}; \n",
      " ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Glires', 'Rodentia', 'Myomorpha', 'Muroidea', 'Muridae', 'Murinae', 'Mus', 'Mus']\n",
      "keywords:['Acetylation', 'Activator', 'Cell projection', 'Cilium', 'Cytoplasm', 'Developmental protein', 'DNA-binding', 'Metal-binding', 'Nucleus', 'Phosphoprotein', 'Reference proteome', 'Repeat', 'Repressor', 'Transcription', 'Transcription regulation', 'Zinc', 'Zinc-finger']\n",
      "\n",
      "> GLI4_XENLA \n",
      " RecName: Full=Zinc finger protein GLI4; AltName: Full=Neural-specific DNA-binding protein xGLI4; Short=xGLI-4; \n",
      " ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Amphibia', 'Batrachia', 'Anura', 'Pipoidea', 'Pipidae', 'Xenopodinae', 'Xenopus', 'Xenopus']\n",
      "keywords:['DNA-binding', 'Metal-binding', 'Nucleus', 'Reference proteome', 'Repeat', 'Transcription', 'Transcription regulation', 'Zinc', 'Zinc-finger']\n",
      "\n",
      "> GLI3_CHICK \n",
      " RecName: Full=Transcriptional activator GLI3; AltName: Full=GLI3 full-length protein; Short=GLI3FL; Contains: RecName: Full=Transcriptional repressor GLI3R; AltName: Full=GLI3 C-terminally truncated form; Flags: Fragment; \n",
      " ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Archelosauria', 'Archosauria', 'Dinosauria', 'Saurischia', 'Theropoda', 'Coelurosauria', 'Aves', 'Neognathae', 'Galloanserae', 'Galliformes', 'Phasianidae', 'Phasianinae', 'Gallus']\n",
      "keywords:['Activator', 'Cytoplasm', 'DNA-binding', 'Metal-binding', 'Nucleus', 'Phosphoprotein', 'Reference proteome', 'Repeat', 'Repressor', 'Transcription', 'Transcription regulation', 'Zinc', 'Zinc-finger']\n",
      "\n",
      "> GLI1_HUMAN \n",
      " RecName: Full=Zinc finger protein GLI1; AltName: Full=Glioma-associated oncogene; AltName: Full=Oncogene GLI; \n",
      " ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "keywords:['3D-structure', 'Acetylation', 'Activator', 'Alternative splicing', 'Cytoplasm', 'Developmental protein', 'Differentiation', 'Disease variant', 'DNA-binding', 'Isopeptide bond', 'Metal-binding', 'Nucleus', 'Phosphoprotein', 'Proto-oncogene', 'Reference proteome', 'Repeat', 'Transcription', 'Transcription regulation', 'Ubl conjugation', 'Zinc', 'Zinc-finger']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for x in lista_ids:\n",
    "    with ExPASy.get_sprot_raw(x) as handle:\n",
    "        seq_record = SwissProt.read(handle)\n",
    "        print(\">\",seq_record.entry_name,\"\\n\",seq_record.description, \"\\n\", seq_record.organism_classification)\n",
    "        print(f\"keywords:{seq_record.keywords}\")\n",
    "        print()\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "381d6d11",
   "metadata": {},
   "source": [
    "## Alignment and Phylo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "cfef628d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\"C:\\Program Files (x86)\\ClustalW2\\clustalw2\" -infile=BLASTp_new.fasta\n"
     ]
    }
   ],
   "source": [
    "from Bio.Align.Applications import ClustalwCommandline\n",
    "\n",
    "dir = r'C:\\Program Files (x86)\\ClustalW2\\clustalw2'\n",
    "in_file = r'BLASTp_new.fasta'\n",
    "\n",
    "clustalw_cline = ClustalwCommandline(dir, infile=in_file)\n",
    "clustalw_cline()\n",
    "print(clustalw_cline)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "998c8ae4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ClustalwCommandline(cmd='clustalw', infile='BLASTp_new.fasta', outfile='BLASTp_new.aln')"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cline = ClustalwCommandline(\"clustalw\", infile=\"BLASTp_new.fasta\", outfile=\"BLASTp_new.aln\")\n",
    "cline\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "7f833ea1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CLUSTAL 2.1 multiple sequence alignment\n",
      "\n",
      "\n",
      "P10070                              -METSASATASEKQEAKSGILEAAGFPDPGKKASP---------------\n",
      "Q0VGT2                              -METSAPAPALEKKEAKSGLLEDSSFPDPGKKACP---------------\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              MEAQSHSSTTTEKKKVENSIVKCSNRTDVSEKAVASSTTSNEDESPGQTY\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              --------------------------------------------------\n",
      "Q0VGT2                              --------------------------------------------------\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              HRERRNAITMQPQGGQGLSKISEEPSTSSEERASLIKKEIHGSISHLPEP\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              -------LVVAAAAAAAVAAQGVPQHLLPPFHAPLPIDMRHQEGRYHYEP\n",
      "Q0VGT2                              -------LAVA----AAVAAHGVPQQLLPAFHAPLPIDMRHQEGRYHYDP\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              SVPYRGTLVTMDPRNGYMDPHYHPPHLFQAFHPPVPIDARHHEGRYHYEP\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              HSVHGVHGPPALSGSPVISDISLIRLSPH--PAGPGESPFNAPHPYVNPH\n",
      "Q0VGT2                              HSVHSVHGPPTLSGSPVISDISLIRLSPH--PAGPGESPFSAHHPYVNPH\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              SPIPPLHVPSALSSSPTYSDLPFIRISPHRNPAAASESPFSTPHPYINPY\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              MEHYLRSVHSSPTLSMISAARGLSPADVAQEHLKERGLFGLPAP---GTT\n",
      "Q0VGT2                              MEHYLRSVHSSPTLSMISAARGLSPADVAHEHLKERGLFSLAAP---GTN\n",
      "Q91661                              MEHYLRSVHNSPTLSMISAARGLSPAEVAHEHLKERGIYGLAPPPPPGTT\n",
      "Q9IA31                              MD-YIRSLHSSPSLSMISAARGLSPTDAPHA----------------GVS\n",
      "P08151                              ----MFNSMTPPPISSYGEPCCLRP------------------------L\n",
      "                                        : .  ..*.:*  . .  * *                         \n",
      "\n",
      "P10070                              PSDYYHQMTLVAGHPAPYGDLLMQSGGAASAPHLHDYLNPVDVSRFSSPR\n",
      "Q0VGT2                              PSDYYHQMTLMASHPTPYGDLLMQSGGAASAPHLHDYLNPVDASRFSSPR\n",
      "Q91661                              PTEYCHQMAFLASHASPYGELLVQSAAAGNTSHLHDYLTPMDVSRFSSPR\n",
      "Q9IA31                              PAEYYHQMALLAGQRSPYADIIPSAATAGAGALHMEYLHAMDSTRFPSPR\n",
      "P08151                              PSQGAPSVGTEGLSGPPFCHQANLMSGPHSYGPARETNSCTEGPLFSSPR\n",
      "                                    *::   .:   .   .*: .     . .       :     : . *.***\n",
      "\n",
      "P10070                              VTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGS\n",
      "Q0VGT2                              VTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGS\n",
      "Q91661                              VTPRLSRKRALSISPLSDASIDLQTMIRTSPNSLVAYINNSRSSSAASGS\n",
      "Q9IA31                              LSARPSRKRTLSISPLSDHSFDLQTMIRTSPNSLVTILNNSCSSWSASGS\n",
      "P08151                              SAVKLTKKRALSISPLSDASLDLQTVIRTSPSSLVAFINSRCTS--PGGS\n",
      "                                     : : ::**:******** *:*** :*****.***: :*.  :*  ..**\n",
      "\n",
      "P10070                              YGHLSAGALSPAFTFPHPINPVAYQ----QILSQQRGLGSAFGHTPPLIQ\n",
      "Q0VGT2                              YGHLSAGALSPAFTFPHPINPVAYQ----QILSQQRGLGSAFGHTPPLIQ\n",
      "Q91661                              YGHLSAGAISPAFSFPIHK-PCSLS----AALSQQRSLSSSFGHTP-LLH\n",
      "Q9IA31                              YGHLSASAISPALSFTYPPTPVSLQQMHQQIISRQQTLGSAFGHSPPLIH\n",
      "P08151                              YGHLSIGTMSPSLGFPAQMNHQKGP-------------------------\n",
      "                                    ***** .::**:: *.                                  \n",
      "\n",
      "P10070                              PSPTFLAQQPMALTSINATPTQLSSSS-NCLSDTNQNKQSSESAVSSTVN\n",
      "Q0VGT2                              PSPTFLAQQPMTLTSISTMPTQLSSSSSNCLNDANQNKQNSESAVSSTVN\n",
      "Q91661                              PSPTFASRQQGALTSANPAPPSNNSSA----PDSVLNKVSSESAVSSTVN\n",
      "Q9IA31                              PAPTFPTQRPIPGIPSVLNPVQVSSGP---SESTQHNKPTSESAVSSTGD\n",
      "P08151                              -SPSFGVQPCGPHDSARGGMIPHPQSR-----------GPFPTCQLKSEL\n",
      "                                     :*:*  :   .  .         ..                :.  .:  \n",
      "\n",
      "P10070                              PVAIHKRSKVKTEPEGLRPASPLALTQGQVSGHGSCGCALPLSQEQLADL\n",
      "Q0VGT2                              PITIHKRSKVKTEAEGLRPASPLGLTQ-----------------EQLADL\n",
      "Q91661                              -QVIHKRSKVKTEEE----ADSVRFPQ---------------PPDHLTDL\n",
      "Q9IA31                              -PMHNKRSKIKPDEDLPSPGAGSVQEQ----------------PEGMTPV\n",
      "P08151                              DMLVGKCREEPLEGDMSSPNSTGIQDP----------------LLGMLDG\n",
      "                                         *  :   : :                               :   \n",
      "\n",
      "P10070                              KEDLDRDDCKQEAEVVIYETNCHWEDCTKEYDTQEQLVHHINNEHIHGEK\n",
      "Q0VGT2                              KEDLDRDDCKQEAEVVIYETNCHWADCTKEYDTQEQLVHHINNEHIHGEK\n",
      "Q91661                              KEDLDKDECKQEPE-HIYETNCHWDGCSKEFDTQDQLVHHINNDHIHGEK\n",
      "Q9IA31                              KEEGDKDESKQEPE-VVYETNCHWEGCSREFDTQEQLVHHINNDHIHGEK\n",
      "P08151                              REDLEREE-KREPE-SVYETDCRWDGCSQEFDSQEQLVHHINSEHIHGER\n",
      "                                    :*: :::: *:*.*  :***:*:* .*::*:*:*:*******.:*****:\n",
      "\n",
      "P10070                              KEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRL\n",
      "Q0VGT2                              KEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRL\n",
      "Q91661                              KEFVCRWQDCSREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCFKAYSRL\n",
      "Q9IA31                              KEFVCRWLDCSREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCTKAYSRL\n",
      "P08151                              KEFVCHWGGCSRELRPFKAQYMLVVHMRRHTGEKPHKCTFEGCRKSYSRL\n",
      "                                    *****:*  *:** :**************************** *:****\n",
      "\n",
      "P10070                              ENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICK\n",
      "Q0VGT2                              ENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICK\n",
      "Q91661                              ENLKTHLRSHTGEKPYVCDHEGCNKAFSNASDRAKHQNRTHSNEKPYICK\n",
      "Q9IA31                              ENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYVCK\n",
      "P08151                              ENLKTHLRSHTGEKPYMCEHEGCSKAFSNASDRAKHQNRTHSNEKPYVCK\n",
      "                                    ****************:*:****.***********************:**\n",
      "\n",
      "P10070                              IPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHLRTPLLKENGDSE\n",
      "Q0VGT2                              IPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHVRAPLLKENGDNE\n",
      "Q91661                              VPGCTKRYTDPSSLRKHVKTVHGPEAHVTKKHRNDIIQKPSLPKENGDNE\n",
      "Q9IA31                              IPGCTKRYTDPSSLRKHVKTVHGPEAHVTKKQRGDIHPRPPPPRDPGSHS\n",
      "P08151                              LPGCTKRYTDPSSLRKHVKTVHGPDAHVTKRHRGDGPLPRAPSISTVEPK\n",
      "                                    :***********************:*****::*.*     .   .  . .\n",
      "\n",
      "P10070                              AGTEPGG------PESTEASSTSQAVEDCLHVRAIKTESSGLCQSSPGAQ\n",
      "Q0VGT2                              ASAEPGGRGP---EESVEASSTSHTVEDCLHIKAIKTESSGLCQSSPGAQ\n",
      "Q91661                              ASAKLSG--------REHSDSVSRDQEHCLQTRTIKTEDNMMHQSSPGGQ\n",
      "Q9IA31                              QTRSPGQQTQGATGEQKDLNSTTSRREECLQVKAVKSEKPMTSQPSPGGQ\n",
      "P08151                              REREGGPIRE--------------------ESRLTVPEGAMKPQPSPGAQ\n",
      "                                       . .                        . :   .*     *.***.*\n",
      "\n",
      "P10070                              SSCSSEPSPLGSAPNNDSGVEMPGTGPGSLGDLTALDDTPPGADTSALAA\n",
      "Q0VGT2                              SSCSSEPSPLGSAPNNDSGMEMPGTGPGSLGDLTALADTCPGADTSALAA\n",
      "Q91661                              SSCSSEPSPYGNTNNIDSGVDVSLAWQGSLGDLFGLEETSPVVDSTVSSW\n",
      "Q9IA31                              STCSSEQSPISNYSNN--GIELTLTGGGSVGDLSVIDET-PIMDSTISTA\n",
      "P08151                              SSCSSDHSPAGSAANTDSGVEMTGNAGGSTEDLSSLDEGPCIAG------\n",
      "                                    *:***: ** ..  *   *:::.    **  **  : :     .      \n",
      "\n",
      "P10070                              PSAGGLQLRKHMTTMHRFEQLKKEK--LKSLKDSCSWAGPTPHTRNTKLP\n",
      "Q0VGT2                              PSTGGLQLRKHMSTVHRFEQLKREK--LKSLKDSCSWAGPAPHTRNTKLP\n",
      "Q91661                              QRSG-RPATPETQRIHSAETGTAEEREIKDNERFLLIYEPNATCQNTRLP\n",
      "Q9IA31                              TTALGLQARRNMTGTKWMEQVKLER--LKQVNGMLPRLNPVPPSKAPTLP\n",
      "P08151                              ---------TGLSTLRRLENLRLDQ---------LHQLRPIG-TRGLKLP\n",
      "                                                   :  *    :.              *    :   **\n",
      "\n",
      "P10070                              PLPGSGSILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDSS\n",
      "Q0VGT2                              PLPVNGSVLENFNNTGGGGPAGLLPSQRLPEL--TEVTMLSQLQERRDSS\n",
      "Q91661                              TISANG-----FDVIG--VPSSVLINPRAIELSMNDVTMMNQLNERRDST\n",
      "Q9IA31                              PLIGNG----TQSNSSCSVGGSMTILPNRNELSSTDITVLNMLN-RRDSN\n",
      "P08151                              SLSHTG-------------------------TTVSRRVGPPVSLERRSSS\n",
      "                                    .:  .*                            .  .       **.*.\n",
      "\n",
      "P10070                              TSTVSSAYTVSRRSSGISPYFSSRRSSEASPLGAGRPHNASSADSYDPIS\n",
      "Q0VGT2                              TSTMSSAYTVSRRSSGISPYFSSRRSSEASPLGGLRPHNASSADSYDPIS\n",
      "Q91661                              SSTLSSAYTS-RRSSGISPYFSSRRSSETSQFGG-RLNNSSSADSYDPIS\n",
      "Q9IA31                              TSTISSAYLSSRRSSGISPCFSSRRSSDASQAEG-RPQNVSVADSYDPIS\n",
      "P08151                              SSSISSAYTVSRRSSLASPFPP----------------------------\n",
      "                                    :*::****   ****  **  .                            \n",
      "\n",
      "P10070                              TDASRRSSEASQCSGG-SGLLNLTPAQQYSLRAKYAAATGG-PPPTPLPG\n",
      "Q0VGT2                              TDASRRSSEASQCSGGGPGLLNLTPAQQYNLRAKYAAATGG-PPPTPLPG\n",
      "Q91661                              TDASRRSSEASQHSGL-PNLLNLTPAQHYRLKAKYAAATGG-PPPTPLPN\n",
      "Q9IA31                              TDASRRSSEASQCDDL-PSLLSLTPAQQYRLKAKYAAATGG-PPPTPLPN\n",
      "P08151                              ---------GSPPENGASSLPGLMPAQHYLLRARYASARGGGTSPTAASS\n",
      "                                             .*  ..  ..* .* ***:* *:*:**:* ** ..**. ..\n",
      "\n",
      "P10070                              LERMSLRTRLALLDAPERTLPAGCPRPLGPRRGSDGPTYGHGH----AGA\n",
      "Q0VGT2                              LDRVSLRTRLALLDAPERALPGACPHPLGPRRGSDGPTYSHGHGHGYAGA\n",
      "Q91661                              MDRIGLRNKLSLMDGADFPLPPFRQLPV-PRRCSDGGGN--------AGL\n",
      "Q9IA31                              MERMSLKTRMALLGDCRESGISPLPPVNAPRRCSDGGANG-------YSR\n",
      "P08151                              LDRIGGLPMPPWRSRAEYP-------------------------------\n",
      "                                    ::*:.     .  .    .                               \n",
      "\n",
      "P10070                              APAFPHEAPGGGARRASDPVRRP---DALSLPRVQRFHSTHNVNPGPLPP\n",
      "Q0VGT2                              APAFPHEGPNSSTRRASDPVRRP---DPLILPRVQRFHSTHNMNPGSLPP\n",
      "Q91661                              TPMYPHEIPGNNSRRASDPVRRTAGIDDKPLPRFSRFHSMNSMNTLHPPS\n",
      "Q9IA31                              RHLLSHDALGNGTRRASDPVRMVS--NNLSVPRVQHFNSLNSFNSPALPP\n",
      "P08151                              ----GYNPNAGVTRRASDPAQAA---DRPAPARVQRFKSLGCVHTPPTVA\n",
      "                                         ::   . :******.:     :    .*..:*:*   .:.    .\n",
      "\n",
      "P10070                              CADRRGLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAGPE\n",
      "Q0VGT2                              CADRRGLHVQSHPSVDSNLTRNAYSPRPPSINENVVMEAVAAGVDGPGLE\n",
      "Q91661                              LSERRNGGLQHYTCSDGGLHRHVYSPRPPSISENVAMEAISCDADVPG--\n",
      "Q9IA31                              SMEKRNLVLQNCTHSEGGVFRGFSSPCPPSISENVTLEAVTMEAGGSLN-\n",
      "P08151                              GGGQ---------NFDPYLPTSVYSPQPPSITENAAMDARGLQE------\n",
      "                                       :           :  :     ** ****.**..::*           \n",
      "\n",
      "P10070                              ADLGLPEDDLVLPDDVVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLP\n",
      "Q0VGT2                              CDLGLVEDELVLPDDVVQYIKAHTGGTLDDGIRQGYPTEGTGFPENSKLP\n",
      "Q91661                              -----GDDDLMLPDDVVQYIRSQNREAPEQNLQTEYSSPARNLQSNTKSF\n",
      "Q9IA31                              -------DEDLLPDDVVQYLNSQNQGTCDHLLNNVLDSN--KMHHSVVLG\n",
      "P08151                              -----------EPEVGTSMVGSGLNPYMDFPPTDTLGYGGPEGAAAEPYG\n",
      "                                                *:  .. : :      :                     \n",
      "\n",
      "P10070                              SPGLHGQRRMVAADSNVGPSAPMLGGCQLGFGAPSSLNKNNMPVQWNEVS\n",
      "Q0VGT2                              SPGLQGHRRLAAADSNMGPSAPGLGGCQLSYSPSSNLNKSNMPVQWNEVS\n",
      "Q91661                              HNNTPEQPR--APGAYLSRNFPALAECLG---QTANMQDNNMPVQWNEVS\n",
      "Q9IA31                              NNNPSSFDR--APPASSQP-------------AGSEVSKSDLPIQWNEVS\n",
      "P08151                              ARGPGSLPLGPGPPTNYGP------------------NPCPQQASYPDPT\n",
      "                                      .        .. :                      .      .: : :\n",
      "\n",
      "P10070                              SGTVDALASQVKPPPFPQGNLAVVQQKPAFGQYPGYSPQGLQASPGGLDS\n",
      "Q0VGT2                              SGTVDALPTQVKPPPFPHSNLAVVQQKPAFGQYPGYNPQSVQSSSGGLDS\n",
      "Q91661                              SGTVDVSDLPKQQ--FAAGNLAVVQQKQNFAQYQSFNQAPMQRAHNIMGQ\n",
      "Q9IA31                              SGSSDLSPTKLKC-----SQRSAVQQARAFGLYNNMMVQQQNLQRGNVYQ\n",
      "P08151                              QETWGEFPSHSGLYPGPKALGGTYSQCPRLEHYGQVQVKPEQGCP--VGS\n",
      "                                    . : .             .  .. .*   :  *        :     : .\n",
      "\n",
      "P10070                              TQPHLQPRSGAPSQGIPRVNYMQQLRQPVAGSQCPGMTTTMSPHACYGQV\n",
      "Q0VGT2                              TQPHLQLR-GAPS--ASRGSYTQQPRQPAAGSQCLGMSAAMSPQASYSQA\n",
      "Q91661                              GQESVQRNISVNG---QRFNYLQQRQQQMS--QCQIVSSDFIPQQRYSQS\n",
      "Q9IA31                              -QNGYQNLMENNG----SYSLQQNTVLGSGASSSFGMQPNKPYGESVSRQ\n",
      "P08151                              DSTGLAPCLNAHP-----------------------SEGPPHPQPLFSHY\n",
      "                                     .                                             .: \n",
      "\n",
      "P10070                              HPQLSPSTISGALNQFPQ--SCSNMPAKPGHLGHPQQTEVAPDPTTMGNR\n",
      "Q0VGT2                              HPQLSPNIVSGSLNQFSP--SCSNMAAKPSHLGLPQQMEVVPNATIMNGH\n",
      "Q91661                              QSMLSSRAMQEGQSQISP--SCNNMVERPGVHTHAAPSNTLNHQRLAVHG\n",
      "Q9IA31                              PMIFGAMDSSCGITVQGQKLRSSNMPVSGNQQNFGHPIASSDQATSMANG\n",
      "P08151                              PQPSPPQYLQSGPYTQPP---------PDYLPSEPRPCLDFDSPTHSTGQ\n",
      "                                         .   . .                                      \n",
      "\n",
      "P10070                              HRELGVPDSALAGVPPPHPVQSYPQQSHHLAASMSQEGYHQVPSLLPARQ\n",
      "Q0VGT2                              QREHGVPNSSLAAVSQPHPVLSYPQQ----------DSYQQGSNLLSSHQ\n",
      "Q91661                              APTQGFANNFSVNQDGLHPPNAYTVQ-------------PQKNGLEPQQN\n",
      "Q9IA31                              MQNRNVMEQEYLQNELVGDGIHYQGVN------------QSSPMTLGQVS\n",
      "P08151                              LKAQLVCNYVQSQQELLWEGGGREDAP---------------AQEPSYQS\n",
      "                                         . :                                         .\n",
      "\n",
      "P10070                              PGFMEPQTGPMGVATAGFGLVQPRPPLEPSPTGRHRGVRAVQQQLAYARA\n",
      "Q0VGT2                              PGFMESQQ------NAGFGLMQPRPPLEPNTASRHRGVRSGQQQL-YART\n",
      "Q91661                              TLGMSGQA-------FNHGMIQPRPPAAPHPPNRPRNIHPVHHPP-YMRS\n",
      "Q9IA31                              PTSQSSLH-------QGPQSCPPVSHTIGNQSSGLSVAKSYQPCANYSGN\n",
      "P08151                              PKFLGGSQ------------------VSPSRAKAPVNTYGPGFGPNLPNH\n",
      "                                    .                              .                  \n",
      "\n",
      "P10070                              TGHAMAAMPSSQETAEAVPKGAMGNMGSVPPQPPPQDAGGAPDHSMLYYY\n",
      "Q0VGT2                              TGQAMVTS-ANQETAEAMPKGPAGTMVSLAPQPS-QDTGRAQDQNTLYYY\n",
      "Q91661                              P-HPVSELSPGQQTAEATPKRTSENTDPTP-----------KDNNLLYYS\n",
      "Q9IA31                              RRQNVLRNNLAQQQGHVSDGNQTYRVNTIKLEMQ--GQSQQFCSNMQNYS\n",
      "P08151                              KSGSYPTPSPCHENFVVGANRASHRAAAPPRLLP----------PLPTCY\n",
      "                                               ::   .          .                      \n",
      "\n",
      "P10070                              GQIHMYEQDGGLENLGS-CQVMRSQPPQPQACQDSIQPQPLPSPGVNQVS\n",
      "Q0VGT2                              GQIHMYEQNGG-------CPAVQPQPPQPQACSDSIQPEPLPSPGVNQVS\n",
      "Q91661                              GQIHMYEPNGN--------------FGSGIDCTVRQLP-TMPSPGANQVT\n",
      "Q9IA31                              GQLYDQTTGFSHQAMKIGSSFFVSEANCLLQETATANSSELLSPGANQVS\n",
      "P08151                              GPLKVGGTNPS---------------CGHPEVGRLGGGPALYPPPEGQVC\n",
      "                                    * :     . .                             : .*  .** \n",
      "\n",
      "P10070                              STVD---SQLLE-APQIDFDAIMDDGDHSSLFSGALSPSLLHSLSQNSSR\n",
      "Q0VGT2                              STVD---SQLLE-PPQIDFDAIMDDGDHSSLFSGALSPTLLHNLSQNSSR\n",
      "Q91661                              STVD---SQGLEHPPVIDFDAMMDDGDHSSLMSGTLSPGLLQSFSQTSSR\n",
      "Q9IA31                              STVDSLDSNSLE-GVQIDFDAIIDDGDHVSLISGALSPSIIQNLSRNSSR\n",
      "P08151                              NPLD---SLDLD-NTQLDFVAILDE-------PQGLSPPPSHDQRGSSGH\n",
      "                                    ..:*   *  *:    :** *::*:       .  ***   :.   .*.:\n",
      "\n",
      "P10070                              LTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT\n",
      "Q0VGT2                              LTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT\n",
      "Q91661                              LTTPRNSLTLPSIPAGINNMAIGDMSSMLTTLAEESKFLNLMS\n",
      "Q9IA31                              LTTPRASLTFP--------------------------------\n",
      "P08151                              TPPPS----------GPPNMAVGNMSVLLRSLPGETEFLNSSA\n",
      "                                     ..*                                       \n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import AlignIO\n",
    "align = AlignIO.read(\"BLASTp_new.aln\", \"clustal\")\n",
    "\n",
    "print(format(align, \"clustal\")) #formato string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "aa9d4da5",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alignment with 5 rows and 1693 columns\n",
      "-METSASATASEKQEAKSGILEAAGFPDPGKKASP---------...MMT P10070\n",
      "-METSAPAPALEKKEAKSGLLEDSSFPDPGKKACP---------...MMT Q0VGT2\n",
      "--------------------------------------------...LMS Q91661\n",
      "MEAQSHSSTTTEKKKVENSIVKCSNRTDVSEKAVASSTTSNEDE...--- Q9IA31\n",
      "--------------------------------------------...SSA P08151\n",
      "Número de linhas: 5\n",
      "CLUSTAL 2.1 multiple sequence alignment\n",
      "\n",
      "\n",
      "P10070                              -METSASATASEKQEAKSGILEAAGFPDPGKKASP---------------\n",
      "Q0VGT2                              -METSAPAPALEKKEAKSGLLEDSSFPDPGKKACP---------------\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              MEAQSHSSTTTEKKKVENSIVKCSNRTDVSEKAVASSTTSNEDESPGQTY\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              --------------------------------------------------\n",
      "Q0VGT2                              --------------------------------------------------\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              HRERRNAITMQPQGGQGLSKISEEPSTSSEERASLIKKEIHGSISHLPEP\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              -------LVVAAAAAAAVAAQGVPQHLLPPFHAPLPIDMRHQEGRYHYEP\n",
      "Q0VGT2                              -------LAVA----AAVAAHGVPQQLLPAFHAPLPIDMRHQEGRYHYDP\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              SVPYRGTLVTMDPRNGYMDPHYHPPHLFQAFHPPVPIDARHHEGRYHYEP\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              HSVHGVHGPPALSGSPVISDISLIRLSPH--PAGPGESPFNAPHPYVNPH\n",
      "Q0VGT2                              HSVHSVHGPPTLSGSPVISDISLIRLSPH--PAGPGESPFSAHHPYVNPH\n",
      "Q91661                              --------------------------------------------------\n",
      "Q9IA31                              SPIPPLHVPSALSSSPTYSDLPFIRISPHRNPAAASESPFSTPHPYINPY\n",
      "P08151                              --------------------------------------------------\n",
      "                                                                                      \n",
      "\n",
      "P10070                              MEHYLRSVHSSPTLSMISAARGLSPADVAQEHLKERGLFGLPAP---GTT\n",
      "Q0VGT2                              MEHYLRSVHSSPTLSMISAARGLSPADVAHEHLKERGLFSLAAP---GTN\n",
      "Q91661                              MEHYLRSVHNSPTLSMISAARGLSPAEVAHEHLKERGIYGLAPPPPPGTT\n",
      "Q9IA31                              MD-YIRSLHSSPSLSMISAARGLSPTDAPHA----------------GVS\n",
      "P08151                              ----MFNSMTPPPISSYGEPCCLRP------------------------L\n",
      "                                        : .  ..*.:*  . .  * *                         \n",
      "\n",
      "P10070                              PSDYYHQMTLVAGHPAPYGDLLMQSGGAASAPHLHDYLNPVDVSRFSSPR\n",
      "Q0VGT2                              PSDYYHQMTLMASHPTPYGDLLMQSGGAASAPHLHDYLNPVDASRFSSPR\n",
      "Q91661                              PTEYCHQMAFLASHASPYGELLVQSAAAGNTSHLHDYLTPMDVSRFSSPR\n",
      "Q9IA31                              PAEYYHQMALLAGQRSPYADIIPSAATAGAGALHMEYLHAMDSTRFPSPR\n",
      "P08151                              PSQGAPSVGTEGLSGPPFCHQANLMSGPHSYGPARETNSCTEGPLFSSPR\n",
      "                                    *::   .:   .   .*: .     . .       :     : . *.***\n",
      "\n",
      "P10070                              VTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGS\n",
      "Q0VGT2                              VTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGS\n",
      "Q91661                              VTPRLSRKRALSISPLSDASIDLQTMIRTSPNSLVAYINNSRSSSAASGS\n",
      "Q9IA31                              LSARPSRKRTLSISPLSDHSFDLQTMIRTSPNSLVTILNNSCSSWSASGS\n",
      "P08151                              SAVKLTKKRALSISPLSDASLDLQTVIRTSPSSLVAFINSRCTS--PGGS\n",
      "                                     : : ::**:******** *:*** :*****.***: :*.  :*  ..**\n",
      "\n",
      "P10070                              YGHLSAGALSPAFTFPHPINPVAYQ----QILSQQRGLGSAFGHTPPLIQ\n",
      "Q0VGT2                              YGHLSAGALSPAFTFPHPINPVAYQ----QILSQQRGLGSAFGHTPPLIQ\n",
      "Q91661                              YGHLSAGAISPAFSFPIHK-PCSLS----AALSQQRSLSSSFGHTP-LLH\n",
      "Q9IA31                              YGHLSASAISPALSFTYPPTPVSLQQMHQQIISRQQTLGSAFGHSPPLIH\n",
      "P08151                              YGHLSIGTMSPSLGFPAQMNHQKGP-------------------------\n",
      "                                    ***** .::**:: *.                                  \n",
      "\n",
      "P10070                              PSPTFLAQQPMALTSINATPTQLSSSS-NCLSDTNQNKQSSESAVSSTVN\n",
      "Q0VGT2                              PSPTFLAQQPMTLTSISTMPTQLSSSSSNCLNDANQNKQNSESAVSSTVN\n",
      "Q91661                              PSPTFASRQQGALTSANPAPPSNNSSA----PDSVLNKVSSESAVSSTVN\n",
      "Q9IA31                              PAPTFPTQRPIPGIPSVLNPVQVSSGP---SESTQHNKPTSESAVSSTGD\n",
      "P08151                              -SPSFGVQPCGPHDSARGGMIPHPQSR-----------GPFPTCQLKSEL\n",
      "                                     :*:*  :   .  .         ..                :.  .:  \n",
      "\n",
      "P10070                              PVAIHKRSKVKTEPEGLRPASPLALTQGQVSGHGSCGCALPLSQEQLADL\n",
      "Q0VGT2                              PITIHKRSKVKTEAEGLRPASPLGLTQ-----------------EQLADL\n",
      "Q91661                              -QVIHKRSKVKTEEE----ADSVRFPQ---------------PPDHLTDL\n",
      "Q9IA31                              -PMHNKRSKIKPDEDLPSPGAGSVQEQ----------------PEGMTPV\n",
      "P08151                              DMLVGKCREEPLEGDMSSPNSTGIQDP----------------LLGMLDG\n",
      "                                         *  :   : :                               :   \n",
      "\n",
      "P10070                              KEDLDRDDCKQEAEVVIYETNCHWEDCTKEYDTQEQLVHHINNEHIHGEK\n",
      "Q0VGT2                              KEDLDRDDCKQEAEVVIYETNCHWADCTKEYDTQEQLVHHINNEHIHGEK\n",
      "Q91661                              KEDLDKDECKQEPE-HIYETNCHWDGCSKEFDTQDQLVHHINNDHIHGEK\n",
      "Q9IA31                              KEEGDKDESKQEPE-VVYETNCHWEGCSREFDTQEQLVHHINNDHIHGEK\n",
      "P08151                              REDLEREE-KREPE-SVYETDCRWDGCSQEFDSQEQLVHHINSEHIHGER\n",
      "                                    :*: :::: *:*.*  :***:*:* .*::*:*:*:*******.:*****:\n",
      "\n",
      "P10070                              KEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRL\n",
      "Q0VGT2                              KEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRL\n",
      "Q91661                              KEFVCRWQDCSREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCFKAYSRL\n",
      "Q9IA31                              KEFVCRWLDCSREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCTKAYSRL\n",
      "P08151                              KEFVCHWGGCSRELRPFKAQYMLVVHMRRHTGEKPHKCTFEGCRKSYSRL\n",
      "                                    *****:*  *:** :**************************** *:****\n",
      "\n",
      "P10070                              ENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICK\n",
      "Q0VGT2                              ENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICK\n",
      "Q91661                              ENLKTHLRSHTGEKPYVCDHEGCNKAFSNASDRAKHQNRTHSNEKPYICK\n",
      "Q9IA31                              ENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYVCK\n",
      "P08151                              ENLKTHLRSHTGEKPYMCEHEGCSKAFSNASDRAKHQNRTHSNEKPYVCK\n",
      "                                    ****************:*:****.***********************:**\n",
      "\n",
      "P10070                              IPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHLRTPLLKENGDSE\n",
      "Q0VGT2                              IPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHVRAPLLKENGDNE\n",
      "Q91661                              VPGCTKRYTDPSSLRKHVKTVHGPEAHVTKKHRNDIIQKPSLPKENGDNE\n",
      "Q9IA31                              IPGCTKRYTDPSSLRKHVKTVHGPEAHVTKKQRGDIHPRPPPPRDPGSHS\n",
      "P08151                              LPGCTKRYTDPSSLRKHVKTVHGPDAHVTKRHRGDGPLPRAPSISTVEPK\n",
      "                                    :***********************:*****::*.*     .   .  . .\n",
      "\n",
      "P10070                              AGTEPGG------PESTEASSTSQAVEDCLHVRAIKTESSGLCQSSPGAQ\n",
      "Q0VGT2                              ASAEPGGRGP---EESVEASSTSHTVEDCLHIKAIKTESSGLCQSSPGAQ\n",
      "Q91661                              ASAKLSG--------REHSDSVSRDQEHCLQTRTIKTEDNMMHQSSPGGQ\n",
      "Q9IA31                              QTRSPGQQTQGATGEQKDLNSTTSRREECLQVKAVKSEKPMTSQPSPGGQ\n",
      "P08151                              REREGGPIRE--------------------ESRLTVPEGAMKPQPSPGAQ\n",
      "                                       . .                        . :   .*     *.***.*\n",
      "\n",
      "P10070                              SSCSSEPSPLGSAPNNDSGVEMPGTGPGSLGDLTALDDTPPGADTSALAA\n",
      "Q0VGT2                              SSCSSEPSPLGSAPNNDSGMEMPGTGPGSLGDLTALADTCPGADTSALAA\n",
      "Q91661                              SSCSSEPSPYGNTNNIDSGVDVSLAWQGSLGDLFGLEETSPVVDSTVSSW\n",
      "Q9IA31                              STCSSEQSPISNYSNN--GIELTLTGGGSVGDLSVIDET-PIMDSTISTA\n",
      "P08151                              SSCSSDHSPAGSAANTDSGVEMTGNAGGSTEDLSSLDEGPCIAG------\n",
      "                                    *:***: ** ..  *   *:::.    **  **  : :     .      \n",
      "\n",
      "P10070                              PSAGGLQLRKHMTTMHRFEQLKKEK--LKSLKDSCSWAGPTPHTRNTKLP\n",
      "Q0VGT2                              PSTGGLQLRKHMSTVHRFEQLKREK--LKSLKDSCSWAGPAPHTRNTKLP\n",
      "Q91661                              QRSG-RPATPETQRIHSAETGTAEEREIKDNERFLLIYEPNATCQNTRLP\n",
      "Q9IA31                              TTALGLQARRNMTGTKWMEQVKLER--LKQVNGMLPRLNPVPPSKAPTLP\n",
      "P08151                              ---------TGLSTLRRLENLRLDQ---------LHQLRPIG-TRGLKLP\n",
      "                                                   :  *    :.              *    :   **\n",
      "\n",
      "P10070                              PLPGSGSILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDSS\n",
      "Q0VGT2                              PLPVNGSVLENFNNTGGGGPAGLLPSQRLPEL--TEVTMLSQLQERRDSS\n",
      "Q91661                              TISANG-----FDVIG--VPSSVLINPRAIELSMNDVTMMNQLNERRDST\n",
      "Q9IA31                              PLIGNG----TQSNSSCSVGGSMTILPNRNELSSTDITVLNMLN-RRDSN\n",
      "P08151                              SLSHTG-------------------------TTVSRRVGPPVSLERRSSS\n",
      "                                    .:  .*                            .  .       **.*.\n",
      "\n",
      "P10070                              TSTVSSAYTVSRRSSGISPYFSSRRSSEASPLGAGRPHNASSADSYDPIS\n",
      "Q0VGT2                              TSTMSSAYTVSRRSSGISPYFSSRRSSEASPLGGLRPHNASSADSYDPIS\n",
      "Q91661                              SSTLSSAYTS-RRSSGISPYFSSRRSSETSQFGG-RLNNSSSADSYDPIS\n",
      "Q9IA31                              TSTISSAYLSSRRSSGISPCFSSRRSSDASQAEG-RPQNVSVADSYDPIS\n",
      "P08151                              SSSISSAYTVSRRSSLASPFPP----------------------------\n",
      "                                    :*::****   ****  **  .                            \n",
      "\n",
      "P10070                              TDASRRSSEASQCSGG-SGLLNLTPAQQYSLRAKYAAATGG-PPPTPLPG\n",
      "Q0VGT2                              TDASRRSSEASQCSGGGPGLLNLTPAQQYNLRAKYAAATGG-PPPTPLPG\n",
      "Q91661                              TDASRRSSEASQHSGL-PNLLNLTPAQHYRLKAKYAAATGG-PPPTPLPN\n",
      "Q9IA31                              TDASRRSSEASQCDDL-PSLLSLTPAQQYRLKAKYAAATGG-PPPTPLPN\n",
      "P08151                              ---------GSPPENGASSLPGLMPAQHYLLRARYASARGGGTSPTAASS\n",
      "                                             .*  ..  ..* .* ***:* *:*:**:* ** ..**. ..\n",
      "\n",
      "P10070                              LERMSLRTRLALLDAPERTLPAGCPRPLGPRRGSDGPTYGHGH----AGA\n",
      "Q0VGT2                              LDRVSLRTRLALLDAPERALPGACPHPLGPRRGSDGPTYSHGHGHGYAGA\n",
      "Q91661                              MDRIGLRNKLSLMDGADFPLPPFRQLPV-PRRCSDGGGN--------AGL\n",
      "Q9IA31                              MERMSLKTRMALLGDCRESGISPLPPVNAPRRCSDGGANG-------YSR\n",
      "P08151                              LDRIGGLPMPPWRSRAEYP-------------------------------\n",
      "                                    ::*:.     .  .    .                               \n",
      "\n",
      "P10070                              APAFPHEAPGGGARRASDPVRRP---DALSLPRVQRFHSTHNVNPGPLPP\n",
      "Q0VGT2                              APAFPHEGPNSSTRRASDPVRRP---DPLILPRVQRFHSTHNMNPGSLPP\n",
      "Q91661                              TPMYPHEIPGNNSRRASDPVRRTAGIDDKPLPRFSRFHSMNSMNTLHPPS\n",
      "Q9IA31                              RHLLSHDALGNGTRRASDPVRMVS--NNLSVPRVQHFNSLNSFNSPALPP\n",
      "P08151                              ----GYNPNAGVTRRASDPAQAA---DRPAPARVQRFKSLGCVHTPPTVA\n",
      "                                         ::   . :******.:     :    .*..:*:*   .:.    .\n",
      "\n",
      "P10070                              CADRRGLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAGPE\n",
      "Q0VGT2                              CADRRGLHVQSHPSVDSNLTRNAYSPRPPSINENVVMEAVAAGVDGPGLE\n",
      "Q91661                              LSERRNGGLQHYTCSDGGLHRHVYSPRPPSISENVAMEAISCDADVPG--\n",
      "Q9IA31                              SMEKRNLVLQNCTHSEGGVFRGFSSPCPPSISENVTLEAVTMEAGGSLN-\n",
      "P08151                              GGGQ---------NFDPYLPTSVYSPQPPSITENAAMDARGLQE------\n",
      "                                       :           :  :     ** ****.**..::*           \n",
      "\n",
      "P10070                              ADLGLPEDDLVLPDDVVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLP\n",
      "Q0VGT2                              CDLGLVEDELVLPDDVVQYIKAHTGGTLDDGIRQGYPTEGTGFPENSKLP\n",
      "Q91661                              -----GDDDLMLPDDVVQYIRSQNREAPEQNLQTEYSSPARNLQSNTKSF\n",
      "Q9IA31                              -------DEDLLPDDVVQYLNSQNQGTCDHLLNNVLDSN--KMHHSVVLG\n",
      "P08151                              -----------EPEVGTSMVGSGLNPYMDFPPTDTLGYGGPEGAAAEPYG\n",
      "                                                *:  .. : :      :                     \n",
      "\n",
      "P10070                              SPGLHGQRRMVAADSNVGPSAPMLGGCQLGFGAPSSLNKNNMPVQWNEVS\n",
      "Q0VGT2                              SPGLQGHRRLAAADSNMGPSAPGLGGCQLSYSPSSNLNKSNMPVQWNEVS\n",
      "Q91661                              HNNTPEQPR--APGAYLSRNFPALAECLG---QTANMQDNNMPVQWNEVS\n",
      "Q9IA31                              NNNPSSFDR--APPASSQP-------------AGSEVSKSDLPIQWNEVS\n",
      "P08151                              ARGPGSLPLGPGPPTNYGP------------------NPCPQQASYPDPT\n",
      "                                      .        .. :                      .      .: : :\n",
      "\n",
      "P10070                              SGTVDALASQVKPPPFPQGNLAVVQQKPAFGQYPGYSPQGLQASPGGLDS\n",
      "Q0VGT2                              SGTVDALPTQVKPPPFPHSNLAVVQQKPAFGQYPGYNPQSVQSSSGGLDS\n",
      "Q91661                              SGTVDVSDLPKQQ--FAAGNLAVVQQKQNFAQYQSFNQAPMQRAHNIMGQ\n",
      "Q9IA31                              SGSSDLSPTKLKC-----SQRSAVQQARAFGLYNNMMVQQQNLQRGNVYQ\n",
      "P08151                              QETWGEFPSHSGLYPGPKALGGTYSQCPRLEHYGQVQVKPEQGCP--VGS\n",
      "                                    . : .             .  .. .*   :  *        :     : .\n",
      "\n",
      "P10070                              TQPHLQPRSGAPSQGIPRVNYMQQLRQPVAGSQCPGMTTTMSPHACYGQV\n",
      "Q0VGT2                              TQPHLQLR-GAPS--ASRGSYTQQPRQPAAGSQCLGMSAAMSPQASYSQA\n",
      "Q91661                              GQESVQRNISVNG---QRFNYLQQRQQQMS--QCQIVSSDFIPQQRYSQS\n",
      "Q9IA31                              -QNGYQNLMENNG----SYSLQQNTVLGSGASSSFGMQPNKPYGESVSRQ\n",
      "P08151                              DSTGLAPCLNAHP-----------------------SEGPPHPQPLFSHY\n",
      "                                     .                                             .: \n",
      "\n",
      "P10070                              HPQLSPSTISGALNQFPQ--SCSNMPAKPGHLGHPQQTEVAPDPTTMGNR\n",
      "Q0VGT2                              HPQLSPNIVSGSLNQFSP--SCSNMAAKPSHLGLPQQMEVVPNATIMNGH\n",
      "Q91661                              QSMLSSRAMQEGQSQISP--SCNNMVERPGVHTHAAPSNTLNHQRLAVHG\n",
      "Q9IA31                              PMIFGAMDSSCGITVQGQKLRSSNMPVSGNQQNFGHPIASSDQATSMANG\n",
      "P08151                              PQPSPPQYLQSGPYTQPP---------PDYLPSEPRPCLDFDSPTHSTGQ\n",
      "                                         .   . .                                      \n",
      "\n",
      "P10070                              HRELGVPDSALAGVPPPHPVQSYPQQSHHLAASMSQEGYHQVPSLLPARQ\n",
      "Q0VGT2                              QREHGVPNSSLAAVSQPHPVLSYPQQ----------DSYQQGSNLLSSHQ\n",
      "Q91661                              APTQGFANNFSVNQDGLHPPNAYTVQ-------------PQKNGLEPQQN\n",
      "Q9IA31                              MQNRNVMEQEYLQNELVGDGIHYQGVN------------QSSPMTLGQVS\n",
      "P08151                              LKAQLVCNYVQSQQELLWEGGGREDAP---------------AQEPSYQS\n",
      "                                         . :                                         .\n",
      "\n",
      "P10070                              PGFMEPQTGPMGVATAGFGLVQPRPPLEPSPTGRHRGVRAVQQQLAYARA\n",
      "Q0VGT2                              PGFMESQQ------NAGFGLMQPRPPLEPNTASRHRGVRSGQQQL-YART\n",
      "Q91661                              TLGMSGQA-------FNHGMIQPRPPAAPHPPNRPRNIHPVHHPP-YMRS\n",
      "Q9IA31                              PTSQSSLH-------QGPQSCPPVSHTIGNQSSGLSVAKSYQPCANYSGN\n",
      "P08151                              PKFLGGSQ------------------VSPSRAKAPVNTYGPGFGPNLPNH\n",
      "                                    .                              .                  \n",
      "\n",
      "P10070                              TGHAMAAMPSSQETAEAVPKGAMGNMGSVPPQPPPQDAGGAPDHSMLYYY\n",
      "Q0VGT2                              TGQAMVTS-ANQETAEAMPKGPAGTMVSLAPQPS-QDTGRAQDQNTLYYY\n",
      "Q91661                              P-HPVSELSPGQQTAEATPKRTSENTDPTP-----------KDNNLLYYS\n",
      "Q9IA31                              RRQNVLRNNLAQQQGHVSDGNQTYRVNTIKLEMQ--GQSQQFCSNMQNYS\n",
      "P08151                              KSGSYPTPSPCHENFVVGANRASHRAAAPPRLLP----------PLPTCY\n",
      "                                               ::   .          .                      \n",
      "\n",
      "P10070                              GQIHMYEQDGGLENLGS-CQVMRSQPPQPQACQDSIQPQPLPSPGVNQVS\n",
      "Q0VGT2                              GQIHMYEQNGG-------CPAVQPQPPQPQACSDSIQPEPLPSPGVNQVS\n",
      "Q91661                              GQIHMYEPNGN--------------FGSGIDCTVRQLP-TMPSPGANQVT\n",
      "Q9IA31                              GQLYDQTTGFSHQAMKIGSSFFVSEANCLLQETATANSSELLSPGANQVS\n",
      "P08151                              GPLKVGGTNPS---------------CGHPEVGRLGGGPALYPPPEGQVC\n",
      "                                    * :     . .                             : .*  .** \n",
      "\n",
      "P10070                              STVD---SQLLE-APQIDFDAIMDDGDHSSLFSGALSPSLLHSLSQNSSR\n",
      "Q0VGT2                              STVD---SQLLE-PPQIDFDAIMDDGDHSSLFSGALSPTLLHNLSQNSSR\n",
      "Q91661                              STVD---SQGLEHPPVIDFDAMMDDGDHSSLMSGTLSPGLLQSFSQTSSR\n",
      "Q9IA31                              STVDSLDSNSLE-GVQIDFDAIIDDGDHVSLISGALSPSIIQNLSRNSSR\n",
      "P08151                              NPLD---SLDLD-NTQLDFVAILDE-------PQGLSPPPSHDQRGSSGH\n",
      "                                    ..:*   *  *:    :** *::*:       .  ***   :.   .*.:\n",
      "\n",
      "P10070                              LTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT\n",
      "Q0VGT2                              LTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT\n",
      "Q91661                              LTTPRNSLTLPSIPAGINNMAIGDMSSMLTTLAEESKFLNLMS\n",
      "Q9IA31                              LTTPRASLTFP--------------------------------\n",
      "P08151                              TPPPS----------GPPNMAVGNMSVLLRSLPGETEFLNSSA\n",
      "                                     ..*                                       \n",
      "\n",
      "\n",
      "\n",
      "-METSASATASEKQEAKSGILEAAGFPDPGKKASP------------------------------------------------------------------------LVVAAAAAAAVAAQGVPQHLLPPFHAPLPIDMRHQEGRYHYEPHSVHGVHGPPALSGSPVISDISLIRLSPH--PAGPGESPFNAPHPYVNPHMEHYLRSVHSSPTLSMISAARGLSPADVAQEHLKERGLFGLPAP---GTTPSDYYHQMTLVAGHPAPYGDLLMQSGGAASAPHLHDYLNPVDVSRFSSPRVTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGSYGHLSAGALSPAFTFPHPINPVAYQ----QILSQQRGLGSAFGHTPPLIQPSPTFLAQQPMALTSINATPTQLSSSS-NCLSDTNQNKQSSESAVSSTVNPVAIHKRSKVKTEPEGLRPASPLALTQGQVSGHGSCGCALPLSQEQLADLKEDLDRDDCKQEAEVVIYETNCHWEDCTKEYDTQEQLVHHINNEHIHGEKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRLENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKIPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHLRTPLLKENGDSEAGTEPGG------PESTEASSTSQAVEDCLHVRAIKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGVEMPGTGPGSLGDLTALDDTPPGADTSALAAPSAGGLQLRKHMTTMHRFEQLKKEK--LKSLKDSCSWAGPTPHTRNTKLPPLPGSGSILENFSGSGGGGPAGLLPNPRLSELSASEVTMLSQLQERRDSSTSTVSSAYTVSRRSSGISPYFSSRRSSEASPLGAGRPHNASSADSYDPISTDASRRSSEASQCSGG-SGLLNLTPAQQYSLRAKYAAATGG-PPPTPLPGLERMSLRTRLALLDAPERTLPAGCPRPLGPRRGSDGPTYGHGH----AGAAPAFPHEAPGGGARRASDPVRRP---DALSLPRVQRFHSTHNVNPGPLPPCADRRGLRLQSHPSTDGGLARGAYSPRPPSISENVAMEAVAAGVDGAGPEADLGLPEDDLVLPDDVVQYIKAHASGALDEGTGQVYPTESTGFSDNPRLPSPGLHGQRRMVAADSNVGPSAPMLGGCQLGFGAPSSLNKNNMPVQWNEVSSGTVDALASQVKPPPFPQGNLAVVQQKPAFGQYPGYSPQGLQASPGGLDSTQPHLQPRSGAPSQGIPRVNYMQQLRQPVAGSQCPGMTTTMSPHACYGQVHPQLSPSTISGALNQFPQ--SCSNMPAKPGHLGHPQQTEVAPDPTTMGNRHRELGVPDSALAGVPPPHPVQSYPQQSHHLAASMSQEGYHQVPSLLPARQPGFMEPQTGPMGVATAGFGLVQPRPPLEPSPTGRHRGVRAVQQQLAYARATGHAMAAMPSSQETAEAVPKGAMGNMGSVPPQPPPQDAGGAPDHSMLYYYGQIHMYEQDGGLENLGS-CQVMRSQPPQPQACQDSIQPQPLPSPGVNQVSSTVD---SQLLE-APQIDFDAIMDDGDHSSLFSGALSPSLLHSLSQNSSRLTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT - P10070\n",
      "-METSAPAPALEKKEAKSGLLEDSSFPDPGKKACP------------------------------------------------------------------------LAVA----AAVAAHGVPQQLLPAFHAPLPIDMRHQEGRYHYDPHSVHSVHGPPTLSGSPVISDISLIRLSPH--PAGPGESPFSAHHPYVNPHMEHYLRSVHSSPTLSMISAARGLSPADVAHEHLKERGLFSLAAP---GTNPSDYYHQMTLMASHPTPYGDLLMQSGGAASAPHLHDYLNPVDASRFSSPRVTPRLSRKRALSISPLSDASLDLQRMIRTSPNSLVAYINNSRSSSAASGSYGHLSAGALSPAFTFPHPINPVAYQ----QILSQQRGLGSAFGHTPPLIQPSPTFLAQQPMTLTSISTMPTQLSSSSSNCLNDANQNKQNSESAVSSTVNPITIHKRSKVKTEAEGLRPASPLGLTQ-----------------EQLADLKEDLDRDDCKQEAEVVIYETNCHWADCTKEYDTQEQLVHHINNEHIHGEKKEFVCRWQACTREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCSKAYSRLENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYICKIPGCTKRYTDPSSLRKHVKTVHGPDAHVTKKQRNDVHVRAPLLKENGDNEASAEPGGRGP---EESVEASSTSHTVEDCLHIKAIKTESSGLCQSSPGAQSSCSSEPSPLGSAPNNDSGMEMPGTGPGSLGDLTALADTCPGADTSALAAPSTGGLQLRKHMSTVHRFEQLKREK--LKSLKDSCSWAGPAPHTRNTKLPPLPVNGSVLENFNNTGGGGPAGLLPSQRLPEL--TEVTMLSQLQERRDSSTSTMSSAYTVSRRSSGISPYFSSRRSSEASPLGGLRPHNASSADSYDPISTDASRRSSEASQCSGGGPGLLNLTPAQQYNLRAKYAAATGG-PPPTPLPGLDRVSLRTRLALLDAPERALPGACPHPLGPRRGSDGPTYSHGHGHGYAGAAPAFPHEGPNSSTRRASDPVRRP---DPLILPRVQRFHSTHNMNPGSLPPCADRRGLHVQSHPSVDSNLTRNAYSPRPPSINENVVMEAVAAGVDGPGLECDLGLVEDELVLPDDVVQYIKAHTGGTLDDGIRQGYPTEGTGFPENSKLPSPGLQGHRRLAAADSNMGPSAPGLGGCQLSYSPSSNLNKSNMPVQWNEVSSGTVDALPTQVKPPPFPHSNLAVVQQKPAFGQYPGYNPQSVQSSSGGLDSTQPHLQLR-GAPS--ASRGSYTQQPRQPAAGSQCLGMSAAMSPQASYSQAHPQLSPNIVSGSLNQFSP--SCSNMAAKPSHLGLPQQMEVVPNATIMNGHQREHGVPNSSLAAVSQPHPVLSYPQQ----------DSYQQGSNLLSSHQPGFMESQQ------NAGFGLMQPRPPLEPNTASRHRGVRSGQQQL-YARTTGQAMVTS-ANQETAEAMPKGPAGTMVSLAPQPS-QDTGRAQDQNTLYYYGQIHMYEQNGG-------CPAVQPQPPQPQACSDSIQPEPLPSPGVNQVSSTVD---SQLLE-PPQIDFDAIMDDGDHSSLFSGALSPTLLHNLSQNSSRLTTPRNSLTLPSIPAGISNMAVGDMSSMLTSLAEESKFLNMMT - Q0VGT2\n",
      "--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------MEHYLRSVHNSPTLSMISAARGLSPAEVAHEHLKERGIYGLAPPPPPGTTPTEYCHQMAFLASHASPYGELLVQSAAAGNTSHLHDYLTPMDVSRFSSPRVTPRLSRKRALSISPLSDASIDLQTMIRTSPNSLVAYINNSRSSSAASGSYGHLSAGAISPAFSFPIHK-PCSLS----AALSQQRSLSSSFGHTP-LLHPSPTFASRQQGALTSANPAPPSNNSSA----PDSVLNKVSSESAVSSTVN-QVIHKRSKVKTEEE----ADSVRFPQ---------------PPDHLTDLKEDLDKDECKQEPE-HIYETNCHWDGCSKEFDTQDQLVHHINNDHIHGEKKEFVCRWQDCSREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCFKAYSRLENLKTHLRSHTGEKPYVCDHEGCNKAFSNASDRAKHQNRTHSNEKPYICKVPGCTKRYTDPSSLRKHVKTVHGPEAHVTKKHRNDIIQKPSLPKENGDNEASAKLSG--------REHSDSVSRDQEHCLQTRTIKTEDNMMHQSSPGGQSSCSSEPSPYGNTNNIDSGVDVSLAWQGSLGDLFGLEETSPVVDSTVSSWQRSG-RPATPETQRIHSAETGTAEEREIKDNERFLLIYEPNATCQNTRLPTISANG-----FDVIG--VPSSVLINPRAIELSMNDVTMMNQLNERRDSTSSTLSSAYTS-RRSSGISPYFSSRRSSETSQFGG-RLNNSSSADSYDPISTDASRRSSEASQHSGL-PNLLNLTPAQHYRLKAKYAAATGG-PPPTPLPNMDRIGLRNKLSLMDGADFPLPPFRQLPV-PRRCSDGGGN--------AGLTPMYPHEIPGNNSRRASDPVRRTAGIDDKPLPRFSRFHSMNSMNTLHPPSLSERRNGGLQHYTCSDGGLHRHVYSPRPPSISENVAMEAISCDADVPG-------GDDDLMLPDDVVQYIRSQNREAPEQNLQTEYSSPARNLQSNTKSFHNNTPEQPR--APGAYLSRNFPALAECLG---QTANMQDNNMPVQWNEVSSGTVDVSDLPKQQ--FAAGNLAVVQQKQNFAQYQSFNQAPMQRAHNIMGQGQESVQRNISVNG---QRFNYLQQRQQQMS--QCQIVSSDFIPQQRYSQSQSMLSSRAMQEGQSQISP--SCNNMVERPGVHTHAAPSNTLNHQRLAVHGAPTQGFANNFSVNQDGLHPPNAYTVQ-------------PQKNGLEPQQNTLGMSGQA-------FNHGMIQPRPPAAPHPPNRPRNIHPVHHPP-YMRSP-HPVSELSPGQQTAEATPKRTSENTDPTP-----------KDNNLLYYSGQIHMYEPNGN--------------FGSGIDCTVRQLP-TMPSPGANQVTSTVD---SQGLEHPPVIDFDAMMDDGDHSSLMSGTLSPGLLQSFSQTSSRLTTPRNSLTLPSIPAGINNMAIGDMSSMLTTLAEESKFLNLMS - Q91661\n",
      "MEAQSHSSTTTEKKKVENSIVKCSNRTDVSEKAVASSTTSNEDESPGQTYHRERRNAITMQPQGGQGLSKISEEPSTSSEERASLIKKEIHGSISHLPEPSVPYRGTLVTMDPRNGYMDPHYHPPHLFQAFHPPVPIDARHHEGRYHYEPSPIPPLHVPSALSSSPTYSDLPFIRISPHRNPAAASESPFSTPHPYINPYMD-YIRSLHSSPSLSMISAARGLSPTDAPHA----------------GVSPAEYYHQMALLAGQRSPYADIIPSAATAGAGALHMEYLHAMDSTRFPSPRLSARPSRKRTLSISPLSDHSFDLQTMIRTSPNSLVTILNNSCSSWSASGSYGHLSASAISPALSFTYPPTPVSLQQMHQQIISRQQTLGSAFGHSPPLIHPAPTFPTQRPIPGIPSVLNPVQVSSGP---SESTQHNKPTSESAVSSTGD-PMHNKRSKIKPDEDLPSPGAGSVQEQ----------------PEGMTPVKEEGDKDESKQEPE-VVYETNCHWEGCSREFDTQEQLVHHINNDHIHGEKKEFVCRWLDCSREQKPFKAQYMLVVHMRRHTGEKPHKCTFEGCTKAYSRLENLKTHLRSHTGEKPYVCEHEGCNKAFSNASDRAKHQNRTHSNEKPYVCKIPGCTKRYTDPSSLRKHVKTVHGPEAHVTKKQRGDIHPRPPPPRDPGSHSQTRSPGQQTQGATGEQKDLNSTTSRREECLQVKAVKSEKPMTSQPSPGGQSTCSSEQSPISNYSNN--GIELTLTGGGSVGDLSVIDET-PIMDSTISTATTALGLQARRNMTGTKWMEQVKLER--LKQVNGMLPRLNPVPPSKAPTLPPLIGNG----TQSNSSCSVGGSMTILPNRNELSSTDITVLNMLN-RRDSNTSTISSAYLSSRRSSGISPCFSSRRSSDASQAEG-RPQNVSVADSYDPISTDASRRSSEASQCDDL-PSLLSLTPAQQYRLKAKYAAATGG-PPPTPLPNMERMSLKTRMALLGDCRESGISPLPPVNAPRRCSDGGANG-------YSRRHLLSHDALGNGTRRASDPVRMVS--NNLSVPRVQHFNSLNSFNSPALPPSMEKRNLVLQNCTHSEGGVFRGFSSPCPPSISENVTLEAVTMEAGGSLN--------DEDLLPDDVVQYLNSQNQGTCDHLLNNVLDSN--KMHHSVVLGNNNPSSFDR--APPASSQP-------------AGSEVSKSDLPIQWNEVSSGSSDLSPTKLKC-----SQRSAVQQARAFGLYNNMMVQQQNLQRGNVYQ-QNGYQNLMENNG----SYSLQQNTVLGSGASSSFGMQPNKPYGESVSRQPMIFGAMDSSCGITVQGQKLRSSNMPVSGNQQNFGHPIASSDQATSMANGMQNRNVMEQEYLQNELVGDGIHYQGVN------------QSSPMTLGQVSPTSQSSLH-------QGPQSCPPVSHTIGNQSSGLSVAKSYQPCANYSGNRRQNVLRNNLAQQQGHVSDGNQTYRVNTIKLEMQ--GQSQQFCSNMQNYSGQLYDQTTGFSHQAMKIGSSFFVSEANCLLQETATANSSELLSPGANQVSSTVDSLDSNSLE-GVQIDFDAIIDDGDHVSLISGALSPSIIQNLSRNSSRLTTPRASLTFP-------------------------------- - Q9IA31\n",
      "------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------MFNSMTPPPISSYGEPCCLRP------------------------LPSQGAPSVGTEGLSGPPFCHQANLMSGPHSYGPARETNSCTEGPLFSSPRSAVKLTKKRALSISPLSDASLDLQTVIRTSPSSLVAFINSRCTS--PGGSYGHLSIGTMSPSLGFPAQMNHQKGP--------------------------SPSFGVQPCGPHDSARGGMIPHPQSR-----------GPFPTCQLKSELDMLVGKCREEPLEGDMSSPNSTGIQDP----------------LLGMLDGREDLEREE-KREPE-SVYETDCRWDGCSQEFDSQEQLVHHINSEHIHGERKEFVCHWGGCSRELRPFKAQYMLVVHMRRHTGEKPHKCTFEGCRKSYSRLENLKTHLRSHTGEKPYMCEHEGCSKAFSNASDRAKHQNRTHSNEKPYVCKLPGCTKRYTDPSSLRKHVKTVHGPDAHVTKRHRGDGPLPRAPSISTVEPKREREGGPIRE--------------------ESRLTVPEGAMKPQPSPGAQSSCSSDHSPAGSAANTDSGVEMTGNAGGSTEDLSSLDEGPCIAG---------------TGLSTLRRLENLRLDQ---------LHQLRPIG-TRGLKLPSLSHTG-------------------------TTVSRRVGPPVSLERRSSSSSSISSAYTVSRRSSLASPFPP-------------------------------------GSPPENGASSLPGLMPAQHYLLRARYASARGGGTSPTAASSLDRIGGLPMPPWRSRAEYP-----------------------------------GYNPNAGVTRRASDPAQAA---DRPAPARVQRFKSLGCVHTPPTVAGGGQ---------NFDPYLPTSVYSPQPPSITENAAMDARGLQE-----------------EPEVGTSMVGSGLNPYMDFPPTDTLGYGGPEGAAAEPYGARGPGSLPLGPGPPTNYGP------------------NPCPQQASYPDPTQETWGEFPSHSGLYPGPKALGGTYSQCPRLEHYGQVQVKPEQGCP--VGSDSTGLAPCLNAHP-----------------------SEGPPHPQPLFSHYPQPSPPQYLQSGPYTQPP---------PDYLPSEPRPCLDFDSPTHSTGQLKAQLVCNYVQSQQELLWEGGGREDAP---------------AQEPSYQSPKFLGGSQ------------------VSPSRAKAPVNTYGPGFGPNLPNHKSGSYPTPSPCHENFVVGANRASHRAAAPPRLLP----------PLPTCYGPLKVGGTNPS---------------CGHPEVGRLGGGPALYPPPEGQVCNPLD---SLDLD-NTQLDFVAILDE-------PQGLSPPPSHDQRGSSGHTPPPS----------GPPNMAVGNMSVLLRSLPGETEFLNSSA - P08151\n",
      "      A     C     D     E     F     G     H     I     K     L     M     N     P     Q     R     S     T     V    W     Y\n",
      "A 386.0   8.5  14.0  18.5   9.0  61.0  11.5  16.0   4.5  29.5  15.5  22.5  82.5  22.5  17.5  75.0  57.5  43.5  2.0  11.0\n",
      "C   8.5 142.0   2.0   2.0   0.5  10.5   4.0   1.0   1.5   8.0   2.5   4.0  13.5   4.5  11.0  17.5   2.5   4.0  0.0   5.5\n",
      "D  14.0   2.0 306.0  89.5   0.5  24.0   4.5   1.0   2.0   3.5   1.5  18.0  18.0   5.5   6.0  18.0   6.0   7.5  0.0   1.5\n",
      "E  18.5   2.0  89.5 349.0   2.0  26.0   8.0   1.5  13.5  10.5   2.0  10.5  14.0  12.5   5.0  17.5   9.0  11.5  0.0   1.0\n",
      "F   9.0   0.5   0.5   2.0 143.0   8.0   4.0   3.0   1.0  22.5   6.5   0.0   8.5   9.0   4.5  10.0   3.0  10.0  0.0  14.0\n",
      "G  61.0  10.5  24.0  26.0   8.0 425.0  19.5   9.0   8.0  42.0  11.0  57.5  55.5  33.5  15.0  95.5  21.0  28.0  2.0  11.5\n",
      "H  11.5   4.0   4.5   8.0   4.0  19.5 277.0   5.0   6.0  14.5   4.0  18.0  31.0  50.0  16.5  18.0   3.5   6.5  1.5   7.5\n",
      "I  16.0   1.0   1.0   1.5   3.0   9.0   5.0 137.0   2.5  43.5  14.5   3.5  13.5   7.0   3.0   9.5   9.5  35.0  1.0   6.0\n",
      "K   4.5   1.5   2.0  13.5   1.0   8.0   6.0   2.5 288.0   5.0   2.5   5.0  12.0  10.0  45.0   8.0   6.5   5.0  0.0   1.0\n",
      "L  29.5   8.0   3.5  10.5  22.5  42.0  14.5  43.5   5.0 523.0  46.5  17.5  57.0  36.0  19.0  35.5  33.0  40.5  2.0  16.0\n",
      "M  15.5   2.5   1.5   2.0   6.5  11.0   4.0  14.5   2.5  46.5 118.0   4.0  11.0   9.5   4.5  18.5  11.5  34.5  0.0   4.5\n",
      "N  22.5   4.0  18.0  10.5   0.0  57.5  18.0   3.5   5.0  17.5   4.0 247.0  30.5  24.5  14.0  85.0  27.5   9.5  0.0   7.5\n",
      "P  82.5  13.5  18.0  14.0   8.5  55.5  31.0  13.5  12.0  57.0  11.0  30.5 677.0  66.0  20.0  95.5  56.5  22.5  0.0   5.0\n",
      "Q  22.5   4.5   5.5  12.5   9.0  33.5  50.0   7.0  10.0  36.0   9.5  24.5  66.0 338.0  31.0  52.0  15.5  21.0  1.0   4.0\n",
      "R  17.5  11.0   6.0   5.0   4.5  15.0  16.5   3.0  45.0  19.0   4.5  14.0  20.0  31.0 420.0  29.0  25.0  11.5  2.5   4.0\n",
      "S  75.0  17.5  18.0  17.5  10.0  95.5  18.0   9.5   8.0  35.5  18.5  85.0  95.5  52.0  29.0 878.0 100.0  29.0  2.5  11.0\n",
      "T  57.5   2.5   6.0   9.0   3.0  21.0   3.5   9.5   6.5  33.0  11.5  27.5  56.5  15.5  25.0 100.0 301.0  25.5  0.0   7.5\n",
      "V  43.5   4.0   7.5  11.5  10.0  28.0   6.5  35.0   5.0  40.5  34.5   9.5  22.5  21.0  11.5  29.0  25.5 234.0  1.5   8.5\n",
      "W   2.0   0.0   0.0   0.0   0.0   2.0   1.5   1.0   0.0   2.0   0.0   0.0   0.0   1.0   2.5   2.5   0.0   1.5 27.0   2.0\n",
      "Y  11.0   5.5   1.5   1.0  14.0  11.5   7.5   6.0   1.0  16.0   4.5   7.5   5.0   4.0   4.0  11.0   7.5   8.5  2.0 211.0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import AlignIO\n",
    "align = AlignIO.read(\"BLASTp_new.aln\", \"clustal\")\n",
    "print(align)\n",
    "print(\"Número de linhas: %i\" % len(align)) \n",
    "print(format(align, \"clustal\"))\n",
    "\n",
    "for record in align:\n",
    "    print(\"%s - %s\" % (record.seq, record.id)) \n",
    "\n",
    "#ver com que frequência as letras são substituidas\n",
    "subs = align.substitutions\n",
    "print(subs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "50d6a250",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Convertidos 1 alinhamentos\n"
     ]
    }
   ],
   "source": [
    "count = AlignIO.convert(\"BLASTp_new.aln\", \"clustal\",\"BLASTp_new.sth\", \"stockholm\") \n",
    "print (\"Convertidos %i alinhamentos\" % count )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "a7bc57d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "alignment = AlignIO.read(\"BLASTp_new.sth\", \"stockholm\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "911a1d5d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "P10070\t0\n",
      "Q0VGT2\t0.14710947109471095\t0\n",
      "Q91661\t0.4521350823332403\t0.45592151366503153\t0\n",
      "Q9IA31\t0.5736807817589576\t0.5734578455872682\t0.5834924364811279\t0\n",
      "P08151\t0.7367896311066799\t0.7434188603798734\t0.7457283031635933\t0.7485773409208485\t0\n",
      "\tP10070\tQ0VGT2\tQ91661\tQ9IA31\tP08151\n"
     ]
    }
   ],
   "source": [
    "calculator = DistanceCalculator('blosum62')\n",
    "dm = calculator.get_distance(alignment)\n",
    "print(dm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "eb918ec1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tree(rooted=True)\n",
      "    Clade(branch_length=0, name='Inner4')\n",
      "        Clade(name='P08151')\n",
      "        Clade(name='Inner3')\n",
      "            Clade(name='Q9IA31')\n",
      "            Clade(name='Inner2')\n",
      "                Clade(name='Q91661')\n",
      "                Clade(name='Inner1')\n",
      "                    Clade(name='Q0VGT2')\n",
      "                    Clade(name='P10070')\n"
     ]
    }
   ],
   "source": [
    "constructor = DistanceTreeConstructor() \n",
    "upgmatree = constructor.upgma(dm)\n",
    "print(upgmatree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "75418271",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tree(rooted=False)\n",
      "    Clade(branch_length=0, name='Inner3')\n",
      "        Clade(name='Inner2')\n",
      "            Clade(name='Inner1')\n",
      "                Clade(name='P10070')\n",
      "                Clade(name='Q0VGT2')\n",
      "            Clade(name='Q91661')\n",
      "        Clade(name='Q9IA31')\n",
      "        Clade(name='P08151')\n"
     ]
    }
   ],
   "source": [
    "njtree = constructor.nj(dm)\n",
    "print(njtree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "763fddd3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Phylo.write([upgmatree, njtree],\"phylotree_GLI2.nhx\",\"newick\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "9335a59d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                  __________ P10070\n",
      "           ______________________|\n",
      "  ________|                      |__________ Q0VGT2\n",
      " |        |\n",
      " |        |__________________________________ Q91661\n",
      "_|\n",
      " |___________________________________________ Q9IA31\n",
      " |\n",
      " |_____________________________________________________________________ P08151\n",
      "\n"
     ]
    }
   ],
   "source": [
    "Phylo.draw_ascii(njtree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "08876f9f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  _____________________________________________________________________ P08151\n",
      "_|\n",
      " |               ______________________________________________________ Q9IA31\n",
      " |______________|\n",
      "                |            __________________________________________ Q91661\n",
      "                |___________|\n",
      "                            |                             _____________ Q0VGT2\n",
      "                            |____________________________|\n",
      "                                                         |_____________ P10070\n",
      "\n"
     ]
    }
   ],
   "source": [
    "Phylo.draw_ascii(upgmatree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d3710b7",
   "metadata": {},
   "outputs": [],
   "source": [
    " "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
